<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>EMBL-EBI (RNA-seq)</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="embl_ebi_rna-seq_files/libs/clipboard/clipboard.min.js"></script>
<script src="embl_ebi_rna-seq_files/libs/quarto-html/quarto.js"></script>
<script src="embl_ebi_rna-seq_files/libs/quarto-html/popper.min.js"></script>
<script src="embl_ebi_rna-seq_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="embl_ebi_rna-seq_files/libs/quarto-html/anchor.min.js"></script>
<link href="embl_ebi_rna-seq_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="embl_ebi_rna-seq_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="embl_ebi_rna-seq_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="embl_ebi_rna-seq_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="embl_ebi_rna-seq_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<link href="embl_ebi_rna-seq_files/libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="embl_ebi_rna-seq_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="embl_ebi_rna-seq_files/libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="embl_ebi_rna-seq_files/libs/typedarray-0.1/typedarray.min.js"></script>
<script src="embl_ebi_rna-seq_files/libs/jquery-3.5.1/jquery.min.js"></script>
<link href="embl_ebi_rna-seq_files/libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="embl_ebi_rna-seq_files/libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="embl_ebi_rna-seq_files/libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet">
<script src="embl_ebi_rna-seq_files/libs/plotly-main-2.11.1/plotly-latest.min.js"></script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">EMBL-EBI (RNA-seq)</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This data used here is from (<a href="https://www.sciencedirect.com/science/article/pii/S2211124723014043">Yu et al., 2023</a>).</p>
<p>The study performed transcriptomic profiling of brain, lung and liver endothelial cells (ECs) from adult mice (2-4 months).</p>
<section id="load-libraries" class="level2">
<h2 class="anchored" data-anchor-id="load-libraries">Load libraries</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tximport)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gprofiler2)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(clusterProfiler)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(enrichplot)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DOSE)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(apeglm)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(biomaRt)</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(memes)</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TFBSTools)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(JASPAR2024)</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(universalmotif)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vsn)</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(grid)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="load-rna-seq-data" class="level2">
<h2 class="anchored" data-anchor-id="load-rna-seq-data">Load RNA-seq data</h2>
<p>Load the data frame that contains the mappings from transcript IDs to gene IDs and gene names.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define file path to the tx2gene.tsv file</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># here::here() constructs file paths relative to the root of the R project</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>tx2genefile <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">'data/count_tables/tx2gene.tsv'</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the contents of the .tsv file, selecting only the first three columns of data</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>tx2gene <span class="ot">=</span> <span class="fu">read.delim</span>(tx2genefile, <span class="at">header=</span><span class="cn">FALSE</span>)[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(tx2gene))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">V1</th>
<th style="text-align: left;">V2</th>
<th style="text-align: left;">V3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUST00000193812</td>
<td style="text-align: left;">ENSMUSG00000102693</td>
<td style="text-align: left;">4933401J01Rik</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUST00000082908</td>
<td style="text-align: left;">ENSMUSG00000064842</td>
<td style="text-align: left;">Gm26206</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUST00000162897</td>
<td style="text-align: left;">ENSMUSG00000051951</td>
<td style="text-align: left;">Xkr4</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUST00000159265</td>
<td style="text-align: left;">ENSMUSG00000051951</td>
<td style="text-align: left;">Xkr4</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUST00000070533</td>
<td style="text-align: left;">ENSMUSG00000051951</td>
<td style="text-align: left;">Xkr4</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUST00000192857</td>
<td style="text-align: left;">ENSMUSG00000102851</td>
<td style="text-align: left;">Gm18956</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The file, <code>tx2gene.tsv</code>, is a common format used in RNA-seq data analysis, especially when working with transcript quantification tools like Salmon, Kallisto, or other tools that assign read counts to transcripts. This file is used to map transcript IDs to gene IDs and gene names. There are three columns:</p>
<ul>
<li><p><strong>Transcript ID</strong>: contains transcript identifiers - each transcript ID corresponds to a specific mRNA transcript produced by a gene.</p></li>
<li><p><strong>Gene ID</strong>: contains gene identifiers - these IDs represent the genes associated with the transcripts in column 1.</p></li>
<li><p><strong>Gene name</strong>: contains gene names.</p></li>
</ul>
<p>Create a list of <code>quant.sf</code> files in the <code>data/count_tables</code> directory.</p>
<p><code>quant.sf</code> files are output files from tools like Salmon or Kallisto, which contain transcript quantification data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path to the count_tables directory</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>directory_path <span class="ot">=</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"data/count_tables"</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># List all files in the count_tables directory (and subdirectories) named quant.sf</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># recurvive=TRUE allows the function to search subdirectories</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co"># full.names=TRUE returns the full file path, not just file names</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>files <span class="ot">=</span> <span class="fu">list.files</span>(<span class="at">path=</span>directory_path, <span class="at">pattern=</span><span class="st">"quant.sf"</span>, <span class="at">recursive=</span><span class="cn">TRUE</span>, <span class="at">full.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>files</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/BR1-EC_bRNA/quant.sf"
[2] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/BR2-EC_bRNA/quant.sf"
[3] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/BR3-EC_bRNA/quant.sf"
[4] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/LG1-EC_bRNA/quant.sf"
[5] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/LG2-EC_bRNA/quant.sf"
[6] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/LG3-EC_bRNA/quant.sf"
[7] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/LV1-EC_bRNA/quant.sf"
[8] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/LV2-EC_bRNA/quant.sf"
[9] "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/LV3-EC_bRNA/quant.sf"</code></pre>
</div>
</div>
<p>Using <code>tximport()</code>, import and aggregate the transcript quantification data (from Salmon output files) into gene-level counts.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># tximport() imports and aggregates transcript-level abundance estimates (like those produced by Salmon/Kallisto) into gene-level counts</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># files is a list of file paths to the quantification files</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co"># tx2gene=tx2gene provide the mapping between transcipts and genes</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>txi <span class="ot">=</span> <span class="fu">tximport</span>(files, <span class="at">type=</span><span class="st">"salmon"</span>, <span class="at">tx2gene=</span>tx2gene)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>reading in files with read_tsv</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1 2 3 4 5 6 7 8 9 
summarizing abundance
summarizing counts
summarizing length</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(txi) <span class="co"># txi is a list containing matricies of abundance, counts, lengths and countsFromAbundance</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "abundance"           "counts"              "length"             
[4] "countsFromAbundance"</code></pre>
</div>
</div>
<p>The <code>txi</code> object contains several components:</p>
<ul>
<li><p><strong>Abundance</strong>: contains transcript abundance estimates, typically expressed as TPM (Transcripts per Million) or FPKM/RPKM (Fragments/Kilobases per Million reads). TPM and FPKM/RPKM are normalised measures that account for both transcript length and library size, making them useful for comparing expression levels across samples.</p></li>
<li><p><strong>Counts</strong>: represents the raw number of reads or fragments that have been assigned to each gene.</p></li>
<li><p><strong>Length</strong>: contains the effective lengths of the transcripts (the effective length accounts for length of the transcript that is actually mappable, which might be shorter than the full transcript length). Effective lengths are used in the calculation of TPM or FPKM/RPKM.</p></li>
<li><p><strong>CountsFromAbundance</strong>: indicates how the counts were derived from the abundance estimates.</p>
<ul>
<li><code>"no"</code>: indicates that the counts are raw and have not been derived from the abundance values.</li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>txi<span class="sc">$</span>countsFromAbundance</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "no"</code></pre>
</div>
</div>
<p>Create a data frame that contains condition information for each sample.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract sample names from the file paths stored in the files variable</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># sapply() applies a function to each element of a list and simplifies the result into a vector or matrix</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># dirname(path) extracts the directory name from the full file path:</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>    <span class="co"># e.g. "/Users/laurahuggon/RProjects/embl_ebi_rna-seq/data/count_tables/BR1-EC_bRNA"</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="co"># basename(dirname(path)) then extracts the base name of this directory:</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>    <span class="co"># e.g. "BR1-EC_bRNA"</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">=</span> <span class="fu">sapply</span>(files, <span class="cf">function</span>(path) {<span class="fu">basename</span>(<span class="fu">dirname</span>(path))})</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the condition from the sample names</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the base directory name (the sample name) and uses the sub() function to replace any characters (.*) after "1-", "2-", or "3-" with an empty string</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>condition <span class="ot">=</span> <span class="fu">sapply</span>(files, <span class="cf">function</span>(path) {</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  folder_name <span class="ot">=</span> <span class="fu">basename</span>(<span class="fu">dirname</span>(path))</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sub</span>(<span class="st">"([123]-.*)"</span>, <span class="st">""</span>, folder_name)</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame to store the condition information</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co"># The resulting data frame has as many rows as there are files, each corresponding to a sample</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>data <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">condition=</span>condition)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Set the row names to the sample names</span></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(data) <span class="ot">=</span> samples</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">condition</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR2-EC_bRNA</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BR3-EC_bRNA</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG1-EC_bRNA</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LG2-EC_bRNA</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG3-EC_bRNA</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV1-EC_bRNA</td>
<td style="text-align: left;">LV</td>
</tr>
<tr class="even">
<td style="text-align: left;">LV2-EC_bRNA</td>
<td style="text-align: left;">LV</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV3-EC_bRNA</td>
<td style="text-align: left;">LV</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section id="data-exploration" class="level2">
<h2 class="anchored" data-anchor-id="data-exploration">Data exploration</h2>
<section id="remove-unexpressed-genes" class="level3">
<h3 class="anchored" data-anchor-id="remove-unexpressed-genes">Remove unexpressed genes</h3>
<p>Extract the raw counts matrix from <code>txi</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>counts_matrix <span class="ot">=</span> txi<span class="sc">$</span>counts</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign sample names as column names</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(counts_matrix) <span class="ot">=</span> <span class="fu">rownames</span>(data)</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(counts_matrix))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 14%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">BR1-EC_bRNA</th>
<th style="text-align: right;">BR2-EC_bRNA</th>
<th style="text-align: right;">BR3-EC_bRNA</th>
<th style="text-align: right;">LG1-EC_bRNA</th>
<th style="text-align: right;">LG2-EC_bRNA</th>
<th style="text-align: right;">LG3-EC_bRNA</th>
<th style="text-align: right;">LV1-EC_bRNA</th>
<th style="text-align: right;">LV2-EC_bRNA</th>
<th style="text-align: right;">LV3-EC_bRNA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000001</td>
<td style="text-align: right;">1570</td>
<td style="text-align: right;">1097</td>
<td style="text-align: right;">739</td>
<td style="text-align: right;">1085</td>
<td style="text-align: right;">105</td>
<td style="text-align: right;">426</td>
<td style="text-align: right;">857</td>
<td style="text-align: right;">926</td>
<td style="text-align: right;">1764</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000003</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000028</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">32</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000031</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">15</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000037</td>
<td style="text-align: right;">7</td>
<td style="text-align: right;">15</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000049</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">210</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Genes with low expression levels across all samples should be filtered out (e.g.&nbsp;not all mouse genes in the genome will be expressed in these samples). This removes noise from the data, making results more reliable and improves statistical power.</p>
<p>Determine the optimal filtering criteria.</p>
<p>Use Counts Per Million (CPM) to normalise for raw count variability (by expressing counts relative to the total number of reads in the sample). This is calculated by:</p>
<p><span class="math display">\[\text{CPM} = \left(\frac{\text{Raw Count}}{\text{Total Reads for Sample}}\right) \times 1{,}000{,}000\]</span></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the function to plot the number of genes retained based on CPM thresholds</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The min_samples parameter is used to define the minimum number of samples with CPM above min_cpm to keep a gene to be considered retained</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>plot_genes_retained_by_cpm <span class="ot">&lt;-</span> <span class="cf">function</span>(data, <span class="at">min_samples=</span><span class="dv">2</span>) {</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert raw counts to CPM (Counts Per Million)</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># 2 indicates that the operation should be performed across columns</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># colSums(data) calculates the sums of counts for each sample (i.e. sum for each column)</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Therefore, for each column in the data matrix, divide each entry by the total sum of that column</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  cpm <span class="ot">&lt;-</span> <span class="fu">sweep</span>(data, <span class="dv">2</span>, <span class="fu">colSums</span>(data), <span class="at">FUN =</span> <span class="st">"/"</span>) <span class="sc">*</span> <span class="fl">1e6</span></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Define a range of CPM thresholds to test (from 0 to 5 with increments of 0.1)</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>  thresholds <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">5</span>, <span class="at">by =</span> <span class="fl">0.1</span>)</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialise a vector to store the number of genes retained for each threshold</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Creates an empty numeric vector with the same length as thresholds</span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>  genes_retained <span class="ot">&lt;-</span> <span class="fu">numeric</span>(<span class="fu">length</span>(thresholds))</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Loop through each threshold value and calculate the number of genes retained</span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(thresholds)) { <span class="co"># Generates a sequence of indices from 1 to the length of thresholds</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>    min_cpm <span class="ot">&lt;-</span> thresholds[i] <span class="co"># Set min_cpm to the current threshold value being processed</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># cpm &gt; min_cpm creates a logical matrix where each element is TRUE if the CPM for that gene in that sample is greater than the current threshold (min_cpm)</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># rowSums(cpm &gt; min_cpm) calculates the number of samples for which each gene has a CPM greater than min_cpm - this creates a vector where each element represents the number of samples meeting the threshold for each gene</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Then create logical vector (mask) where each element is TRUE if the rowSums() element is greater than or equal to the min_samples requirement</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>    mask <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(cpm <span class="sc">&gt;</span> min_cpm) <span class="sc">&gt;=</span> min_samples</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Count the number of TRUE values in the mask vector, which corresponds to the number of genes retained for the current threshold</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># This count is stored at the index corresponding to the current threshold</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>    genes_retained[i] <span class="ot">&lt;-</span> <span class="fu">sum</span>(mask)</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create data frame used for plotting</span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>  plot_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">thresholds =</span> thresholds, <span class="at">genes_retained =</span> genes_retained)</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Plot the number of genes retained as a function of CPM threshold using ggplot2</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> thresholds, <span class="at">y =</span> genes_retained)) <span class="sc">+</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_line</span>(<span class="at">color =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span>  <span class="co"># Line plot for the gene counts</span></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"darkgreen"</span>) <span class="sc">+</span>  <span class="co"># Add points to the plot</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">1</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">label =</span> <span class="st">"CPM = 1"</span>) <span class="sc">+</span>  <span class="co"># Vertical line at CPM = 1</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Threshold (CPM)"</span>, <span class="at">y =</span> <span class="st">"Number of Genes Retained"</span>) <span class="sc">+</span></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_minimal</span>()</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_genes_retained_by_cpm</span>(counts_matrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in geom_vline(xintercept = 1, color = "red", linetype = "dashed", :
Ignoring unknown parameters: `label`</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This plot displays the number of genes retained as a function of different CPM thresholds. For many bulk RNA-seq datasets, a CPM threshold of 1 is a common filtering point.</p>
<p>Count the number of rows (genes) in <code>txi</code> before filtering.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(txi<span class="sc">$</span>counts) <span class="co"># 45,706 genes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 45706</code></pre>
</div>
</div>
<p>Here, we will filter genes with an expression threshold of &gt;0.75 CPM in at least two samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>filter_genes_by_cpm <span class="ot">&lt;-</span> <span class="cf">function</span>(data, <span class="at">min_cpm =</span> <span class="fl">0.75</span>, <span class="at">min_samples =</span> <span class="dv">2</span>) {</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>    <span class="co"># min_cpm: min counts per million (CPM) threshold for filtering genes.</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>    <span class="co"># min_sample: min number of samples with CPM above min_cpm to keep a gene.</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert raw counts to CPM (Counts Per Million)</span></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  cpm <span class="ot">&lt;-</span> <span class="fu">sweep</span>(data, <span class="dv">2</span>, <span class="fu">colSums</span>(data), <span class="at">FUN =</span> <span class="st">"/"</span>) <span class="sc">*</span> <span class="fl">1e6</span>  <span class="co"># Divide by total counts and multiply by 1M to get CPM</span></span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter genes based on CPM threshold</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create a logical mask where CPM &gt; min_cpm in at least min_samples samples</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  mask <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(cpm <span class="sc">&gt;</span> min_cpm) <span class="sc">&gt;=</span> min_samples</span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply the filter to retain only genes that pass the mask</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only keep the rows (genes) where the mask is TRUE</span></span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  filtered_data <span class="ot">&lt;-</span> data[mask, ]</span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(filtered_data)</span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the filtering to the counts matrix in txi</span></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a><span class="co"># filtered_genes contains only the genes that pass the filter</span></span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>filtered_genes <span class="ot">&lt;-</span> <span class="fu">filter_genes_by_cpm</span>(txi<span class="sc">$</span>counts)</span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Update txi to include only the filtered genes</span></span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a><span class="co"># The matrices are filtered to only keep the rows (genes) that are present in filtered_genes</span></span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>txi<span class="sc">$</span>abundance <span class="ot">&lt;-</span> txi<span class="sc">$</span>abundance[<span class="fu">rownames</span>(filtered_genes), ]</span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>txi<span class="sc">$</span>counts <span class="ot">&lt;-</span> txi<span class="sc">$</span>counts[<span class="fu">rownames</span>(filtered_genes), ]</span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>txi<span class="sc">$</span>length <span class="ot">&lt;-</span> txi<span class="sc">$</span>length[<span class="fu">rownames</span>(filtered_genes), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Count the number of rows (genes) after filtering.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(txi<span class="sc">$</span>counts) <span class="co"># 16,230 genes</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 16230</code></pre>
</div>
</div>
</section>
<section id="library-size-differences" class="level3">
<h3 class="anchored" data-anchor-id="library-size-differences">Library size differences</h3>
<p>Differences in the total number of reads (that have been assigned to genes) between samples typically occur for technical reasons. In practice, it means that we can not simply compare a geneâ€™s raw read count directly between samples and conclude that a sample with a higher read count also expresses the gene more strongly - the higher count may be caused by an overall higher number of reads in that sample.</p>
<p>Update the counts matrix from <code>txi</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>counts_matrix <span class="ot">=</span> txi<span class="sc">$</span>counts</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Assign sample names as column names</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(counts_matrix) <span class="ot">=</span> <span class="fu">rownames</span>(data)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(counts_matrix))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 14%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">BR1-EC_bRNA</th>
<th style="text-align: right;">BR2-EC_bRNA</th>
<th style="text-align: right;">BR3-EC_bRNA</th>
<th style="text-align: right;">LG1-EC_bRNA</th>
<th style="text-align: right;">LG2-EC_bRNA</th>
<th style="text-align: right;">LG3-EC_bRNA</th>
<th style="text-align: right;">LV1-EC_bRNA</th>
<th style="text-align: right;">LV2-EC_bRNA</th>
<th style="text-align: right;">LV3-EC_bRNA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000001</td>
<td style="text-align: right;">1570.000</td>
<td style="text-align: right;">1097</td>
<td style="text-align: right;">739</td>
<td style="text-align: right;">1085</td>
<td style="text-align: right;">105</td>
<td style="text-align: right;">426</td>
<td style="text-align: right;">857</td>
<td style="text-align: right;">926</td>
<td style="text-align: right;">1764.000</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000028</td>
<td style="text-align: right;">13.000</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">32.000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000049</td>
<td style="text-align: right;">0.000</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">210.000</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000056</td>
<td style="text-align: right;">262.000</td>
<td style="text-align: right;">74</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">82</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">427</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">107.000</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000058</td>
<td style="text-align: right;">1531.001</td>
<td style="text-align: right;">1449</td>
<td style="text-align: right;">1619</td>
<td style="text-align: right;">2517</td>
<td style="text-align: right;">479</td>
<td style="text-align: right;">1305</td>
<td style="text-align: right;">1300</td>
<td style="text-align: right;">1660</td>
<td style="text-align: right;">1412.001</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000078</td>
<td style="text-align: right;">8184.000</td>
<td style="text-align: right;">8183</td>
<td style="text-align: right;">7587</td>
<td style="text-align: right;">4709</td>
<td style="text-align: right;">359</td>
<td style="text-align: right;">1711</td>
<td style="text-align: right;">1772</td>
<td style="text-align: right;">3543</td>
<td style="text-align: right;">1758.000</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Compare the library sizes of all samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate library sizes (total counts per sample) by summing the counts for each column</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>libSize <span class="ot">&lt;-</span> <span class="fu">colSums</span>(counts_matrix)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a data frame for plotting</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">sample =</span> <span class="fu">names</span>(libSize),</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">libSize =</span> libSize</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with group information</span></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">=</span> <span class="fu">cbind</span>(plot_data, data)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove row names</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(plot_data) <span class="ot">=</span> <span class="cn">NULL</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(plot_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">sample</th>
<th style="text-align: right;">libSize</th>
<th style="text-align: left;">condition</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: right;">15136134</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR2-EC_bRNA</td>
<td style="text-align: right;">16196362</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BR3-EC_bRNA</td>
<td style="text-align: right;">20279540</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG1-EC_bRNA</td>
<td style="text-align: right;">9094897</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LG2-EC_bRNA</td>
<td style="text-align: right;">2283948</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG3-EC_bRNA</td>
<td style="text-align: right;">5551379</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV1-EC_bRNA</td>
<td style="text-align: right;">21208832</td>
<td style="text-align: left;">LV</td>
</tr>
<tr class="even">
<td style="text-align: left;">LV2-EC_bRNA</td>
<td style="text-align: right;">28461320</td>
<td style="text-align: left;">LV</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV3-EC_bRNA</td>
<td style="text-align: right;">24075596</td>
<td style="text-align: left;">LV</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot library sizes</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(plot_data, <span class="fu">aes</span>(<span class="at">x =</span> sample, <span class="at">y =</span> libSize <span class="sc">/</span> <span class="fl">1e6</span>, <span class="at">fill =</span> condition)) <span class="sc">+</span> </span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>) <span class="sc">+</span> </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"Total count in millions"</span>) <span class="sc">+</span> </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This plot helps visualise the overall sequencing depth/library size across the samples. Ideally, the sequencing depth should be consistent across samples.</p>
</section>
<section id="variability-and-distribution-of-gene-expression-counts" class="level3">
<h3 class="anchored" data-anchor-id="variability-and-distribution-of-gene-expression-counts">Variability and distribution of gene expression counts</h3>
<p>Use a box and whisker plot to assess the variability and distribution of gene expression counts across the samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Log transform the counts</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>log_counts_matrix <span class="ot">=</span> <span class="fu">log2</span>(counts_matrix <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Reshape the counts matrix (convert to long format)</span></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>counts_long <span class="ot">=</span> <span class="fu">melt</span>(log_counts_matrix)</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename columns</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(counts_long) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Gene"</span>, <span class="st">"Sample"</span>, <span class="st">"Count"</span>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create condition data frame</span></span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract sample names (which were previously set as row names)</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>condition <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Sample=</span><span class="fu">rownames</span>(data), <span class="at">condition=</span>data<span class="sc">$</span>condition)</span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge with condition information</span></span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>counts_long <span class="ot">&lt;-</span> <span class="fu">merge</span>(counts_long, condition, <span class="at">by=</span><span class="st">"Sample"</span>)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(counts_long))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;">Sample</th>
<th style="text-align: left;">Gene</th>
<th style="text-align: right;">Count</th>
<th style="text-align: left;">condition</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">ENSMUSG00000000001</td>
<td style="text-align: right;">10.617468</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">ENSMUSG00000000028</td>
<td style="text-align: right;">3.807355</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">ENSMUSG00000000049</td>
<td style="text-align: right;">0.000000</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">ENSMUSG00000000056</td>
<td style="text-align: right;">8.038919</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">ENSMUSG00000000058</td>
<td style="text-align: right;">10.581202</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">ENSMUSG00000000078</td>
<td style="text-align: right;">12.998767</td>
<td style="text-align: left;">BR</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot box and whisker</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(counts_long, <span class="fu">aes</span>(<span class="at">x =</span> Sample, <span class="at">y =</span> Count, <span class="at">fill =</span> condition)) <span class="sc">+</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span>  <span class="co"># Avoid showing outliers</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Sample"</span>, <span class="at">y =</span> <span class="st">"log2(counts + 1)"</span>) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This plot shows the log2 transformed gene expression counts per sample. The boxes represent the interquartile range (middle 50% of the data) and the whiskers represent variability outside the upper and lower quartiles.</p>
<p>Ideally, the boxes and whiskers should be of similar sizes across the samples, indicating a consistent data distribution. By comparing the boxes across samples, you can see if certain samples have higher or lower expression levels overall. If the median varies significantly between samples, this suggests some samples have more overall expression than others. If the spread/height of the boxes (IQR) differs between samples, this suggests these samples have greater variability in gene expression.</p>
</section>
</section>
<section id="differential-expression-analysis-dea" class="level2">
<h2 class="anchored" data-anchor-id="differential-expression-analysis-dea">Differential Expression Analysis (DEA)</h2>
<p>Use the <code>DESeq2</code> library to perform differential gene expression analysis and normalise counts for making plots such as PCA, heatmaps, etc.</p>
<p>Create a DESeqDataSet object using the <code>txi</code> data. This is the primary data structure used by the DESeq2 package for DEA.</p>
<p>The <code>dds</code> object is created using <code>DESeqDataSetFromTximport()</code> function, which takes the imported transcipt data (<code>txi</code>), sample metadata (<code>data</code>), and a design formula (<code>condition</code>) as inputs.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># colData is a data frame that contains metadata about the samples</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="co"># design=~condition specifies that the DEA will be based on the condition variable, which represents different experimental groups</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="co"># The tilde (~) indicates that the analysis will model how the condition affects gene expression (i.e. the result will be a comparison of expression levels across the different conditions)</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeqDataSetFromTximport</span>(txi, <span class="at">colData=</span>data, <span class="at">design=</span><span class="sc">~</span>condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
design formula are characters, converting to factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using counts and average transcript lengths from tximport</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>dds <span class="co"># dds contains 16,230 genes and 9 samples</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>class: DESeqDataSet 
dim: 16230 9 
metadata(1): version
assays(2): counts avgTxLength
rownames(16230): ENSMUSG00000000001 ENSMUSG00000000028 ...
  ENSMUSG00000107383 ENSMUSG00000107390
rowData names(0):
colnames(9): BR1-EC_bRNA BR2-EC_bRNA ... LV2-EC_bRNA LV3-EC_bRNA
colData names(1): condition</code></pre>
</div>
</div>
<p><code>dds</code> contains:</p>
<ul>
<li><p><strong>Counts</strong>: the raw read counts for each gene in each sample</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(<span class="fu">counts</span>(dds)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 14%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">BR1-EC_bRNA</th>
<th style="text-align: right;">BR2-EC_bRNA</th>
<th style="text-align: right;">BR3-EC_bRNA</th>
<th style="text-align: right;">LG1-EC_bRNA</th>
<th style="text-align: right;">LG2-EC_bRNA</th>
<th style="text-align: right;">LG3-EC_bRNA</th>
<th style="text-align: right;">LV1-EC_bRNA</th>
<th style="text-align: right;">LV2-EC_bRNA</th>
<th style="text-align: right;">LV3-EC_bRNA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000001</td>
<td style="text-align: right;">1570</td>
<td style="text-align: right;">1097</td>
<td style="text-align: right;">739</td>
<td style="text-align: right;">1085</td>
<td style="text-align: right;">105</td>
<td style="text-align: right;">426</td>
<td style="text-align: right;">857</td>
<td style="text-align: right;">926</td>
<td style="text-align: right;">1764</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000028</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">23</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">8</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">32</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000049</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">29</td>
<td style="text-align: right;">24</td>
<td style="text-align: right;">210</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000056</td>
<td style="text-align: right;">262</td>
<td style="text-align: right;">74</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">82</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">5</td>
<td style="text-align: right;">427</td>
<td style="text-align: right;">11</td>
<td style="text-align: right;">107</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000000058</td>
<td style="text-align: right;">1531</td>
<td style="text-align: right;">1449</td>
<td style="text-align: right;">1619</td>
<td style="text-align: right;">2517</td>
<td style="text-align: right;">479</td>
<td style="text-align: right;">1305</td>
<td style="text-align: right;">1300</td>
<td style="text-align: right;">1660</td>
<td style="text-align: right;">1412</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000000078</td>
<td style="text-align: right;">8184</td>
<td style="text-align: right;">8183</td>
<td style="text-align: right;">7587</td>
<td style="text-align: right;">4709</td>
<td style="text-align: right;">359</td>
<td style="text-align: right;">1711</td>
<td style="text-align: right;">1772</td>
<td style="text-align: right;">3543</td>
<td style="text-align: right;">1758</td>
</tr>
</tbody>
</table>
</div>
</div></li>
<li><p><strong>Sample metadata</strong>: information about each sample, such as experimental conditions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">colData</span>(dds))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">condition</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR2-EC_bRNA</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BR3-EC_bRNA</td>
<td style="text-align: left;">BR</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG1-EC_bRNA</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LG2-EC_bRNA</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG3-EC_bRNA</td>
<td style="text-align: left;">LG</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV1-EC_bRNA</td>
<td style="text-align: left;">LV</td>
</tr>
<tr class="even">
<td style="text-align: left;">LV2-EC_bRNA</td>
<td style="text-align: left;">LV</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV3-EC_bRNA</td>
<td style="text-align: left;">LV</td>
</tr>
</tbody>
</table>
</div>
</div></li>
<li><p><strong>Design formula</strong>: the model formula used to explain how gene expression varies with conditions.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">design</span>(dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>~condition</code></pre>
</div>
</div></li>
<li><p><strong>Gene/feature metadata</strong>: additional information about each gene/feature.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(dds) <span class="co"># Gene identifiers are row names (not a column)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 16230 rows and 0 columns</code></pre>
</div>
</div></li>
</ul>
<p>Use the <code>DESeq()</code> function to perform a series of steps to analyse RNA-seq count data for differential expression between experimental conditions:</p>
<ul>
<li><p><strong>Estimation of size factors</strong>: normalises the count data to account for differences in sequencing depth or library size across samples.</p></li>
<li><p><strong>Estimation of dispersions</strong>: estimates the dispersion parameter for each gene, which reflects the variability of counts across replicates.</p></li>
<li><p><strong>Fitting the model</strong>: fits a generalised linear model (GLM) for each gene, based on the design formula specified (<code>~condition</code> in this case).</p></li>
<li><p><strong>Wald Test or Likelihood Ratio Test</strong>: performs statistical tests (typically a Wald test) to determine if there is a significant difference in gene expression between conditions.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run the DEA piepline on dds</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">=</span> <span class="fu">DESeq</span>(dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'avgTxLength' from assays(dds), correcting for library size</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
</div>
<p>After running this function, the <code>dds</code> object contains additional information such as:</p>
<ul>
<li><p><strong>Normalised counts</strong>: the count data normalised by the estimated size factors.</p></li>
<li><p><strong>Dispersions</strong>: estimates of dispersion for each gene.</p></li>
<li><p><strong>Fitted model parameters</strong>: coefficients from the GLM.</p></li>
<li><p><strong>Test results</strong>: p-values and log2 fold changes indicating differential expression.</p></li>
</ul>
</section>
<section id="transformation" class="level2">
<h2 class="anchored" data-anchor-id="transformation">Transformation</h2>
<p>Most downstream analyses work best in situations where the variance of the input data (here, each gene) is relatively independent of the average value. For read count data such as RNA-seq, this is not the case. In fact, the variance increases with the average read count.</p>
<p>Generate a plot to show how the standard deviation of genes varies with their mean (across all samples)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(dds), <span class="at">ranks =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Perform a variance stabilising transformation (VST) on the normalised counts.</p>
<p>VST aims to stabilise variance across the range of mean values. This is important as RNA-seq data typically exhibit a mean-variance relationship, which is where expression level of the gene affects the variability of that gene.</p>
<p>Generate the plot to verify that it has removed the correlation between average read count and variance.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The resultant vsd object contains variance-stabilised normalised counts (this is a log2-like scale but with variance stabilised across the range of counts)</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">=</span> <span class="fu">vst</span>(dds, <span class="at">blind=</span><span class="cn">TRUE</span>) <span class="co"># Do not take into account the experimental design</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate plot</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(vsd), <span class="at">ranks =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-26-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Apply log2(n + 1) transformation to the normalised counts using <code>normTransform()</code>.</p>
<p>log2(n + 1) transformation helps to make the data more interpretable, especially when dealing with count data that has a wide range of expression levels. n + 1 handles zero values before performing the log transformation.</p>
<p>Generate same plot to verify that it has removed the correlation between average read count and variance.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The resultant ntd object contains log2-transformed normalised counts</span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>ntd <span class="ot">=</span> <span class="fu">normTransform</span>(dds)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate plot</span></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a><span class="fu">meanSdPlot</span>(<span class="fu">assay</span>(ntd), <span class="at">ranks =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="pca" class="level2">
<h2 class="anchored" data-anchor-id="pca">PCA</h2>
<p>Perform PCA on the variance-stabilised data (<code>vsd</code>) and generate a PCA plot using <code>PlotPCA()</code>.</p>
<p>PCA is a dimensionality reduction technique that helps to visualise the variation in gene expression data across samples, often revealing patterns such as clustering of samples by condition or batch effects.</p>
<p>The variance-stabilised data is used here as PCA aims to capture the structure in the data by identifying principal components that explain the maximum variance. If variance is not stabilisated, the genes with high variability (which is determined by expression levels) might disproportionately influence the principal components.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># intgroup="condition" specifies the grouping variable(s) of interest that will be used to colour the samples in the PCA plot, making it easier to observe how samples group according to this variable</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a><span class="co"># returnData=TRUE returns the data used to generate the PCA plot instead of the plot itself</span></span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a>pcaData <span class="ot">=</span> <span class="fu">plotPCA</span>(vsd, <span class="at">intgroup=</span><span class="st">"condition"</span>, <span class="at">returnData=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
</div>
<p><code>pcaData</code> contains:</p>
<ul>
<li><p><strong>PC1</strong>: the coordinates of each sample along the first principal component.</p></li>
<li><p><strong>PC2</strong>: the coordinates of each sample along the second principal component.</p></li>
<li><p><strong>group</strong>: the grouping variable associated with each sample.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(pcaData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">PC1</th>
<th style="text-align: right;">PC2</th>
<th style="text-align: left;">group</th>
<th style="text-align: left;">condition</th>
<th style="text-align: left;">name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BR1-EC_bRNA</td>
<td style="text-align: right;">-47.157598</td>
<td style="text-align: right;">31.45050</td>
<td style="text-align: left;">BR</td>
<td style="text-align: left;">BR</td>
<td style="text-align: left;">BR1-EC_bRNA</td>
</tr>
<tr class="even">
<td style="text-align: left;">BR2-EC_bRNA</td>
<td style="text-align: right;">-48.944877</td>
<td style="text-align: right;">31.33826</td>
<td style="text-align: left;">BR</td>
<td style="text-align: left;">BR</td>
<td style="text-align: left;">BR2-EC_bRNA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BR3-EC_bRNA</td>
<td style="text-align: right;">-49.052319</td>
<td style="text-align: right;">30.21645</td>
<td style="text-align: left;">BR</td>
<td style="text-align: left;">BR</td>
<td style="text-align: left;">BR3-EC_bRNA</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG1-EC_bRNA</td>
<td style="text-align: right;">-10.514917</td>
<td style="text-align: right;">-45.18792</td>
<td style="text-align: left;">LG</td>
<td style="text-align: left;">LG</td>
<td style="text-align: left;">LG1-EC_bRNA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LG2-EC_bRNA</td>
<td style="text-align: right;">-8.934462</td>
<td style="text-align: right;">-52.72137</td>
<td style="text-align: left;">LG</td>
<td style="text-align: left;">LG</td>
<td style="text-align: left;">LG2-EC_bRNA</td>
</tr>
<tr class="even">
<td style="text-align: left;">LG3-EC_bRNA</td>
<td style="text-align: right;">-9.217820</td>
<td style="text-align: right;">-48.55511</td>
<td style="text-align: left;">LG</td>
<td style="text-align: left;">LG</td>
<td style="text-align: left;">LG3-EC_bRNA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV1-EC_bRNA</td>
<td style="text-align: right;">56.053741</td>
<td style="text-align: right;">16.79258</td>
<td style="text-align: left;">LV</td>
<td style="text-align: left;">LV</td>
<td style="text-align: left;">LV1-EC_bRNA</td>
</tr>
<tr class="even">
<td style="text-align: left;">LV2-EC_bRNA</td>
<td style="text-align: right;">61.847745</td>
<td style="text-align: right;">19.18357</td>
<td style="text-align: left;">LV</td>
<td style="text-align: left;">LV</td>
<td style="text-align: left;">LV2-EC_bRNA</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LV3-EC_bRNA</td>
<td style="text-align: right;">55.920506</td>
<td style="text-align: right;">17.48304</td>
<td style="text-align: left;">LV</td>
<td style="text-align: left;">LV</td>
<td style="text-align: left;">LV3-EC_bRNA</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>List all the attributes associated with <code>pcaData</code>. An attribute is a piece of metadata or a property that provides additional information about an object.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">attributes</span>(pcaData)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$names
[1] "PC1"       "PC2"       "group"     "condition" "name"     

$class
[1] "data.frame"

$row.names
[1] "BR1-EC_bRNA" "BR2-EC_bRNA" "BR3-EC_bRNA" "LG1-EC_bRNA" "LG2-EC_bRNA"
[6] "LG3-EC_bRNA" "LV1-EC_bRNA" "LV2-EC_bRNA" "LV3-EC_bRNA"

$percentVar
[1] 0.5859993 0.3712017</code></pre>
</div>
</div>
<p>Calculate the percentage of variance explained by the first two principal components (PC1 and PC2).</p>
<p>This indicates how much of the total variability in the data is captured by these two components.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Retrieve the "percentVar" attribute for the pcaData object, and convert the fraction of variance explained to a percentage</span></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">=</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(pcaData, <span class="st">"percentVar"</span>))</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>percentVar</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 59 37</code></pre>
</div>
</div>
<p>Visualise the PCA plot.</p>
<p>This plot helps to visualise how samples group based on their experimental conditions in the reduced PC space and understand the variance captured by the first two PCs. This can reveal if there are distinct clusters or overlaps between groups.</p>
<p>You can also identify if any samples are outliers.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pcaData, <span class="fu">aes</span>(PC1, PC2, <span class="at">color=</span>condition)) <span class="sc">+</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: "</span>,percentVar[<span class="dv">1</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: "</span>,percentVar[<span class="dv">2</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span> </span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="co"># Ensures that that scaling of the x- and y-axis are fixed</span></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_few</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-32-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>PC1 and PC2 account for 96% of the total variance in the dataset - these two components capture a substantial portion of the dataâ€™s variability. This indicates that the reduced 2D representation is capturing most of the essential information, making it easier to visualise meaningful patterns/clusters/relationships.</p>
</section>
<section id="quality-control" class="level2">
<h2 class="anchored" data-anchor-id="quality-control">Quality control</h2>
<p>These samples were collected directly after FACs sorting so these cells might be stressed.</p>
<p>Look at highly expressed genes in each condition and see if these genes are enriched in stress-related GO terms.</p>
<p>Select the top 300 genes based on average normalised counts to identify the the genes with the highest expression levels across all the samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># counts(dds, normalized=TRUE) retrieves the normalised counts</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the mean normalised count for each gene across all samples</span></span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Order the genes based on their mean normalised counts in descending order (this function returns the indices of the genes)</span></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select the top 300 genes with the highest average expression</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>select <span class="ot">&lt;-</span> <span class="fu">order</span>(<span class="fu">rowMeans</span>(<span class="fu">counts</span>(dds,<span class="at">normalized=</span><span class="cn">TRUE</span>)),</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>                <span class="at">decreasing=</span><span class="cn">TRUE</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">300</span>]</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(select) <span class="co"># Each integer represents the index of a gene in the original dataset</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 12137 12136 15973 14900 14328 12140</code></pre>
</div>
</div>
<p>Once these genes have been selected, extract the counts data for the top 300 genes from the <code>ntd</code> object (which contains the log2 normalised counts).</p>
<p>GO enrichment analysis focuses on expression levels and the log2 transformation helps in compressing the range of expression values and handling zero values. This transformation also mitigates issues like skewness which impacts the assumptions of statistical methods that assume normally distributed data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Only include the counts of the selected genes</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>top300 <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">assay</span>(ntd)[select,])</span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(top300))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 14%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: right;">BR1-EC_bRNA</th>
<th style="text-align: right;">BR2-EC_bRNA</th>
<th style="text-align: right;">BR3-EC_bRNA</th>
<th style="text-align: right;">LG1-EC_bRNA</th>
<th style="text-align: right;">LG2-EC_bRNA</th>
<th style="text-align: right;">LG3-EC_bRNA</th>
<th style="text-align: right;">LV1-EC_bRNA</th>
<th style="text-align: right;">LV2-EC_bRNA</th>
<th style="text-align: right;">LV3-EC_bRNA</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000064339</td>
<td style="text-align: right;">20.63240</td>
<td style="text-align: right;">20.36137</td>
<td style="text-align: right;">20.75712</td>
<td style="text-align: right;">19.35866</td>
<td style="text-align: right;">18.43775</td>
<td style="text-align: right;">19.14895</td>
<td style="text-align: right;">21.65061</td>
<td style="text-align: right;">21.84351</td>
<td style="text-align: right;">21.12461</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000064337</td>
<td style="text-align: right;">19.67134</td>
<td style="text-align: right;">19.01064</td>
<td style="text-align: right;">19.32008</td>
<td style="text-align: right;">17.96961</td>
<td style="text-align: right;">17.39532</td>
<td style="text-align: right;">17.99099</td>
<td style="text-align: right;">20.71149</td>
<td style="text-align: right;">21.03126</td>
<td style="text-align: right;">20.24610</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000106106</td>
<td style="text-align: right;">18.50385</td>
<td style="text-align: right;">18.83705</td>
<td style="text-align: right;">19.21683</td>
<td style="text-align: right;">18.77296</td>
<td style="text-align: right;">19.46000</td>
<td style="text-align: right;">19.13979</td>
<td style="text-align: right;">20.10907</td>
<td style="text-align: right;">20.74802</td>
<td style="text-align: right;">20.29679</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000098178</td>
<td style="text-align: right;">18.64633</td>
<td style="text-align: right;">19.12970</td>
<td style="text-align: right;">19.37125</td>
<td style="text-align: right;">19.04266</td>
<td style="text-align: right;">19.24107</td>
<td style="text-align: right;">19.28725</td>
<td style="text-align: right;">18.51664</td>
<td style="text-align: right;">19.40545</td>
<td style="text-align: right;">17.51050</td>
</tr>
<tr class="odd">
<td style="text-align: left;">ENSMUSG00000092341</td>
<td style="text-align: right;">19.26558</td>
<td style="text-align: right;">18.79547</td>
<td style="text-align: right;">18.78624</td>
<td style="text-align: right;">17.42506</td>
<td style="text-align: right;">16.15052</td>
<td style="text-align: right;">17.21982</td>
<td style="text-align: right;">17.11530</td>
<td style="text-align: right;">16.12312</td>
<td style="text-align: right;">17.39293</td>
</tr>
<tr class="even">
<td style="text-align: left;">ENSMUSG00000064351</td>
<td style="text-align: right;">17.16200</td>
<td style="text-align: right;">17.26503</td>
<td style="text-align: right;">17.35215</td>
<td style="text-align: right;">15.88960</td>
<td style="text-align: right;">15.05999</td>
<td style="text-align: right;">15.60946</td>
<td style="text-align: right;">19.03051</td>
<td style="text-align: right;">19.34770</td>
<td style="text-align: right;">18.65586</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Perform GO enrichment analysis to identify GO terms that are overrepresented among the top 300 genes.</p>
<p><code>gost()</code>, of the <a href="https://cran.r-project.org/web/packages/gprofiler2/gprofiler2.pdf">gprofiler2 package</a>, performs functional enrichment analysis (FEA) on gene lists. FEA is a method to identify classes of genes that are over-represented in a large set of genes. Several data sources are used:</p>
<ul>
<li><p><strong>Gene Ontology (GO)</strong></p>
<ul>
<li><p><strong>GO:MF</strong>: functional roles at the molecular level e.g.&nbsp;â€œenzyme activityâ€.</p></li>
<li><p><strong>GO:CC</strong>: subcellular localisation where gene product functions e.g.&nbsp;â€œnucleusâ€.</p></li>
<li><p><strong>GO:BP</strong>: broader biological processes and pathways e.g.&nbsp;â€œcell divisionâ€.</p></li>
</ul></li>
</ul>
<!-- -->
<ul>
<li><p><strong>Pathway and Functional databases</strong></p>
<ul>
<li><p><strong>KEGG</strong>: biological pathways and interactions between different biological molecules.</p></li>
<li><p><strong>REACTOME</strong>: biological pathways and processes.</p></li>
</ul></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment analysis using the gene identifiers provided in the row name</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>go_enrich <span class="ot">=</span> <span class="fu">gost</span>(<span class="fu">rownames</span>(top300), <span class="at">organism =</span> <span class="st">"mmusculus"</span>,<span class="at">ordered_query =</span> <span class="cn">TRUE</span>) <span class="co"># Specify the organism is a mouse and that the gene list is provided in a specific order</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise result of GO enrichment analysis</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">gostplot</span>(go_enrich, <span class="at">interactive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-9390ab8c97ee2d4e9970" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-9390ab8c97ee2d4e9970">{"x":{"data":[{"x":[148.1032480554253],"y":[1.5296506313926308],"text":"CORUM:538 (12) <br> Cytochrome c oxidase, mitochondrial <br> 2.954e-02","key":["CORUM:538"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(102,170,0,1)","opacity":0.80000000000000004,"size":12.058806804819245,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(102,170,0,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"CORUM","legendgroup":"CORUM","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[79.300658338305112,79.019376587791626,65.133601600388999,54.394666196856058,78.544713633800114,52.453319829472768,46.739784272167469,46.232472543562778,68.581814488380289,46.174709326939471,90.253066498924198,62.956179478110457,111.04782448331447,68.501448273947858,71.053075582177385,52.239847072386631,79.333307112918291,58.62393823136248,71.05056413797638,78.808415274906508,79.926007944357437,78.542202189599095,58.234664380205416,111.05284737171648,62.958690922311469,62.722615167416215,79.330795668717272,71.065632803182453,78.549736522202139,78.559782299006187,89.148031050478338,78.552247966403158,86.480877309002196,79.707512298869275,80.586517769223946,78.469370307769708,86.483388753203215,45.94114501624523,78.562293743207206,52.696929916971051,65.131090156187994,80.375556456338828,79.717558075673324,54.841703264636429,81.407760022955287,62.531745408139201,62.725126611617227,87.661256083478449,53.332325299827424,62.40868464228955,52.267472958597779,71.266548339263522,95.65016008690182,51.895779216847806,79.715046631472319,62.639737508782773,78.682843064855845,54.33941442443377,110.76654273280097,80.589029213424951,100.35911796380178,54.854260485641497,78.93901037335921,63.749795845630658,56.135097028158292,62.752752497828375,52.237335628185619,78.858644158926779,75.463171599156766,52.264961514396767,54.299231317217554,80.498617222188471,75.503354706372988,62.604577289968596,46.292747204387098,73.642374553422115,54.510192630102672,68.953508230130254,97.066614616273313,52.269984402798798,79.19517768186256,78.685354509056864,46.295258648588117,55.085313352134719,95.652671531102811,50.348729589023606,54.982344139893179,72.22340857984959,79.305681226707151,53.319768078822364,56.961362170291672,70.123841227802458,87.121295580260607,63.852765057872212,88.193682254093275,80.312770351313489,63.953222825912739,51.659703461952553,56.200394577384635,87.616050087860202,80.573960548218878,93.92228647660464,51.845550332827536,91.978428665020331,62.581974292159465,66.050278733758859,62.602065845767576,88.316743019942919,88.603047658858458,91.973405776618307,80.596563546027994,79.33581855711931,53.553332389516591,58.202015605592244,58.204527049793263,52.71451002637815,71.437326544932418,80.486060001183404,54.984855584094184,60.834009128254209,56.238066240399831,45.474016394856761,49.027709939290602,78.861155603127798,58.191969828788189,62.57946284795846,83.243625733896025,75.078920636401733,54.849237597239465,56.029616371715726,85.644566390064753,51.968611098677187,73.908587638729529,71.957195494542177,72.228431468251628,59.872125999266103,73.037116500977902,74.855402102511547,63.860299390475248,51.637100464143444,51.629566131540393,93.550592734854675,80.378067900539833,63.352987661870557,72.61519387520768,73.250589258064039,47.626324075125169,50.290966372400291,102.91827960463436,59.867103110864079,80.488571445384409,67.592305473181042,51.996236984888341,102.83289050179989,62.63722606458176,102.83791339020193,80.48103711278138,105.78634888219156,67.589794028980023,86.968097483998775,52.935517116067317,91.343033282163972,63.847742169470173,82.849328994336943,46.237495431964803,72.494644553559041,54.999924249300264,67.160337070606758,53.334836744028436,56.192860244781592,86.965586039797756,93.590775842070897,52.717021470579162,101.05478800748247,46.995951580670827,56.202906021585648,80.599074990228999,75.465683043357785,50.93640753206072,55.09033624053675,66.118087727186222,75.073897747999709,52.061534534114692,62.54179118494325,58.546083461131076,72.62021676360969,100.8312694735923,62.607088734169601,56.240577684600851,81.269630591899556,93.942378030212751,54.304254205619578,64.156649806194821,52.249892849190687,91.837787789763595,56.5720883191346,52.252404293391699,95.647648642700801,103.51600332447552,82.919649431965311,81.395202801950234,49.012641274084523,72.612682431006661,72.780949192474552,97.254972931349315,84.652545930664502,62.49909663352603,87.455317658995355,63.897971053490444,65.814202978863605,97.24492715454528,62.504119521928054,69.998269017751795,49.344151908618279,46.717181274358346,75.794682233690537,56.0421735927208,46.958279917655631,46.724715606961389,62.65480617398886,71.550341533978028,56.373684227254543,55.097870573139787,63.958245714314764,51.812901558214371,54.663390726364483,46.727227051162394,101.50433651946386,53.194195868771693,56.57459976333562,79.664817747452048,53.262004862199049,79.579428644617593,53.337348188229448,58.219595714999343,73.639863109221096,72.230942912452633,79.591985865622661,70.508092190557491,78.687865953257869,74.903119542330799,80.239938469484102,52.292587400607914,54.314299982423634,73.044650833580945,54.324345759227683,80.237427025283097,60.590399040755912,88.271537024324687,51.815413002415383,88.650765098677695,58.19699271719022,71.040518361172332,56.112494030349161,56.207928909987672,83.052755974619004,48.364688670223089,51.662214906153565,73.238032037058971,46.754852937373549,75.995597769771592,81.289722145507668,46.450968189050933,80.380579344740838,79.669840635854072,69.614018054996762,54.525261295308752,81.003417506592143,50.351241033224618,71.055587026378404,53.407668625857823,54.844214708837448,62.60960017837062,81.709133327076884,56.243089128801863,91.928199781000075,81.578538228624197,65.186341928610275,74.840333437305475,59.741530900813409,49.040267160295663,56.027104927514706,73.04213938937994,46.759875825775573,103.82742240540118,101.04976511908043,53.892377356653398,51.559245693912018,97.257484375550348,66.123110615588246,100.96688746044701,83.374220832348726,62.692477837004056,62.69750072540608,62.938599368703358,52.275007291200822,63.855276502073224,53.741690704592592,53.877308691447318,58.465717246698645,62.936087924502353,45.456436285449662,52.747158800991322,73.594657113602864,53.894888800854403,80.483548556982399,54.83919182043541,53.744202148793612,52.732090135785242,51.664726350354584,55.90153271746405,88.962184179603341,83.544999038017622,46.23498398776379,97.036477285861167,67.06741363516926,92.29989352275004,80.59154065762597,53.882331579849343,56.115005474550181,46.631792171523898,53.736667816190568,52.739624468388286,54.987367028295203,53.734156371989556,81.714156215478923,83.005038534799766,77.294014421695479,58.244710157009465,102.61941774471377,77.597899170018081,63.850253613671192,88.969718512206384,91.905596783190958,57.531460003921694,57.119583154955507,73.888496085121417,72.53231621657423,83.509838819203438,78.697911730061918,58.644029784970591,77.743562933676856,74.767501555476073,66.848917989681098,55.529838975714078,79.014353699389602,65.816714423064624,64.500717661733646,111.12065636514384,105.38954069843146,100.3616294080028,66.01009562654265,62.961202366512481,77.703379826460662,55.559976306126238,66.035210068552772,52.059023089913673,103.51851476867654,78.881247156735895,52.001259873290365,110.01562091669797,79.983771160980737,54.851749041440478,46.72220416276037,78.695400285860913,62.820561491255738,79.082162692816965,71.238922453052382,71.176136348027043,103.83244529380319,101.05227656328145,111.05033592751546,79.93354227696048,95.640114310097758,53.405157181656811,79.672352080055092,95.411572887805548,83.389289497554799,49.871555190831074,67.856007114287436,73.516802343371452,85.421047856174567,85.428582188777611,82.15617039485727,46.64937228093099,82.834260329130856,51.87317621903869,100.83629236199431,80.601586434430018,71.557875866581071,56.137608472359297,93.954935251217819,57.536482892323718,67.162848514807763,54.668413614766507,93.937355141810727,71.555364422380052,48.369711558625113,75.69171302144899,64.957800506318065,73.64739744182414,101.05729945168348,65.977446851929471,88.904420962980041,53.761782258200697,103.07398914509717,65.771508427446392,82.424894924365688,77.34424330571575,47.345042324611683,46.270144206577982,63.955734270113759,102.63950929832185,81.156615602853961,77.208625318861039,97.172095272715879,96.85314185918719,54.02799534350811,72.082767704592854,51.898290661048826,57.491276896705479,62.700012169607092,85.375841860556321,88.688436761692898,73.896030417724461,79.262986675289923,79.350887222325383,76.774145472085721,97.948131530828988,79.873267616136161,82.251605274495759,79.348375778124378,93.94991236281578,63.398193657488804,88.685925317491879,50.403981361445901,80.31025890711247,88.605559103059448,66.233614160432822,81.000906062391138,56.220486130992747,63.069194467156052,78.612522627227477,75.787147901087479,65.791599981054489,81.005928950793162,80.614143655435086,80.194732473865855,79.986282605181756,84.167837199868927,86.973120372400814,79.597008754024685,90.391195929979929,81.581049672825216,64.505740550135684,49.331594687613212,60.989718668717025,83.215999847684884,63.528788755941491,81.38766846934719,85.654612166868816,63.084263132362132,95.343763894378171,69.528628952162308,105.07561017330478,46.536357291885388,88.452361006797645,46.533845847684368,64.174229915601913,73.074788163993105,80.616655099636091,74.800150330089252,62.772844051436486,67.416504379110108,79.885824837141215,110.57818441772497,57.48876545250446,56.908621842070382,66.570147683368617,72.605148098403617,102.39087632242156,79.275543896294991,52.428205387462633,49.464701230266911,63.594086305167842,64.952777617916041,75.789659345288499,53.791919588612863,63.890436720887408,49.178396591351394,100.27121741676632,71.043029805373337,45.340909852203048,59.03832652452968,89.05259617083982,89.050084726638801,83.69819713427944,88.427246564787524,54.15356755355878,83.015084311603815,78.803392386504484,80.551357550409747,79.265498119490942,62.534256852340221,46.732249939564426,75.478240264362853,97.259995819751339,75.80975089889661,67.60988558258812,88.366971903963204,54.665902170565502,75.694224465649995,91.888016673783866,83.170793852066637,103.07650058929818,112.12272260134816,72.529804772373211,64.651404313794444,72.720674531650232,62.657317618189872,49.462189786065906,92.563595163856434,46.742295716368481,73.742832321462657,79.062071139208854,85.383376193159378,71.467463875344578,65.121044379383932,79.677374968457116,52.950585781273396,84.572179716232071,54.570467290926992,82.254116718696778,95.795823850560581,46.219915322557711,79.855687506729055,94.944444266417065,87.96262938760006,89.409221247383712,80.415739563555036,67.599839805784072,93.894660590393499,50.750560661185737,57.262735474413269,62.516676742933129,79.270521007892953,80.609120767033062,51.938473768265041,89.436847133594881,95.082573697472796,58.955448865896244,89.421778468388794,53.314745190420325,67.958976326528983,52.259938625994742,46.63430361572491,54.316811426624646,50.768140770592829,74.885539432923707,52.433228275864657,95.7983352947616,54.585535956133072,74.61430345921427,47.284767663787363,81.588584005428245,56.91364473047242,82.495215361994056,48.460123549861585,110.5078639800966,97.953154419231026,52.435739720065669,54.578001623530028,52.295098844808919,46.282701427583042,56.092402476741057,80.433319672962128,46.541380180287419,91.446002494405505,63.857787946274229,99.959798335840659,77.449723962158302,48.367200114424094,85.373330416355316,77.45474685056034,75.538514925187158,56.032127815916745,84.823324136333397,48.394826000635241,70.65375595421628,82.16119328325928,67.514450702949631,62.509142410330085,90.534348249437699,54.846726153038453,49.469724118668942,72.698071533841102,84.165325755667922,60.359346174262683,46.729738495363414,100.969398904648,56.582134095938656,100.40432395942001,50.730469107577633,58.935357312288133,48.399848889037273,64.252084685833324,59.083532520147919,79.338330001320315,91.893039562185905,52.257427181793723,46.885448035826236,62.750241053627363,49.52748733529225,63.900482497691463,50.720423330773578,107.80806146400727],"y":[17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,17,15.978236750297237,15.859203195921873,15.229114074592335,14.939190525467795,14.799076066019213,14.700479271479812,14.445676178535711,14.444671652853584,14.337853897100528,13.93910626615563,13.833409335358716,13.421224907676415,13.393468418094141,13.389840566881917,13.377759860406497,13.377737494443025,13.049885102218674,12.915081567848544,12.82775479671265,12.81187879177563,12.672014730125523,12.664854250247172,12.57743889912577,12.564910591277782,12.301231607494302,12.078459154609504,12.07551868387238,11.977922999874824,11.971103719625663,11.925488466075027,11.89452157620687,11.89438103078842,11.76694416166281,11.725584453613068,11.594405010725238,11.578643373243569,11.537054030746248,11.468490948794479,11.304823640156902,11.281958786115334,11.231446414257247,11.230179223727408,11.172270309661389,11.14007341761755,11.14007341761755,11.03504615950739,10.805780846394921,10.790088500022796,10.571661192546184,10.552178679287307,10.416510514004502,10.338957196938493,10.297489978624869,10.254198061398448,10.245025102973022,10.211745937805192,10.211095969846692,10.206938299402387,10.187229327985174,10.091866299187906,10.057860246611895,10.040278425819864,10.031136494574996,10.0287986098975,9.9977290937125378,9.9961265260887764,9.993687349461748,9.9831862422784017,9.9831862422784017,9.8338537936205324,9.7572153894969027,9.6431842635899496,9.6332226522460829,9.611038793501999,9.5970405842098909,9.5884678362667604,9.5785321263597289,9.5513204925993982,9.539787755329769,9.4920058709611794,9.4494602393234235,9.4238546340896914,9.2128534624133493,9.1919713032499466,9.1892087963565032,9.1637902165513179,9.0870036571071751,9.0691712061951542,9.0592696763951057,8.967860418362898,8.9629686296563218,8.941455241565599,8.9263703856275178,8.8651808767019169,8.6851703213343185,8.6144404190761534,8.4756142234032215,8.2403496467580091,8.2211130076368697,8.1668161756315545,8.1575755688494205,8.1298027641636992,8.0835436462582226,8.0644158416373077,8.051522503800312,8.0080560917860666,7.9934888018903729,7.9775604521363652,7.9491393174640441,7.9489755547030478,7.9426832426849687,7.9344545566308264,7.870776657854635,7.8673013939061676,7.8268661431482105,7.79929277769285,7.7321435288159401,7.6760717616423477,7.6733680353818068,7.5716327972072106,7.5690590627903518,7.4710188558234432,7.3672940575695121,7.3641149630430043,7.2900315131997351,7.1605662610483529,7.1605662610483529,7.1324570725775729,6.9863058790850081,6.9825812639048532,6.9789715257153082,6.9779772443153911,6.9329268612055399,6.9077145916251688,6.8611548952484531,6.8194476032783591,6.7329586352197754,6.7244144284395437,6.7072442849201748,6.661585986465048,6.6312879204915571,6.6222348601274641,6.6222348601274641,6.6135187706593008,6.5999173954817483,6.5859467188825134,6.5805794393327242,6.5799718821861033,6.5341884245654329,6.5239110206188018,6.4630365764406594,6.4615883733628463,6.4579012534903484,6.4518697764106436,6.4125798636095848,6.4016839448890801,6.3929787557139548,6.3797545443601535,6.3783615147889439,6.3299035770201701,6.3115871082389869,6.2943460087683212,6.2653937736679834,6.2348209471999709,6.1831516913021645,6.1790231851229374,6.1788396888771775,6.1274028627974229,6.0542694354449313,6.0388985693252888,6.0388985693252888,6.0331726496876055,5.958294757484321,5.9263181144195993,5.881411271705729,5.8574080803467297,5.8346436718446242,5.8001741392978206,5.7842106845198025,5.7612724476581114,5.7209749249252146,5.6839577199341917,5.5831104796419124,5.5093955214011254,5.4770492949963367,5.4574892187598909,5.4414922281682614,5.4324223077852807,5.383504270422943,5.3823214192235751,5.375905357234136,5.373159798844112,5.3608422112371734,5.351351462311519,5.3318849704570797,5.3045592406192856,5.2970490961810688,5.278153505851944,5.2660747787274804,5.2599215872409983,5.2460760442346457,5.2103396716675316,5.1932507537241142,5.1776213615913571,5.1440483168573738,5.1415907653850805,5.108202544027149,5.0645392647228862,5.0393369522317037,5.0291507994715818,5.0238957816494176,5.0184450508611667,4.9745768683689979,4.9745768683689979,4.9493921387557478,4.9481586642448097,4.9155042752673737,4.9132422969947696,4.8616751909461735,4.851721624375215,4.7944864568178112,4.7687730336665348,4.7517687068481926,4.7105594176176249,4.6661709741148769,4.6364872181134889,4.6333202479329536,4.6282900165822003,4.6094579331949479,4.5977028151521324,4.5977028151521324,4.5815664649477537,4.5742398822822885,4.5355477411301841,4.5008646902495357,4.4910166616904341,4.4424959757604894,4.3791869009630568,4.3637505821248705,4.3513545550963588,4.3382828592026836,4.3116330086544732,4.3060820681059493,4.303540479160354,4.2664170603757414,4.2564859253511029,4.2438079064827212,4.2432474564721439,4.2142363045478453,4.1751391436939755,4.1701075232048597,4.1499215382424408,4.1499215382424408,4.1463826538027977,4.1407796917337043,4.116716662415131,4.1081789212922004,4.1059104128185746,4.0876422517506938,4.0759676071787574,4.072601537757456,4.0684061988253122,4.0643604739182884,4.050850124085593,4.0372654331336353,4.0320574977240238,4.0290791425843304,3.9958356430881437,3.9860240401894504,3.983993085986933,3.9803204734533288,3.9758601402899014,3.9747640793261749,3.8564181823609527,3.8535806525363934,3.8502638954373989,3.7771500093028845,3.7729357029095292,3.7719630163795892,3.7231075015806723,3.7207011281101336,3.7079407419076111,3.6924405026842786,3.6922397998391134,3.6863856378403064,3.6671035313591904,3.6239494496209881,3.5879092807495305,3.5845548371907041,3.5753585878903462,3.5626045406095992,3.5553428067171642,3.5477550417467905,3.4901453547997781,3.4810383839572667,3.4789309133127082,3.4758114822223143,3.4658621153637705,3.4617471710343897,3.4351888615888044,3.4323492857041735,3.4202709930475708,3.412901408114168,3.3796894037188818,3.3793490083520918,3.3680396241344468,3.3661465090624203,3.3517440880515768,3.3248151235648766,3.3248151235648766,3.3072287780184446,3.2965554215131556,3.2857655523328866,3.2719789650647044,3.2310622969026377,3.2047790328387693,3.1875143538298705,3.1844979895436936,3.181081862478782,3.1550871244894414,3.1381828295694754,3.100381846168101,3.0937166529837321,3.0936041380355257,3.0864698525403385,3.085054392791271,3.0694548361998661,3.0681831731713141,3.0680978755322541,3.0449096792516741,3.0224870016411756,3.0026383858910544,2.9981916310635635,2.9878523192061825,2.985923496113903,2.9842270272783797,2.9842270272783797,2.9616668969133872,2.954828882384406,2.954828882384406,2.9309989439810686,2.929344204302418,2.9165164827982029,2.9023762615126762,2.8995678204253061,2.8951670211925928,2.8862929576497396,2.8570096331936852,2.8232931329492166,2.8097799387381048,2.8091980797892777,2.7976573525085517,2.790783239416013,2.7860975513429098,2.7846240261578727,2.7792293423684939,2.7725851423000152,2.762537828342829,2.7575578431520693,2.733350481925894,2.7254197344966626,2.7180826069042481,2.6803202527584498,2.6799816524244249,2.6742732774797924,2.6700835752648588,2.6606012840035937,2.6516043944944738,2.6358394074938749,2.6313094973520825,2.6307869470762162,2.6092783853969723,2.6004175219751859,2.5977739135987172,2.5958209205060379,2.5847353040072574,2.5814018882328371,2.577597094663036,2.5757510992699038,2.5728000866496985,2.5605640079543384,2.5528230901344937,2.5457812273022693,2.5329894247103422,2.5273733751832843,2.5208912200268929,2.518603278382503,2.5148343492967293,2.4832607715740389,2.4620366602332129,2.4607392158172865,2.4567975249094927,2.4527928769068903,2.4516318867922022,2.4463986810735956,2.4454129057767955,2.4356983653236011,2.4302029535504315,2.4186167748203,2.4185096847951155,2.4064224065155968,2.404989815051028,2.4033154575725031,2.3998504883904328,2.3942832293267484,2.3928254758729688,2.3921857695976483,2.3888817305782948,2.3855609674570708,2.3847225553260234,2.382956452436114,2.3733008504281417,2.3584329136626718,2.3369609739717894,2.3355710558763421,2.3335516653738249,2.3196884662456618,2.3192972901867694,2.3086763969809621,2.3005796274372527,2.2710472306655558,2.2583848699816071,2.2423815198036467,2.2365911547516864,2.2355246469596017,2.2304955969612585,2.2304955969612585,2.2257128583622849,2.2202083286214238,2.2162722828002801,2.2115888780366011,2.2086565756663785,2.203973519742854,2.1974624822672486,2.1956349219554747,2.1956349219554747,2.1901253179088229,2.1853401259365461,2.1842361621902411,2.1789365992773342,2.1771531175302274,2.1495942969825119,2.1381232747424397,2.1353882198716918,2.1268468416530646,2.1208303470178338,2.1100186997013521,2.1092480579154551,2.1013555751979331,2.1013555751979331,2.0906842092223394,2.0875589290291057,2.0734448352163355,2.0704755343240588,2.0626039551543056,2.0604920283836909,2.0546270439175722,2.0472472598612583,2.034360972532661,2.0240300488454888,2.0191409926025461,2.0124366497430839,1.9988789685801882,1.9879961657187153,1.9875729563094218,1.9872610636115202,1.9753730325543974,1.9712302337684429,1.9692812490600555,1.955732975061808,1.9528263718871985,1.9473574241238043,1.9186114353743502,1.9186114353743502,1.9125467078687481,1.9035654940531561,1.9016949117094148,1.9002981514624735,1.8891808750477634,1.8870834883137291,1.8523078491664853,1.8512496127433593,1.8502341175568862,1.8452054857703335,1.8332862535956493,1.8286119238519545,1.8013840410711277,1.7934578705164637,1.7911768067979612,1.7887333128479068,1.788101849934828,1.788101849934828,1.782262628639202,1.7789226200420514,1.7708456971664692,1.7663689948647547,1.7652374789851388,1.754341206343399,1.7539259092377408,1.7521531434940292,1.7514012407313142,1.746506663426703,1.7461613502012368,1.7434327933231319,1.7337538107138342,1.7314910071540406,1.7297803244169419,1.7289536638488148,1.7282250299111162,1.7268942898028294,1.7228766035962939,1.6925830240160606,1.6826011154486851,1.6741979641542983,1.6721822594903459,1.6690412053731485,1.6673554767537186,1.6627190849180344,1.6538176834294429,1.6489472831636189,1.6365477772559069,1.636206167076673,1.6350894319501199,1.6185253228077152,1.5973923772914052,1.5877971531512123,1.5753582782216939,1.571975531988679,1.5557341308487094,1.5541831866793476,1.5539382577820413,1.544491383945739,1.5390183661768557,1.5337013485978974,1.5323654398295252,1.5299746806514056,1.5184085932307281,1.5128211930546156,1.4890163561904934,1.4754183067269553,1.4737214652379542,1.4705033697373431,1.469496591428924,1.4689114303668647,1.4618107802680798,1.4516379027814723,1.4484646314251974,1.4390838114219404,1.4257908150838159,1.417089493236493,1.4138691999233981,1.413478024728793,1.4063838770627473,1.4048957117129912,1.4016249383469697,1.4005376430946159,1.398998276794611,1.3831083814196796,1.3817001259597512,1.3689360194259146,1.3496309586589292,1.344803634092284,1.3365345178683325,1.3357183520446003,1.3309133600973944,1.3275426809309441,1.3211310629560125,1.3207600133607391,1.3130147709486109,1.3062437147801156],"text":["GO:0048856 (6269) <br> anatomical structure development <br> 5.666e-33","GO:0048731 (4117) <br> system development <br> 9.078e-32","GO:0032502 (6854) <br> developmental process <br> 9.630e-32","GO:0009653 (2761) <br> anatomical structure morphogenesis <br> 2.289e-30","GO:0048514 (657) <br> blood vessel morphogenesis <br> 4.355e-30","GO:0007275 (4879) <br> multicellular organism development <br> 9.204e-30","GO:0001944 (793) <br> vasculature development <br> 2.093e-28","GO:0001568 (761) <br> blood vessel development <br> 2.287e-28","GO:0035295 (1181) <br> tube development <br> 2.592e-28","GO:0001525 (561) <br> angiogenesis <br> 4.526e-28","GO:0072359 (1216) <br> circulatory system development <br> 5.468e-28","GO:0030334 (984) <br> regulation of cell migration <br> 2.449e-27","GO:2000145 (1045) <br> regulation of cell motility <br> 9.803e-27","GO:0035239 (927) <br> tube morphogenesis <br> 8.726e-26","GO:0040012 (1089) <br> regulation of locomotion <br> 1.016e-25","GO:0007155 (1465) <br> cell adhesion <br> 1.051e-25","GO:0048870 (1760) <br> cell motility <br> 1.242e-25","GO:0016477 (1538) <br> cell migration <br> 7.149e-25","GO:0040011 (1273) <br> locomotion <br> 2.534e-24","GO:0048646 (1221) <br> anatomical structure formation involved in morphogenesis <br> 4.108e-24","GO:0050896 (10506) <br> response to stimulus <br> 1.547e-23","GO:0048513 (3260) <br> animal organ development <br> 1.563e-23","GO:0016043 (7612) <br> cellular component organization <br> 3.282e-23","GO:2000147 (607) <br> positive regulation of cell motility <br> 5.009e-23","GO:0030335 (577) <br> positive regulation of cell migration <br> 5.199e-23","GO:0030154 (4777) <br> cell differentiation <br> 1.026e-22","GO:0048869 (4778) <br> cellular developmental process <br> 1.046e-22","GO:0040017 (624) <br> positive regulation of locomotion <br> 1.653e-22","GO:0048518 (6589) <br> positive regulation of biological process <br> 1.918e-22","GO:0048522 (6078) <br> positive regulation of cellular process <br> 2.223e-22","GO:0071840 (7815) <br> cellular component organization or biogenesis <br> 8.078e-22","GO:0048519 (5966) <br> negative regulation of biological process <br> 1.759e-20","GO:0065007 (13527) <br> biological regulation <br> 1.233e-19","GO:0050789 (13144) <br> regulation of biological process <br> 1.761e-19","GO:0051239 (3165) <br> regulation of multicellular organismal process <br> 4.720e-19","GO:0048468 (3202) <br> cell development <br> 1.357e-18","GO:0065008 (3134) <br> regulation of biological quality <br> 4.284e-18","GO:0000902 (1045) <br> cell morphogenesis <br> 5.031e-18","GO:0048523 (5599) <br> negative regulation of cellular process <br> 6.873e-18","GO:0007399 (2529) <br> nervous system development <br> 8.553e-18","GO:0032501 (9132) <br> multicellular organismal process <br> 1.606e-17","GO:0051128 (2546) <br> regulation of cellular component organization <br> 3.430e-17","GO:0050794 (12593) <br> regulation of cellular process <br> 3.952e-17","GO:0009888 (2122) <br> tissue development <br> 4.207e-17","GO:0051716 (8752) <br> cellular response to stimulus <br> 7.221e-17","GO:0022603 (904) <br> regulation of anatomical structure morphogenesis <br> 7.614e-17","GO:0030155 (800) <br> regulation of cell adhesion <br> 1.051e-16","GO:0070887 (3162) <br> cellular response to chemical stimulus <br> 1.383e-16","GO:0008283 (2168) <br> cell population proliferation <br> 5.900e-16","GO:0022008 (1907) <br> neurogenesis <br> 1.150e-15","GO:0007166 (2766) <br> cell surface receptor signaling pathway <br> 1.588e-15","GO:0042127 (1799) <br> regulation of cell population proliferation <br> 1.993e-15","GO:0120036 (1666) <br> plasma membrane bounded cell projection organization <br> 3.584e-15","GO:0006950 (3936) <br> response to stress <br> 3.592e-15","GO:0050793 (2698) <br> regulation of developmental process <br> 4.594e-15","GO:0030030 (1706) <br> cell projection organization <br> 1.151e-14","GO:0048583 (3987) <br> regulation of response to stimulus <br> 1.468e-14","GO:0009628 (1300) <br> response to abiotic stimulus <br> 3.791e-14","GO:2000026 (1550) <br> regulation of multicellular organismal development <br> 4.041e-14","GO:0051240 (1804) <br> positive regulation of multicellular organismal process <br> 4.075e-14","GO:1901342 (312) <br> regulation of vasculature development <br> 4.190e-14","GO:0009893 (3745) <br> positive regulation of metabolic process <br> 4.190e-14","GO:0048699 (1651) <br> generation of neurons <br> 8.915e-14","GO:0031175 (1131) <br> neuron projection development <br> 1.216e-13","GO:0010604 (3427) <br> positive regulation of macromolecule metabolic process <br> 1.487e-13","GO:0030182 (1569) <br> neuron differentiation <br> 1.542e-13","GO:0007154 (7582) <br> cell communication <br> 2.128e-13","GO:0048666 (1287) <br> neuron development <br> 2.163e-13","GO:0045765 (306) <br> regulation of angiogenesis <br> 2.646e-13","GO:0007165 (6895) <br> signal transduction <br> 2.723e-13","GO:0009605 (3042) <br> response to external stimulus <br> 4.998e-13","GO:0051179 (5364) <br> localization <br> 8.347e-13","GO:0045785 (504) <br> positive regulation of cell adhesion <br> 8.404e-13","GO:0023052 (7472) <br> signaling <br> 1.052e-12","GO:0001666 (350) <br> response to hypoxia <br> 1.069e-12","GO:0044087 (1010) <br> regulation of cellular component biogenesis <br> 1.187e-12","GO:0009719 (1935) <br> response to endogenous stimulus <br> 1.275e-12","GO:0035556 (2610) <br> intracellular signal transduction <br> 1.275e-12","GO:0141124 (1794) <br> intracellular signaling cassette <br> 1.710e-12","GO:0007167 (970) <br> enzyme-linked receptor protein signaling pathway <br> 1.881e-12","GO:0048812 (699) <br> neuron projection morphogenesis <br> 2.544e-12","GO:0048584 (2359) <br> positive regulation of response to stimulus <br> 2.638e-12","GO:0001667 (446) <br> ameboidal-type cell migration <br> 2.904e-12","GO:0010033 (3435) <br> response to organic substance <br> 3.400e-12","GO:0120039 (714) <br> plasma membrane bounded cell projection morphogenesis <br> 4.957e-12","GO:0006119 (146) <br> oxidative phosphorylation <br> 5.224e-12","GO:0009966 (3005) <br> regulation of signal transduction <br> 5.869e-12","GO:0042773 (83) <br> ATP synthesis coupled electron transport <br> 5.886e-12","GO:0048858 (721) <br> cell projection morphogenesis <br> 6.726e-12","GO:0008219 (2125) <br> cell death <br> 7.243e-12","GO:0012501 (2125) <br> programmed cell death <br> 7.243e-12","GO:0036293 (383) <br> response to decreased oxygen levels <br> 9.225e-12","GO:0070482 (447) <br> response to oxygen levels <br> 1.564e-11","GO:0031325 (3434) <br> positive regulation of cellular metabolic process <br> 1.621e-11","GO:0071310 (2420) <br> cellular response to organic substance <br> 2.681e-11","GO:0051094 (1480) <br> positive regulation of developmental process <br> 2.804e-11","GO:0031399 (1102) <br> regulation of protein modification process <br> 3.833e-11","GO:0006810 (4397) <br> transport <br> 4.582e-11","GO:0010631 (316) <br> epithelial cell migration <br> 5.041e-11","GO:0070848 (761) <br> response to growth factor <br> 5.569e-11","GO:0051234 (4696) <br> establishment of localization <br> 5.688e-11","GO:0098609 (934) <br> cell-cell adhesion <br> 6.141e-11","GO:0006915 (2050) <br> apoptotic process <br> 6.150e-11","GO:0090132 (319) <br> epithelium migration <br> 6.210e-11","GO:0022904 (111) <br> respiratory electron transport chain <br> 6.498e-11","GO:0032989 (824) <br> cellular anatomical entity morphogenesis <br> 8.093e-11","GO:0023051 (3616) <br> regulation of signaling <br> 8.753e-11","GO:0071363 (732) <br> cellular response to growth factor stimulus <br> 9.114e-11","GO:0071495 (1629) <br> cellular response to endogenous stimulus <br> 9.308e-11","GO:0090130 (325) <br> tissue migration <br> 9.358e-11","GO:0051246 (1901) <br> regulation of protein metabolic process <br> 1.005e-10","GO:0048871 (912) <br> multicellular organismal-level homeostasis <br> 1.009e-10","GO:0009060 (202) <br> aerobic respiration <br> 1.015e-10","GO:0015988 (5) <br> energy coupled proton transmembrane transport, against electrochemical gradient <br> 1.039e-10","GO:0015990 (5) <br> electron transport coupled proton transport <br> 1.039e-10","GO:0007409 (481) <br> axonogenesis <br> 1.466e-10","GO:0042221 (5561) <br> response to chemical <br> 1.749e-10","GO:0051173 (2944) <br> positive regulation of nitrogen compound metabolic process <br> 2.274e-10","GO:0009967 (1613) <br> positive regulation of signal transduction <br> 2.327e-10","GO:0019646 (68) <br> aerobic electron transport chain <br> 2.449e-10","GO:0010646 (3616) <br> regulation of cell communication <br> 2.529e-10","GO:0000165 (758) <br> MAPK cascade <br> 2.579e-10","GO:0003013 (559) <br> circulatory system process <br> 2.639e-10","GO:0048667 (647) <br> cell morphogenesis involved in neuron differentiation <br> 2.810e-10","GO:0015980 (354) <br> energy derivation by oxidation of organic compounds <br> 2.885e-10","GO:0022900 (135) <br> electron transport chain <br> 3.221e-10","GO:0060429 (1309) <br> epithelium development <br> 3.553e-10","GO:0045597 (970) <br> positive regulation of cell differentiation <br> 3.768e-10","GO:0009891 (2827) <br> positive regulation of biosynthetic process <br> 6.126e-10","GO:0010557 (2697) <br> positive regulation of macromolecule biosynthetic process <br> 6.427e-10","GO:0061564 (548) <br> axon development <br> 6.468e-10","GO:0006996 (3518) <br> organelle organization <br> 6.858e-10","GO:0044419 (1940) <br> biological process involved in interspecies interaction between organisms <br> 8.185e-10","GO:0042592 (1835) <br> homeostatic process <br> 8.528e-10","GO:0042775 (81) <br> mitochondrial ATP synthesis coupled electron transport <br> 8.724e-10","GO:0019222 (6882) <br> regulation of metabolic process <br> 1.077e-09","GO:0043408 (670) <br> regulation of MAPK cascade <br> 1.089e-09","GO:0045333 (264) <br> cellular respiration <br> 1.144e-09","GO:0031328 (2799) <br> positive regulation of cellular biosynthetic process <br> 1.185e-09","GO:0006796 (2428) <br> phosphate-containing compound metabolic process <br> 1.364e-09","GO:0006793 (2450) <br> phosphorus metabolic process <br> 2.065e-09","GO:0097435 (806) <br> supramolecular fiber organization <br> 2.430e-09","GO:0051129 (751) <br> negative regulation of cellular component organization <br> 3.345e-09","GO:0030855 (777) <br> epithelial cell differentiation <br> 5.750e-09","GO:0043067 (1707) <br> regulation of programmed cell death <br> 6.010e-09","GO:0043542 (231) <br> endothelial cell migration <br> 6.811e-09","GO:0002376 (2819) <br> immune system process <br> 6.957e-09","GO:0006091 (491) <br> generation of precursor metabolites and energy <br> 7.416e-09","GO:1902600 (138) <br> proton transmembrane transport <br> 8.250e-09","GO:0019220 (1041) <br> regulation of phosphate metabolic process <br> 8.622e-09","GO:0051174 (1042) <br> regulation of phosphorus metabolic process <br> 8.881e-09","GO:0034330 (814) <br> cell junction organization <br> 9.816e-09","GO:0007010 (1466) <br> cytoskeleton organization <br> 1.015e-08","GO:1902531 (1722) <br> regulation of intracellular signal transduction <br> 1.053e-08","GO:0030029 (782) <br> actin filament-based process <br> 1.124e-08","GO:1902533 (1050) <br> positive regulation of intracellular signal transduction <br> 1.125e-08","GO:0051171 (5229) <br> regulation of nitrogen compound metabolic process <br> 1.141e-08","GO:1904018 (181) <br> positive regulation of vasculature development <br> 1.163e-08","GO:0034329 (471) <br> cell junction assembly <br> 1.347e-08","GO:0070372 (328) <br> regulation of ERK1 and ERK2 cascade <br> 1.357e-08","GO:0007507 (653) <br> heart development <br> 1.490e-08","GO:0080090 (5407) <br> regulation of primary metabolic process <br> 1.587e-08","GO:0031323 (6425) <br> regulation of cellular metabolic process <br> 1.853e-08","GO:0060255 (6358) <br> regulation of macromolecule metabolic process <br> 2.108e-08","GO:0001570 (92) <br> vasculogenesis <br> 2.121e-08","GO:0042981 (1658) <br> regulation of apoptotic process <br> 2.681e-08","GO:0009987 (20967) <br> cellular process <br> 2.697e-08","GO:0034097 (1188) <br> response to cytokine <br> 3.381e-08","GO:0008284 (1097) <br> positive regulation of cell population proliferation <br> 4.292e-08","GO:0010628 (1285) <br> positive regulation of gene expression <br> 4.324e-08","GO:0070371 (352) <br> ERK1 and ERK2 cascade <br> 5.128e-08","GO:0097485 (250) <br> neuron projection guidance <br> 6.909e-08","GO:0007411 (250) <br> axon guidance <br> 6.909e-08","GO:1901700 (2077) <br> response to oxygen-containing compound <br> 7.371e-08","GO:0002064 (240) <br> epithelial cell development <br> 1.032e-07","GO:0010632 (238) <br> regulation of epithelial cell migration <br> 1.041e-07","GO:0051247 (1130) <br> positive regulation of protein metabolic process <br> 1.050e-07","GO:0045766 (179) <br> positive regulation of angiogenesis <br> 1.052e-07","GO:0006468 (1236) <br> protein phosphorylation <br> 1.167e-07","GO:0010035 (651) <br> response to inorganic substance <br> 1.237e-07","GO:0033036 (3092) <br> macromolecule localization <br> 1.377e-07","GO:0045595 (1761) <br> regulation of cell differentiation <br> 1.515e-07","GO:0007044 (91) <br> cell-substrate junction assembly <br> 1.849e-07","GO:0022607 (4098) <br> cellular component assembly <br> 1.886e-07","GO:0016310 (1470) <br> phosphorylation <br> 1.962e-07","GO:0043069 (1069) <br> negative regulation of programmed cell death <br> 2.180e-07","GO:1901564 (5744) <br> organonitrogen compound metabolic process <br> 2.337e-07","GO:0023056 (1990) <br> positive regulation of signaling <br> 2.387e-07","GO:0010647 (1990) <br> positive regulation of cell communication <br> 2.387e-07","GO:0051641 (3502) <br> cellular localization <br> 2.435e-07","GO:0098657 (916) <br> import into cell <br> 2.512e-07","GO:0009607 (1819) <br> response to biotic stimulus <br> 2.594e-07","GO:0031589 (356) <br> cell-substrate adhesion <br> 2.627e-07","GO:0007159 (418) <br> leukocyte cell-cell adhesion <br> 2.630e-07","GO:0090066 (571) <br> regulation of anatomical structure size <br> 2.923e-07","GO:0010810 (222) <br> regulation of cell-substrate adhesion <br> 2.993e-07","GO:0007160 (224) <br> cell-matrix adhesion <br> 3.443e-07","GO:0120035 (751) <br> regulation of plasma membrane bounded cell projection organization <br> 3.455e-07","GO:1902903 (388) <br> regulation of supramolecular fiber organization <br> 3.484e-07","GO:0060284 (941) <br> regulation of cell development <br> 3.533e-07","GO:0051707 (1778) <br> response to other organism <br> 3.867e-07","GO:0003007 (273) <br> heart morphogenesis <br> 3.966e-07","GO:0043066 (1039) <br> negative regulation of apoptotic process <br> 4.046e-07","GO:0043207 (1782) <br> response to external biotic stimulus <br> 4.171e-07","GO:0150115 (98) <br> cell-substrate junction organization <br> 4.184e-07","GO:0061061 (733) <br> muscle structure development <br> 4.678e-07","GO:0022407 (498) <br> regulation of cell-cell adhesion <br> 4.880e-07","GO:0070727 (2604) <br> cellular macromolecule localization <br> 5.078e-07","GO:0031344 (765) <br> regulation of cell projection organization <br> 5.428e-07","GO:0032879 (2161) <br> regulation of localization <br> 5.823e-07","GO:0150105 (24) <br> protein localization to cell-cell junction <br> 6.559e-07","GO:0022409 (327) <br> positive regulation of cell-cell adhesion <br> 6.622e-07","GO:0036211 (2728) <br> protein modification process <br> 6.625e-07","GO:0003158 (141) <br> endothelium development <br> 7.458e-07","GO:0001932 (816) <br> regulation of protein phosphorylation <br> 8.825e-07","GO:0045937 (680) <br> positive regulation of phosphate metabolic process <br> 9.143e-07","GO:0010562 (680) <br> positive regulation of phosphorus metabolic process <br> 9.143e-07","GO:0002040 (133) <br> sprouting angiogenesis <br> 9.265e-07","GO:0001935 (169) <br> endothelial cell proliferation <br> 1.101e-06","GO:0030036 (695) <br> actin cytoskeleton organization <br> 1.185e-06","GO:0042325 (877) <br> regulation of phosphorylation <br> 1.314e-06","GO:0010720 (535) <br> positive regulation of cell development <br> 1.389e-06","GO:0010038 (395) <br> response to metal ion <br> 1.463e-06","GO:0031401 (742) <br> positive regulation of protein modification process <br> 1.584e-06","GO:0006897 (701) <br> endocytosis <br> 1.644e-06","GO:0009790 (1243) <br> embryo development <br> 1.733e-06","GO:0001936 (146) <br> regulation of endothelial cell proliferation <br> 1.901e-06","GO:1901888 (236) <br> regulation of cell junction assembly <br> 2.070e-06","GO:0008015 (530) <br> blood circulation <br> 2.611e-06","GO:0010811 (136) <br> positive regulation of cell-substrate adhesion <br> 3.095e-06","GO:0050767 (467) <br> regulation of neurogenesis <br> 3.334e-06","GO:0008104 (2594) <br> protein localization <br> 3.487e-06","GO:0050673 (489) <br> epithelial cell proliferation <br> 3.618e-06","GO:0008285 (701) <br> negative regulation of cell population proliferation <br> 3.695e-06","GO:0016032 (350) <br> viral process <br> 4.135e-06","GO:0044085 (4368) <br> cellular component biogenesis <br> 4.146e-06","GO:0042776 (63) <br> proton motive force-driven mitochondrial ATP synthesis <br> 4.208e-06","GO:0050678 (405) <br> regulation of epithelial cell proliferation <br> 4.235e-06","GO:0038084 (44) <br> vascular endothelial growth factor signaling pathway <br> 4.357e-06","GO:0048585 (1723) <br> negative regulation of response to stimulus <br> 4.453e-06","GO:0045446 (118) <br> endothelial cell differentiation <br> 4.657e-06","GO:0051050 (975) <br> positive regulation of transport <br> 4.960e-06","GO:0007178 (400) <br> transmembrane receptor protein serine/threonine kinase signaling pathway <br> 5.046e-06","GO:0009611 (541) <br> response to wounding <br> 5.270e-06","GO:0043412 (2937) <br> macromolecule modification <br> 5.419e-06","GO:0009617 (968) <br> response to bacterium <br> 5.496e-06","GO:0051049 (1749) <br> regulation of transport <br> 5.674e-06","GO:0019538 (4698) <br> protein metabolic process <br> 6.161e-06","GO:0071345 (1074) <br> cellular response to cytokine stimulus <br> 6.408e-06","GO:0006898 (254) <br> receptor-mediated endocytosis <br> 6.643e-06","GO:0071526 (51) <br> semaphorin-plexin signaling pathway <br> 7.177e-06","GO:0015986 (70) <br> proton motive force-driven ATP synthesis <br> 7.218e-06","GO:0040007 (1044) <br> growth <br> 7.795e-06","GO:0010594 (177) <br> regulation of endothelial cell migration <br> 8.619e-06","GO:0010634 (157) <br> positive regulation of epithelial cell migration <br> 9.134e-06","GO:0060341 (1086) <br> regulation of cellular localization <br> 9.351e-06","GO:0002682 (1575) <br> regulation of immune system process <br> 9.465e-06","GO:0006811 (1239) <br> monoatomic ion transport <br> 9.584e-06","GO:0043534 (123) <br> blood vessel endothelial cell migration <br> 1.060e-05","GO:0001952 (123) <br> regulation of cell-matrix adhesion <br> 1.060e-05","GO:0046034 (230) <br> ATP metabolic process <br> 1.124e-05","GO:0051649 (2018) <br> establishment of localization in cell <br> 1.127e-05","GO:0001775 (1160) <br> cell activation <br> 1.215e-05","GO:0051130 (1248) <br> positive regulation of cellular component organization <br> 1.221e-05","GO:0050769 (310) <br> positive regulation of neurogenesis <br> 1.375e-05","GO:0035924 (62) <br> cellular response to vascular endothelial growth factor stimulus <br> 1.407e-05","GO:0009725 (1008) <br> response to hormone <br> 1.605e-05","GO:0051493 (522) <br> regulation of cytoskeleton organization <br> 1.703e-05","GO:0006120 (28) <br> mitochondrial electron transport, NADH to ubiquinone <br> 1.771e-05","GO:0040013 (350) <br> negative regulation of locomotion <br> 1.947e-05","GO:0008361 (214) <br> regulation of cell size <br> 2.157e-05","GO:0009889 (5706) <br> regulation of biosynthetic process <br> 2.309e-05","GO:0023057 (1453) <br> negative regulation of signaling <br> 2.326e-05","GO:0051960 (562) <br> regulation of nervous system development <br> 2.353e-05","GO:0010648 (1456) <br> negative regulation of cell communication <br> 2.458e-05","GO:0090109 (58) <br> regulation of cell-substrate junction assembly <br> 2.525e-05","GO:0051893 (58) <br> regulation of focal adhesion assembly <br> 2.525e-05","GO:0032535 (399) <br> regulation of cellular component size <br> 2.621e-05","GO:0045321 (1031) <br> leukocyte activation <br> 2.665e-05","GO:0019058 (272) <br> viral life cycle <br> 2.914e-05","GO:0003018 (237) <br> vascular process in circulatory system <br> 3.156e-05","GO:0010556 (5506) <br> regulation of macromolecule biosynthetic process <br> 3.228e-05","GO:0043410 (480) <br> positive regulation of MAPK cascade <br> 3.610e-05","GO:0001954 (65) <br> positive regulation of cell-matrix adhesion <br> 4.177e-05","GO:1903037 (380) <br> regulation of leukocyte cell-cell adhesion <br> 4.328e-05","GO:1901698 (1283) <br> response to nitrogen compound <br> 4.453e-05","GO:0009205 (254) <br> purine ribonucleoside triphosphate metabolic process <br> 4.589e-05","GO:0006754 (102) <br> ATP biosynthetic process <br> 4.879e-05","GO:0150116 (63) <br> regulation of cell-substrate junction organization <br> 4.942e-05","GO:0033043 (1178) <br> regulation of organelle organization <br> 4.971e-05","GO:1901652 (568) <br> response to peptide <br> 5.415e-05","GO:0060485 (297) <br> mesenchyme development <br> 5.540e-05","GO:0030097 (1044) <br> hemopoiesis <br> 5.704e-05","GO:0030099 (468) <br> myeloid cell differentiation <br> 5.712e-05","GO:0030324 (234) <br> lung development <br> 6.106e-05","GO:0007169 (599) <br> transmembrane receptor protein tyrosine kinase signaling pathway <br> 6.681e-05","GO:0031326 (5663) <br> regulation of cellular biosynthetic process <br> 6.759e-05","GO:0009144 (262) <br> purine nucleoside triphosphate metabolic process <br> 7.081e-05","GO:0009199 (262) <br> ribonucleoside triphosphate metabolic process <br> 7.081e-05","GO:0016192 (1517) <br> vesicle-mediated transport <br> 7.139e-05","GO:0030323 (237) <br> respiratory tube development <br> 7.231e-05","GO:0000122 (1019) <br> negative regulation of transcription by RNA polymerase II <br> 7.643e-05","GO:0007423 (650) <br> sensory organ development <br> 7.795e-05","GO:0044057 (614) <br> regulation of system process <br> 7.836e-05","GO:0009206 (113) <br> purine ribonucleoside triphosphate biosynthetic process <br> 8.173e-05","GO:0051172 (2219) <br> negative regulation of nitrogen compound metabolic process <br> 8.395e-05","GO:0009887 (1094) <br> animal organ morphogenesis <br> 8.461e-05","GO:0009145 (114) <br> purine nucleoside triphosphate biosynthetic process <br> 8.543e-05","GO:0007417 (1006) <br> central nervous system development <br> 8.623e-05","GO:0006812 (1022) <br> monoatomic cation transport <br> 8.895e-05","GO:0010468 (5382) <br> regulation of gene expression <br> 9.178e-05","GO:0071702 (2419) <br> organic substance transport <br> 9.288e-05","GO:0060560 (271) <br> developmental growth involved in morphogenesis <br> 9.352e-05","GO:0001569 (39) <br> branching involved in blood vessel morphogenesis <br> 1.010e-04","GO:0141091 (351) <br> transforming growth factor beta receptor superfamily signaling pathway <br> 1.033e-04","GO:0033993 (1155) <br> response to lipid <br> 1.038e-04","GO:0090287 (318) <br> regulation of cellular response to growth factor stimulus <br> 1.046e-04","GO:0051241 (1214) <br> negative regulation of multicellular organismal process <br> 1.057e-04","GO:0009201 (119) <br> ribonucleoside triphosphate biosynthetic process <br> 1.060e-04","GO:0010595 (111) <br> positive regulation of endothelial cell migration <br> 1.392e-04","GO:0001885 (71) <br> endothelial cell development <br> 1.401e-04","GO:0009142 (126) <br> nucleoside triphosphate biosynthetic process <br> 1.412e-04","GO:0007420 (723) <br> brain development <br> 1.671e-04","GO:0009968 (1311) <br> negative regulation of signal transduction <br> 1.687e-04","GO:0009141 (279) <br> nucleoside triphosphate metabolic process <br> 1.691e-04","GO:0051962 (368) <br> positive regulation of nervous system development <br> 1.892e-04","GO:0060322 (778) <br> head development <br> 1.902e-04","GO:0046689 (14) <br> response to mercury ion <br> 1.959e-04","GO:0016049 (537) <br> cell growth <br> 2.030e-04","GO:1902396 (7) <br> protein localization to bicellular tight junction <br> 2.031e-04","GO:0046907 (1337) <br> intracellular transport <br> 2.059e-04","GO:0031324 (3108) <br> negative regulation of cellular metabolic process <br> 2.152e-04","GO:0071705 (1959) <br> nitrogen compound transport <br> 2.377e-04","GO:0090100 (102) <br> positive regulation of transmembrane receptor protein serine/threonine kinase signaling pathway <br> 2.583e-04","GO:0015669 (24) <br> gas transport <br> 2.603e-04","GO:0014070 (1137) <br> response to organic cyclic compound <br> 2.659e-04","GO:0044403 (289) <br> biological process involved in symbiotic interaction <br> 2.738e-04","GO:0043010 (375) <br> camera-type eye development <br> 2.784e-04","GO:0060541 (263) <br> respiratory system development <br> 2.833e-04","GO:0048589 (785) <br> developmental growth <br> 3.235e-04","GO:0016525 (109) <br> negative regulation of angiogenesis <br> 3.303e-04","GO:0048041 (80) <br> focal adhesion assembly <br> 3.319e-04","GO:0045216 (229) <br> cell-cell junction organization <br> 3.343e-04","GO:0033554 (1799) <br> cellular response to stress <br> 3.421e-04","GO:0010243 (1161) <br> response to organonitrogen compound <br> 3.453e-04","GO:0048729 (651) <br> tissue morphogenesis <br> 3.671e-04","GO:0032880 (977) <br> regulation of protein localization <br> 3.695e-04","GO:0032101 (1126) <br> regulation of response to external stimulus <br> 3.800e-04","GO:2000181 (111) <br> negative regulation of blood vessel morphogenesis <br> 3.865e-04","GO:1903829 (523) <br> positive regulation of protein localization <br> 4.172e-04","GO:1901343 (112) <br> negative regulation of vasculature development <br> 4.175e-04","GO:0032970 (393) <br> regulation of actin filament-based process <br> 4.285e-04","GO:0030336 (304) <br> negative regulation of cell migration <br> 4.304e-04","GO:0048010 (50) <br> vascular endothelial growth factor receptor signaling pathway <br> 4.449e-04","GO:0010257 (62) <br> NADH dehydrogenase complex assembly <br> 4.734e-04","GO:0032981 (62) <br> mitochondrial respiratory chain complex I assembly <br> 4.734e-04","GO:0007043 (168) <br> cell-cell junction assembly <br> 4.929e-04","GO:1902904 (172) <br> negative regulation of supramolecular fiber organization <br> 5.052e-04","GO:0048675 (135) <br> axon extension <br> 5.179e-04","GO:0007015 (456) <br> actin filament organization <br> 5.346e-04","GO:1990138 (200) <br> neuron projection extension <br> 5.874e-04","GO:0050920 (236) <br> regulation of chemotaxis <br> 6.241e-04","GO:0009892 (3336) <br> negative regulation of metabolic process <br> 6.494e-04","GO:0001934 (567) <br> positive regulation of protein phosphorylation <br> 6.539e-04","GO:0048588 (274) <br> developmental cell growth <br> 6.590e-04","GO:0030218 (155) <br> erythrocyte differentiation <br> 6.997e-04","GO:0048762 (239) <br> mesenchymal cell differentiation <br> 7.275e-04","GO:0042110 (611) <br> T cell activation <br> 7.936e-04","GO:0042060 (377) <br> wound healing <br> 8.059e-04","GO:1903039 (280) <br> positive regulation of leukocyte cell-cell adhesion <br> 8.061e-04","GO:1901699 (823) <br> cellular response to nitrogen compound <br> 8.195e-04","GO:2000146 (319) <br> negative regulation of cell motility <br> 8.221e-04","GO:0050900 (378) <br> leukocyte migration <br> 8.522e-04","GO:0120032 (211) <br> regulation of plasma membrane bounded cell projection assembly <br> 8.547e-04","GO:0008360 (141) <br> regulation of cell shape <br> 8.549e-04","GO:0050770 (174) <br> regulation of axonogenesis <br> 9.018e-04","GO:0110053 (277) <br> regulation of actin filament organization <br> 9.495e-04","GO:0060491 (214) <br> regulation of cell projection assembly <br> 9.939e-04","GO:0003382 (43) <br> epithelial cell morphogenesis <br> 1.004e-03","GO:0034504 (329) <br> protein localization to nucleus <br> 1.028e-03","GO:0043933 (3478) <br> protein-containing complex organization <br> 1.033e-03","GO:0061437 (32) <br> renal system vasculature development <br> 1.037e-03","GO:0061440 (32) <br> kidney vasculature development <br> 1.037e-03","GO:0055008 (68) <br> cardiac muscle tissue morphogenesis <br> 1.092e-03","GO:0001894 (271) <br> tissue homeostasis <br> 1.110e-03","GO:0060249 (271) <br> anatomical structure homeostasis <br> 1.110e-03","GO:0006935 (455) <br> chemotaxis <br> 1.172e-03","GO:1901566 (1707) <br> organonitrogen compound biosynthetic process <br> 1.177e-03","GO:0051248 (686) <br> negative regulation of protein metabolic process <br> 1.212e-03","GO:0042330 (457) <br> taxis <br> 1.252e-03","GO:0010605 (3109) <br> negative regulation of macromolecule metabolic process <br> 1.260e-03","GO:0098662 (774) <br> inorganic cation transmembrane transport <br> 1.273e-03","GO:0015671 (14) <br> oxygen transport <br> 1.299e-03","GO:0034101 (166) <br> erythrocyte homeostasis <br> 1.390e-03","GO:0009792 (808) <br> embryo development ending in birth or egg hatching <br> 1.502e-03","GO:0098655 (797) <br> monoatomic cation transmembrane transport <br> 1.550e-03","GO:0042327 (597) <br> positive regulation of phosphorylation <br> 1.552e-03","GO:0002684 (1105) <br> positive regulation of immune system process <br> 1.593e-03","GO:0045892 (1409) <br> negative regulation of DNA-templated transcription <br> 1.619e-03","GO:0032388 (154) <br> positive regulation of intracellular transport <br> 1.636e-03","GO:0044089 (537) <br> positive regulation of cellular component biogenesis <br> 1.642e-03","GO:1901701 (1497) <br> cellular response to oxygen-containing compound <br> 1.663e-03","GO:0032956 (344) <br> regulation of actin cytoskeleton organization <br> 1.688e-03","GO:0071674 (189) <br> mononuclear cell migration <br> 1.728e-03","GO:0009152 (209) <br> purine ribonucleotide biosynthetic process <br> 1.748e-03","GO:1902679 (1417) <br> negative regulation of RNA biosynthetic process <br> 1.848e-03","GO:0032835 (75) <br> glomerulus development <br> 1.882e-03","GO:0060041 (188) <br> retina development in camera-type eye <br> 1.914e-03","GO:0046718 (124) <br> symbiont entry into host cell <br> 2.088e-03","GO:0002262 (210) <br> myeloid cell homeostasis <br> 2.089e-03","GO:0001654 (431) <br> eye development <br> 2.117e-03","GO:0031400 (337) <br> negative regulation of protein modification process <br> 2.138e-03","GO:1902414 (128) <br> protein localization to cell junction <br> 2.185e-03","GO:0051592 (160) <br> response to calcium ion <br> 2.230e-03","GO:0046649 (848) <br> lymphocyte activation <br> 2.313e-03","GO:0150063 (434) <br> visual system development <br> 2.337e-03","GO:0140962 (84) <br> multicellular organismal-level chemical homeostasis <br> 2.340e-03","GO:0009260 (224) <br> ribonucleotide biosynthetic process <br> 2.459e-03","GO:0042692 (444) <br> muscle cell differentiation <br> 2.509e-03","GO:0006952 (1812) <br> defense response <br> 2.525e-03","GO:0014910 (99) <br> regulation of smooth muscle cell migration <br> 2.536e-03","GO:0030100 (327) <br> regulation of endocytosis <br> 2.602e-03","GO:0061384 (34) <br> heart trabecula morphogenesis <br> 2.622e-03","GO:0071560 (269) <br> cellular response to transforming growth factor beta stimulus <br> 2.645e-03","GO:0044409 (128) <br> symbiont entry into host <br> 2.656e-03","GO:0048841 (33) <br> regulation of axon extension involved in axon guidance <br> 2.674e-03","GO:0048880 (439) <br> sensory system development <br> 2.751e-03","GO:0046390 (230) <br> ribose phosphate biosynthetic process <br> 2.800e-03","GO:1900180 (164) <br> regulation of protein localization to nucleus <br> 2.846e-03","GO:0050870 (256) <br> positive regulation of T cell activation <br> 2.931e-03","GO:0055082 (768) <br> intracellular chemical homeostasis <br> 2.969e-03","GO:0048878 (1091) <br> chemical homeostasis <br> 3.014e-03","GO:0098660 (881) <br> inorganic ion transmembrane transport <br> 3.030e-03","GO:0030900 (447) <br> forebrain development <br> 3.056e-03","GO:0071559 (274) <br> response to transforming growth factor beta <br> 3.287e-03","GO:0006164 (240) <br> purine nucleotide biosynthetic process <br> 3.451e-03","GO:0051093 (1078) <br> negative regulation of developmental process <br> 3.461e-03","GO:0071496 (404) <br> cellular response to external stimulus <br> 3.493e-03","GO:0033108 (102) <br> mitochondrial respiratory chain complex assembly <br> 3.525e-03","GO:0051492 (96) <br> regulation of stress fiber assembly <br> 3.535e-03","GO:0010639 (371) <br> negative regulation of organelle organization <br> 3.578e-03","GO:0030509 (165) <br> BMP signaling pathway <br> 3.586e-03","GO:0048545 (371) <br> response to steroid hormone <br> 3.667e-03","GO:0045934 (1651) <br> negative regulation of nucleobase-containing compound metabolic process <br> 3.714e-03","GO:0032870 (681) <br> cellular response to hormone stimulus <br> 3.814e-03","GO:0051494 (172) <br> negative regulation of cytoskeleton organization <br> 3.815e-03","GO:0051253 (1527) <br> negative regulation of RNA metabolic process <br> 3.923e-03","GO:0051017 (168) <br> actin filament bundle assembly <br> 3.936e-03","GO:0050921 (161) <br> positive regulation of chemotaxis <br> 3.951e-03","GO:0060841 (20) <br> venous blood vessel development <br> 3.982e-03","GO:0070374 (228) <br> positive regulation of ERK1 and ERK2 cascade <br> 4.034e-03","GO:0050680 (152) <br> negative regulation of epithelial cell proliferation <br> 4.047e-03","GO:0072522 (248) <br> purine-containing compound biosynthetic process <br> 4.053e-03","GO:0051894 (28) <br> positive regulation of focal adhesion assembly <br> 4.084e-03","GO:0032103 (654) <br> positive regulation of response to external stimulus <br> 4.116e-03","GO:0003151 (78) <br> outflow tract morphogenesis <br> 4.124e-03","GO:0019725 (869) <br> cellular homeostasis <br> 4.140e-03","GO:0060415 (81) <br> muscle tissue morphogenesis <br> 4.233e-03","GO:0031032 (207) <br> actomyosin structure organization <br> 4.381e-03","GO:0051701 (185) <br> biological process involved in interaction with host <br> 4.603e-03","GO:0061572 (171) <br> actin filament bundle organization <br> 4.618e-03","GO:0030516 (106) <br> regulation of axon extension <br> 4.639e-03","GO:0110020 (109) <br> regulation of actomyosin structure organization <br> 4.790e-03","GO:0035886 (41) <br> vascular associated smooth muscle cell differentiation <br> 4.794e-03","GO:1903670 (44) <br> regulation of sprouting angiogenesis <br> 4.913e-03","GO:0001817 (824) <br> regulation of cytokine production <br> 5.005e-03","GO:0071417 (747) <br> cellular response to organonitrogen compound <br> 5.357e-03","GO:0001816 (830) <br> cytokine production <br> 5.516e-03","GO:0031623 (145) <br> receptor internalization <br> 5.723e-03","GO:0043434 (489) <br> response to peptide hormone <br> 5.800e-03","GO:0051254 (1877) <br> positive regulation of RNA metabolic process <br> 5.814e-03","GO:0045229 (304) <br> external encapsulating structure organization <br> 5.882e-03","GO:0030198 (304) <br> extracellular matrix organization <br> 5.882e-03","GO:0034220 (975) <br> monoatomic ion transmembrane transport <br> 5.947e-03","GO:0050878 (354) <br> regulation of body fluid levels <br> 6.023e-03","GO:1990845 (159) <br> adaptive thermogenesis <br> 6.078e-03","GO:0014909 (110) <br> smooth muscle cell migration <br> 6.143e-03","GO:0010975 (558) <br> regulation of neuron projection development <br> 6.185e-03","GO:0033365 (996) <br> protein localization to organelle <br> 6.252e-03","GO:0043062 (306) <br> extracellular structure organization <br> 6.347e-03","GO:1902284 (38) <br> neuron projection extension involved in neuron projection guidance <br> 6.373e-03","GO:0048846 (38) <br> axon extension involved in axon guidance <br> 6.373e-03","GO:0007264 (437) <br> small GTPase-mediated signal transduction <br> 6.455e-03","GO:0003206 (141) <br> cardiac chamber morphogenesis <br> 6.526e-03","GO:0031100 (84) <br> animal organ regeneration <br> 6.543e-03","GO:0032386 (293) <br> regulation of intracellular transport <br> 6.623e-03","GO:0045935 (2083) <br> positive regulation of nucleobase-containing compound metabolic process <br> 6.650e-03","GO:0009165 (278) <br> nucleotide biosynthetic process <br> 7.086e-03","GO:0031341 (138) <br> regulation of cell killing <br> 7.276e-03","GO:0003073 (120) <br> regulation of systemic arterial blood pressure <br> 7.322e-03","GO:1901293 (281) <br> nucleoside phosphate biosynthetic process <br> 7.467e-03","GO:0040008 (658) <br> regulation of growth <br> 7.571e-03","GO:0000003 (1859) <br> reproduction <br> 7.762e-03","GO:0018193 (650) <br> peptidyl-amino acid modification <br> 7.776e-03","GO:0071773 (179) <br> cellular response to BMP stimulus <br> 7.919e-03","GO:0071772 (179) <br> response to BMP <br> 7.919e-03","GO:0060627 (587) <br> regulation of vesicle-mediated transport <br> 8.116e-03","GO:0071407 (672) <br> cellular response to organic cyclic compound <br> 8.174e-03","GO:0009410 (503) <br> response to xenobiotic stimulus <br> 8.444e-03","GO:0060326 (321) <br> cell chemotaxis <br> 8.502e-03","GO:0048644 (89) <br> muscle organ morphogenesis <br> 8.658e-03","GO:0051222 (300) <br> positive regulation of protein transport <br> 8.700e-03","GO:0048842 (7) <br> positive regulation of axon extension involved in axon guidance <br> 8.818e-03","GO:0022604 (271) <br> regulation of cell morphogenesis <br> 8.969e-03","GO:0001938 (102) <br> positive regulation of endothelial cell proliferation <br> 9.239e-03","GO:0045773 (49) <br> positive regulation of axon extension <br> 9.462e-03","GO:0150117 (33) <br> positive regulation of cell-substrate junction organization <br> 9.569e-03","GO:0045944 (1315) <br> positive regulation of transcription by RNA polymerase II <br> 9.718e-03","GO:0034341 (160) <br> response to type II interferon <br> 1.003e-02","GO:0071383 (228) <br> cellular response to steroid hormone stimulus <br> 1.028e-02","GO:0009791 (118) <br> post-embryonic development <br> 1.029e-02","GO:0045893 (1726) <br> positive regulation of DNA-templated transcription <br> 1.030e-02","GO:0090090 (133) <br> negative regulation of canonical Wnt signaling pathway <br> 1.058e-02","GO:0060395 (104) <br> SMAD protein signal transduction <br> 1.068e-02","GO:1902680 (1729) <br> positive regulation of RNA biosynthetic process <br> 1.073e-02","GO:2000628 (91) <br> regulation of miRNA metabolic process <br> 1.107e-02","GO:0043009 (791) <br> chordate embryonic development <br> 1.115e-02","GO:0032231 (112) <br> regulation of actin filament bundle assembly <br> 1.129e-02","GO:0043149 (113) <br> stress fiber assembly <br> 1.206e-02","GO:0030038 (113) <br> contractile actin filament bundle assembly <br> 1.206e-02","GO:0003205 (186) <br> cardiac chamber development <br> 1.223e-02","GO:0090407 (558) <br> organophosphate biosynthetic process <br> 1.249e-02","GO:0001945 (30) <br> lymph vessel development <br> 1.254e-02","GO:0044271 (4715) <br> cellular nitrogen compound biosynthetic process <br> 1.258e-02","GO:0048754 (173) <br> branching morphogenesis of an epithelial tube <br> 1.291e-02","GO:0061387 (120) <br> regulation of extent of cell growth <br> 1.297e-02","GO:0042255 (69) <br> ribosome assembly <br> 1.405e-02","GO:0032496 (451) <br> response to lipopolysaccharide <br> 1.408e-02","GO:0050772 (94) <br> positive regulation of axonogenesis <br> 1.412e-02","GO:0007517 (360) <br> muscle organ development <br> 1.428e-02","GO:0061028 (52) <br> establishment of endothelial barrier <br> 1.468e-02","GO:0009743 (286) <br> response to carbohydrate <br> 1.484e-02","GO:0055085 (1525) <br> transmembrane transport <br> 1.580e-02","GO:0120161 (140) <br> regulation of cold-induced thermogenesis <br> 1.609e-02","GO:0001558 (447) <br> regulation of cell growth <br> 1.617e-02","GO:0050863 (382) <br> regulation of T cell activation <br> 1.627e-02","GO:0104004 (383) <br> cellular response to environmental stimulus <br> 1.629e-02","GO:0071214 (383) <br> cellular response to abiotic stimulus <br> 1.629e-02","GO:0072001 (358) <br> renal system development <br> 1.651e-02","GO:0051145 (76) <br> smooth muscle cell differentiation <br> 1.664e-02","GO:0034333 (13) <br> adherens junction assembly <br> 1.695e-02","GO:0098542 (1281) <br> defense response to other organism <br> 1.713e-02","GO:0006357 (2485) <br> regulation of transcription by RNA polymerase II <br> 1.717e-02","GO:0014812 (125) <br> muscle cell migration <br> 1.761e-02","GO:0022414 (1850) <br> reproductive process <br> 1.762e-02","GO:0048844 (77) <br> artery morphogenesis <br> 1.769e-02","GO:0051251 (334) <br> positive regulation of lymphocyte activation <br> 1.773e-02","GO:0006972 (40) <br> hyperosmotic response <br> 1.793e-02","GO:0072012 (30) <br> glomerulus vasculature development <br> 1.794e-02","GO:0106106 (142) <br> cold-induced thermogenesis <br> 1.805e-02","GO:0018130 (3963) <br> heterocycle biosynthetic process <br> 1.846e-02","GO:0072006 (169) <br> nephron development <br> 1.856e-02","GO:0008217 (215) <br> regulation of blood pressure <br> 1.863e-02","GO:0034654 (3888) <br> nucleobase-containing compound biosynthetic process <br> 1.867e-02","GO:0007163 (227) <br> establishment or maintenance of cell polarity <br> 1.870e-02","GO:0001886 (14) <br> endothelial cell morphogenesis <br> 1.875e-02","GO:0009612 (261) <br> response to mechanical stimulus <br> 1.893e-02","GO:0006366 (2571) <br> transcription by RNA polymerase II <br> 2.030e-02","GO:0045428 (71) <br> regulation of nitric oxide biosynthetic process <br> 2.077e-02","GO:0007266 (143) <br> Rho protein signal transduction <br> 2.117e-02","GO:0120162 (92) <br> positive regulation of cold-induced thermogenesis <br> 2.127e-02","GO:0009749 (248) <br> response to glucose <br> 2.143e-02","GO:0045087 (977) <br> innate immune response <br> 2.151e-02","GO:0002237 (467) <br> response to molecule of bacterial origin <br> 2.174e-02","GO:0051897 (204) <br> positive regulation of phosphatidylinositol 3-kinase/protein kinase B signal transduction <br> 2.219e-02","GO:0010977 (162) <br> negative regulation of neuron projection development <br> 2.244e-02","GO:0060070 (297) <br> canonical Wnt signaling pathway <br> 2.309e-02","GO:0002720 (106) <br> positive regulation of cytokine production involved in immune response <br> 2.311e-02","GO:1990778 (386) <br> protein localization to cell periphery <br> 2.317e-02","GO:1900182 (101) <br> positive regulation of protein localization to nucleus <br> 2.407e-02","GO:0007267 (1734) <br> cell-cell signaling <br> 2.527e-02","GO:0009746 (253) <br> response to hexose <br> 2.583e-02","GO:0007179 (204) <br> transforming growth factor beta receptor signaling pathway <br> 2.659e-02","GO:0001659 (188) <br> temperature homeostasis <br> 2.679e-02","GO:0010586 (103) <br> miRNA metabolic process <br> 2.781e-02","GO:0051152 (16) <br> positive regulation of smooth muscle cell differentiation <br> 2.791e-02","GO:0001819 (523) <br> positive regulation of cytokine production <br> 2.793e-02","GO:0080164 (75) <br> regulation of nitric oxide metabolic process <br> 2.854e-02","GO:0031327 (2768) <br> negative regulation of cellular biosynthetic process <br> 2.891e-02","GO:1901137 (648) <br> carbohydrate derivative biosynthetic process <br> 2.926e-02","GO:0046822 (135) <br> regulation of nucleocytoplasmic transport <br> 2.935e-02","GO:0002683 (540) <br> negative regulation of immune system process <br> 2.951e-02","GO:0061383 (51) <br> trabecula morphogenesis <br> 3.031e-02","GO:0046824 (75) <br> positive regulation of nucleocytoplasmic transport <br> 3.070e-02","GO:0045807 (192) <br> positive regulation of endocytosis <br> 3.243e-02","GO:0010558 (2710) <br> negative regulation of macromolecule biosynthetic process <br> 3.346e-02","GO:0061138 (207) <br> morphogenesis of a branching epithelium <br> 3.360e-02","GO:0002694 (578) <br> regulation of leukocyte activation <br> 3.385e-02","GO:0038165 (4) <br> oncostatin-M-mediated signaling pathway <br> 3.392e-02","GO:0055010 (52) <br> ventricular cardiac muscle tissue morphogenesis <br> 3.397e-02","GO:0034284 (261) <br> response to monosaccharide <br> 3.453e-02","GO:0022411 (482) <br> cellular component disassembly <br> 3.535e-02","GO:0072659 (306) <br> protein localization to plasma membrane <br> 3.561e-02","GO:0009890 (2791) <br> negative regulation of biosynthetic process <br> 3.638e-02","GO:0003208 (76) <br> cardiac ventricle morphogenesis <br> 3.752e-02","GO:0043129 (29) <br> surfactant homeostasis <br> 3.827e-02","GO:0060840 (110) <br> artery development <br> 3.856e-02","GO:0019438 (3973) <br> aromatic compound biosynthetic process <br> 3.859e-02","GO:0001937 (43) <br> negative regulation of endothelial cell proliferation <br> 3.923e-02","GO:1901653 (398) <br> cellular response to peptide <br> 3.936e-02","GO:0010814 (3) <br> substance P catabolic process <br> 3.966e-02","GO:1901362 (4131) <br> organic cyclic compound biosynthetic process <br> 3.976e-02","GO:0006338 (624) <br> chromatin remodeling <br> 3.990e-02","GO:0018108 (239) <br> peptidyl-tyrosine phosphorylation <br> 4.139e-02","GO:0002696 (366) <br> positive regulation of leukocyte activation <br> 4.152e-02","GO:0031667 (576) <br> response to nutrient levels <br> 4.276e-02","GO:0018212 (241) <br> peptidyl-tyrosine modification <br> 4.471e-02","GO:0048872 (381) <br> homeostasis of number of cells <br> 4.521e-02","GO:0090092 (261) <br> regulation of transmembrane receptor protein serine/threonine kinase signaling pathway <br> 4.608e-02","GO:0007162 (301) <br> negative regulation of cell adhesion <br> 4.616e-02","GO:0002009 (536) <br> morphogenesis of an epithelium <br> 4.668e-02","GO:0030178 (160) <br> negative regulation of Wnt signaling pathway <br> 4.704e-02","GO:0003231 (141) <br> cardiac ventricle development <br> 4.774e-02","GO:0031345 (215) <br> negative regulation of cell projection organization <br> 4.778e-02","GO:0006325 (768) <br> chromatin organization <br> 4.864e-02","GO:1904951 (350) <br> positive regulation of establishment of protein localization <br> 4.940e-02"],"key":["GO:0048856","GO:0048731","GO:0032502","GO:0009653","GO:0048514","GO:0007275","GO:0001944","GO:0001568","GO:0035295","GO:0001525","GO:0072359","GO:0030334","GO:2000145","GO:0035239","GO:0040012","GO:0007155","GO:0048870","GO:0016477","GO:0040011","GO:0048646","GO:0050896","GO:0048513","GO:0016043","GO:2000147","GO:0030335","GO:0030154","GO:0048869","GO:0040017","GO:0048518","GO:0048522","GO:0071840","GO:0048519","GO:0065007","GO:0050789","GO:0051239","GO:0048468","GO:0065008","GO:0000902","GO:0048523","GO:0007399","GO:0032501","GO:0051128","GO:0050794","GO:0009888","GO:0051716","GO:0022603","GO:0030155","GO:0070887","GO:0008283","GO:0022008","GO:0007166","GO:0042127","GO:0120036","GO:0006950","GO:0050793","GO:0030030","GO:0048583","GO:0009628","GO:2000026","GO:0051240","GO:1901342","GO:0009893","GO:0048699","GO:0031175","GO:0010604","GO:0030182","GO:0007154","GO:0048666","GO:0045765","GO:0007165","GO:0009605","GO:0051179","GO:0045785","GO:0023052","GO:0001666","GO:0044087","GO:0009719","GO:0035556","GO:0141124","GO:0007167","GO:0048812","GO:0048584","GO:0001667","GO:0010033","GO:0120039","GO:0006119","GO:0009966","GO:0042773","GO:0048858","GO:0008219","GO:0012501","GO:0036293","GO:0070482","GO:0031325","GO:0071310","GO:0051094","GO:0031399","GO:0006810","GO:0010631","GO:0070848","GO:0051234","GO:0098609","GO:0006915","GO:0090132","GO:0022904","GO:0032989","GO:0023051","GO:0071363","GO:0071495","GO:0090130","GO:0051246","GO:0048871","GO:0009060","GO:0015988","GO:0015990","GO:0007409","GO:0042221","GO:0051173","GO:0009967","GO:0019646","GO:0010646","GO:0000165","GO:0003013","GO:0048667","GO:0015980","GO:0022900","GO:0060429","GO:0045597","GO:0009891","GO:0010557","GO:0061564","GO:0006996","GO:0044419","GO:0042592","GO:0042775","GO:0019222","GO:0043408","GO:0045333","GO:0031328","GO:0006796","GO:0006793","GO:0097435","GO:0051129","GO:0030855","GO:0043067","GO:0043542","GO:0002376","GO:0006091","GO:1902600","GO:0019220","GO:0051174","GO:0034330","GO:0007010","GO:1902531","GO:0030029","GO:1902533","GO:0051171","GO:1904018","GO:0034329","GO:0070372","GO:0007507","GO:0080090","GO:0031323","GO:0060255","GO:0001570","GO:0042981","GO:0009987","GO:0034097","GO:0008284","GO:0010628","GO:0070371","GO:0097485","GO:0007411","GO:1901700","GO:0002064","GO:0010632","GO:0051247","GO:0045766","GO:0006468","GO:0010035","GO:0033036","GO:0045595","GO:0007044","GO:0022607","GO:0016310","GO:0043069","GO:1901564","GO:0023056","GO:0010647","GO:0051641","GO:0098657","GO:0009607","GO:0031589","GO:0007159","GO:0090066","GO:0010810","GO:0007160","GO:0120035","GO:1902903","GO:0060284","GO:0051707","GO:0003007","GO:0043066","GO:0043207","GO:0150115","GO:0061061","GO:0022407","GO:0070727","GO:0031344","GO:0032879","GO:0150105","GO:0022409","GO:0036211","GO:0003158","GO:0001932","GO:0045937","GO:0010562","GO:0002040","GO:0001935","GO:0030036","GO:0042325","GO:0010720","GO:0010038","GO:0031401","GO:0006897","GO:0009790","GO:0001936","GO:1901888","GO:0008015","GO:0010811","GO:0050767","GO:0008104","GO:0050673","GO:0008285","GO:0016032","GO:0044085","GO:0042776","GO:0050678","GO:0038084","GO:0048585","GO:0045446","GO:0051050","GO:0007178","GO:0009611","GO:0043412","GO:0009617","GO:0051049","GO:0019538","GO:0071345","GO:0006898","GO:0071526","GO:0015986","GO:0040007","GO:0010594","GO:0010634","GO:0060341","GO:0002682","GO:0006811","GO:0043534","GO:0001952","GO:0046034","GO:0051649","GO:0001775","GO:0051130","GO:0050769","GO:0035924","GO:0009725","GO:0051493","GO:0006120","GO:0040013","GO:0008361","GO:0009889","GO:0023057","GO:0051960","GO:0010648","GO:0090109","GO:0051893","GO:0032535","GO:0045321","GO:0019058","GO:0003018","GO:0010556","GO:0043410","GO:0001954","GO:1903037","GO:1901698","GO:0009205","GO:0006754","GO:0150116","GO:0033043","GO:1901652","GO:0060485","GO:0030097","GO:0030099","GO:0030324","GO:0007169","GO:0031326","GO:0009144","GO:0009199","GO:0016192","GO:0030323","GO:0000122","GO:0007423","GO:0044057","GO:0009206","GO:0051172","GO:0009887","GO:0009145","GO:0007417","GO:0006812","GO:0010468","GO:0071702","GO:0060560","GO:0001569","GO:0141091","GO:0033993","GO:0090287","GO:0051241","GO:0009201","GO:0010595","GO:0001885","GO:0009142","GO:0007420","GO:0009968","GO:0009141","GO:0051962","GO:0060322","GO:0046689","GO:0016049","GO:1902396","GO:0046907","GO:0031324","GO:0071705","GO:0090100","GO:0015669","GO:0014070","GO:0044403","GO:0043010","GO:0060541","GO:0048589","GO:0016525","GO:0048041","GO:0045216","GO:0033554","GO:0010243","GO:0048729","GO:0032880","GO:0032101","GO:2000181","GO:1903829","GO:1901343","GO:0032970","GO:0030336","GO:0048010","GO:0010257","GO:0032981","GO:0007043","GO:1902904","GO:0048675","GO:0007015","GO:1990138","GO:0050920","GO:0009892","GO:0001934","GO:0048588","GO:0030218","GO:0048762","GO:0042110","GO:0042060","GO:1903039","GO:1901699","GO:2000146","GO:0050900","GO:0120032","GO:0008360","GO:0050770","GO:0110053","GO:0060491","GO:0003382","GO:0034504","GO:0043933","GO:0061437","GO:0061440","GO:0055008","GO:0001894","GO:0060249","GO:0006935","GO:1901566","GO:0051248","GO:0042330","GO:0010605","GO:0098662","GO:0015671","GO:0034101","GO:0009792","GO:0098655","GO:0042327","GO:0002684","GO:0045892","GO:0032388","GO:0044089","GO:1901701","GO:0032956","GO:0071674","GO:0009152","GO:1902679","GO:0032835","GO:0060041","GO:0046718","GO:0002262","GO:0001654","GO:0031400","GO:1902414","GO:0051592","GO:0046649","GO:0150063","GO:0140962","GO:0009260","GO:0042692","GO:0006952","GO:0014910","GO:0030100","GO:0061384","GO:0071560","GO:0044409","GO:0048841","GO:0048880","GO:0046390","GO:1900180","GO:0050870","GO:0055082","GO:0048878","GO:0098660","GO:0030900","GO:0071559","GO:0006164","GO:0051093","GO:0071496","GO:0033108","GO:0051492","GO:0010639","GO:0030509","GO:0048545","GO:0045934","GO:0032870","GO:0051494","GO:0051253","GO:0051017","GO:0050921","GO:0060841","GO:0070374","GO:0050680","GO:0072522","GO:0051894","GO:0032103","GO:0003151","GO:0019725","GO:0060415","GO:0031032","GO:0051701","GO:0061572","GO:0030516","GO:0110020","GO:0035886","GO:1903670","GO:0001817","GO:0071417","GO:0001816","GO:0031623","GO:0043434","GO:0051254","GO:0045229","GO:0030198","GO:0034220","GO:0050878","GO:1990845","GO:0014909","GO:0010975","GO:0033365","GO:0043062","GO:1902284","GO:0048846","GO:0007264","GO:0003206","GO:0031100","GO:0032386","GO:0045935","GO:0009165","GO:0031341","GO:0003073","GO:1901293","GO:0040008","GO:0000003","GO:0018193","GO:0071773","GO:0071772","GO:0060627","GO:0071407","GO:0009410","GO:0060326","GO:0048644","GO:0051222","GO:0048842","GO:0022604","GO:0001938","GO:0045773","GO:0150117","GO:0045944","GO:0034341","GO:0071383","GO:0009791","GO:0045893","GO:0090090","GO:0060395","GO:1902680","GO:2000628","GO:0043009","GO:0032231","GO:0043149","GO:0030038","GO:0003205","GO:0090407","GO:0001945","GO:0044271","GO:0048754","GO:0061387","GO:0042255","GO:0032496","GO:0050772","GO:0007517","GO:0061028","GO:0009743","GO:0055085","GO:0120161","GO:0001558","GO:0050863","GO:0104004","GO:0071214","GO:0072001","GO:0051145","GO:0034333","GO:0098542","GO:0006357","GO:0014812","GO:0022414","GO:0048844","GO:0051251","GO:0006972","GO:0072012","GO:0106106","GO:0018130","GO:0072006","GO:0008217","GO:0034654","GO:0007163","GO:0001886","GO:0009612","GO:0006366","GO:0045428","GO:0007266","GO:0120162","GO:0009749","GO:0045087","GO:0002237","GO:0051897","GO:0010977","GO:0060070","GO:0002720","GO:1990778","GO:1900182","GO:0007267","GO:0009746","GO:0007179","GO:0001659","GO:0010586","GO:0051152","GO:0001819","GO:0080164","GO:0031327","GO:1901137","GO:0046822","GO:0002683","GO:0061383","GO:0046824","GO:0045807","GO:0010558","GO:0061138","GO:0002694","GO:0038165","GO:0055010","GO:0034284","GO:0022411","GO:0072659","GO:0009890","GO:0003208","GO:0043129","GO:0060840","GO:0019438","GO:0001937","GO:1901653","GO:0010814","GO:1901362","GO:0006338","GO:0018108","GO:0002696","GO:0031667","GO:0018212","GO:0048872","GO:0090092","GO:0007162","GO:0002009","GO:0030178","GO:0003231","GO:0031345","GO:0006325","GO:1904951"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[21.328701751057359,20.864226972468124,21.425677032812782,20.410891339422463,18.668407940029581,21.053308382885966,18.908186071536839,18.856016803374629,19.403663741365257,18.464065056701649,19.439378205030948,19.178623821207275,19.253155187873404,19.10432153871146,19.304055367294772,19.665296933776261,19.884686769488304,19.723742560374109,19.495234200737425,19.444389654278609,21.88274512494765,20.600882389716016,21.539016614806243,18.566364821130115,18.500650704377186,21.029896765233428,21.030128864275088,18.602053031309978,21.382882501585598,21.294944028415323,21.567341510517984,21.274620612141288,22.147855770297312,22.117920782678972,20.567218733002786,20.580456506606325,20.555999759703631,19.253155187873404,21.20506702705476,20.309637264065447,21.734036327290926,20.317387988767649,22.073197465212008,20.105330786772971,21.688698070967941,19.072929603778331,18.919293936381997,20.566138165724581,20.130438904075231,19.979681740149239,20.41297214249937,19.910696974658698,19.819362960300751,20.813815051682944,20.384322580975084,19.847632302417448,20.828265580083379,19.5207587936241,19.73306396628913,19.913987772666282,17.678803404595058,20.757865019645447,19.808573485961332,19.350612301236584,20.657596136500928,19.747665325555783,21.534762819396946,19.508541196334715,17.652054395439823,21.432140862389922,20.521993586492155,21.157935465906252,18.323827527744129,21.519011385004998,17.836080781649432,19.210985186098107,19.99688315058361,20.346078537193332,19.907396343450667,19.16081347962028,18.747807006360492,20.228916694392645,18.162139397385626,20.660238472746975,18.77491552544776,16.591163144568295,20.508006265108179,15.718205436949759,18.787355432242641,20.106985976745786,20.106985976745786,17.958161847265735,18.165117717346547,20.659908540054147,20.258577339931808,19.677558310711198,19.318670025149817,20.937736855395432,17.696323875125149,18.856016803374629,21.010922982113723,19.113708638442272,20.064835818205584,17.709305060191621,16.17520418195295,18.956593895682929,20.718327126675387,18.806641510377418,19.792556895803209,17.734869765167502,19.975960473777356,19.083945550207464,17.067055779370321,9.700176674170466,9.700176674170466,18.262266882511316,21.197589923643317,20.48453941951616,19.780761776682866,15.394442462025141,20.718327126675387,18.851004453529015,18.45941212876998,18.648689983393385,17.851536166606468,16.47367201217347,19.529140283062222,19.16081347962028,20.438038906630339,20.383895523846846,18.433492686902198,20.687271427101319,19.999926719230256,19.934174664443724,15.679061062146381,21.430095782274744,18.693559386636259,17.446969261783856,20.426605382369303,20.262407796002723,20.272872290192961,18.928731471263692,18.839224835486206,18.882392547526074,19.848329883984494,17.258790060479736,20.434784581116546,18.289418567675717,16.506747981081492,19.248413576265786,19.249600821436079,18.941197134781184,19.666118552007926,19.858741250062707,18.890514391315275,19.259054715390775,21.129856269510995,16.908082832631191,18.23449212306971,17.747458444011631,18.660560141784572,21.166721650673466,21.355472807418838,21.344060494459796,15.882025628956763,19.813621623959669,22.598716152232853,19.410897151015615,19.313071133340067,19.506649771460314,17.843832550907738,17.370496430649634,17.370496430649634,20.080197546425548,17.312920622487489,17.301087667952569,19.349525622658916,16.891883825567096,19.459292312560617,18.656616635122333,20.540609330678421,19.885361398822894,15.864739656123454,20.859047200288121,19.669399006449162,19.281199497415457,21.233110611717237,20.029907540636515,20.029907540636515,20.682113566939638,19.089414393008106,19.923801732074157,17.85919220198322,18.075648223452593,18.487061994888009,17.202276580201648,17.215051415662369,18.839224835486206,17.975649234564241,19.123019968340142,19.896767575161896,17.493805663105405,19.246035386564525,19.899434365837209,15.981472643387029,18.808379179592698,18.308068239435634,20.343420943996556,18.862666714102716,20.126655136290552,13.529597157502854,17.743276315884209,20.397056181904627,16.539029588337922,18.944292818623428,18.712548647837508,18.712548647837508,16.451161238395375,16.807751972475245,18.740471361246467,19.034956694710743,18.402120044834064,17.999721415049571,18.823903449615361,18.751457762021811,19.466180311913369,16.591163144568295,17.289144360138966,18.389831878455684,16.484787908550889,18.22320150408035,20.338977032003182,18.284036783537221,18.751457762021811,17.836080781649432,20.930358247280399,15.267977608713885,18.033312175105056,14.653923996014182,19.859431874601995,16.269217457803322,19.167206025081374,18.016631675229984,18.416701738523138,20.481813323999734,19.158246482824286,19.877238566453983,21.011395652500902,19.286956478423942,17.39281863124878,14.910589323184952,15.442083462890102,19.251971624778417,16.875482427096053,16.699160992818289,19.300656010320601,19.752236819622194,19.462249447371921,16.332614376859173,16.332614376859173,17.252632081632278,20.046345943226477,19.381682892005362,19.471074765142809,17.669950360661655,15.241306099833178,19.208527977891112,18.369905656339025,13.827468853955285,17.836080781649432,17.149872545660465,21.225834640143709,19.655390189649136,18.466384753046551,19.657875105700629,15.129460953286518,15.129460953286518,18.013268198377471,19.236472936783183,17.488686282102407,17.295129931743453,21.186671045685987,18.259517895881778,15.319895742096389,17.947548986300461,19.504755172408313,17.39281863124878,16.044026258231838,15.267977608713885,19.400549543737782,18.480209239927877,17.610830123528174,19.251971624778417,18.226034028415874,17.277088702611685,18.549192406633704,21.217538933871005,17.436322104692952,17.436322104692952,19.707240356936772,17.295129931743453,19.221977953879332,18.654639942998166,18.581189986235994,16.202730409602214,20.157617355326259,19.309698562268999,16.216288914496698,19.206065496864341,19.225618918894778,21.161622484891499,20.258097580088869,17.483546122126494,14.43962568218266,17.839962721699276,19.376386084792355,17.704992955229532,19.437367402415497,16.282135104415108,16.17520418195295,15.465325129039176,16.369280411622906,18.790885633929687,19.530994406739804,17.524094483509998,17.904160093122737,18.884021439413772,12.407605671852071,18.407000319038037,10.702412765043418,19.554824474688868,20.546498530296468,20.011416026277963,16.044026258231838,13.529597157502854,19.357110680949631,17.573015855014475,17.929655297260286,17.441656841322331,18.895360528242147,16.147114504766865,15.659075333497245,17.246444436276072,19.910696974658698,19.382739298201539,18.656616635122333,19.169753128871644,19.345168508722086,16.17520418195295,18.372414567657582,16.189036371100354,17.992891601414531,17.643009717726557,14.876507174207195,15.241306099833178,15.241306099833178,16.79903557952581,16.833560798433115,16.47367201217347,18.191599502487591,17.052723906189033,17.289144360138966,20.627067755849026,18.477916229644741,17.498904426206344,16.680165266632077,17.307017816653882,18.574858964839876,17.936843956031275,17.529072820646299,18.955063351489322,17.709305060191621,17.940422635738837,17.129659550094892,16.539029588337922,16.850490053745229,17.514078591395506,17.149872545660465,14.613409556027271,17.751626595051512,20.674329442977385,14.078535733677407,14.078535733677407,15.394442462025141,17.483546122126494,17.483546122126494,18.188685401228863,19.848329883984494,18.723797222092436,18.194506632993676,20.546865529502192,18.877492204588115,12.407605671852071,16.781428259236037,18.931860412320365,18.914546369101146,18.544860315044854,19.322016216221716,19.618297106189353,16.670564656334751,18.407000319038037,19.691293962001481,17.81253066459243,16.970947385124298,17.116006618603713,19.625133218529758,15.554697906781431,16.963252535010657,16.344946758710758,17.122851084106237,18.116568226369793,17.784479219279955,16.39318500134382,16.727154664841223,18.992735342150937,18.125819736421583,15.737378400651465,17.215051415662369,18.156160809921754,19.919232737763505,15.997377539550557,17.743276315884209,14.190524430536781,17.47320281161134,16.39318500134382,14.135529956784948,18.141084679903194,17.252632081632278,16.76358319901939,17.403834749391049,18.867629457595761,19.306315994545855,19.040661510155601,18.165117717346547,17.498904426206344,17.312920622487489,19.291541269852484,18.029994167237007,16.044026258231838,15.949106100033569,17.915151124757344,16.772535830865632,17.915151124757344,19.808573485961332,18.714430875962282,16.833560798433115,19.715128982032727,16.79903557952581,16.736356135602332,13.165094704134018,17.240226848372863,16.651153144759785,17.359187054355758,13.827468853955285,18.662526977488223,15.618238820841244,19.023462108095462,15.679061062146381,17.102208213102088,16.939892475914366,16.82501387237734,16.103873952107552,16.147114504766865,14.528988762818752,14.653923996014182,18.956593895682929,18.832440049345617,18.965735120751209,16.580897563777473,18.284036783537221,19.960948340102021,17.643009717726557,17.643009717726557,19.167206025081374,17.851536166606468,16.717888898684524,16.161231149937279,18.457078863208451,19.19367322615927,17.652054395439823,14.392913471576092,14.392913471576092,18.135001614954604,16.539029588337922,15.737378400651465,17.592064992028025,20.08358219071777,17.519096457241538,16.506747981081492,16.294931486381042,17.53403161741592,18.670361780713947,19.949553436874549,18.654639942998166,16.891883825567096,16.891883825567096,18.522960374722658,18.697381754131872,18.321215232823697,17.717884894111318,15.829513939414692,17.624722444228865,10.702412765043418,17.483546122126494,16.044026258231838,14.84162824092461,14.135529956784948,19.534693531361903,16.727154664841223,17.240226848372863,16.269217457803322,19.861501168226596,16.451161238395375,16.074274264307061,19.86356660275386,15.864739656123454,18.904992873760889,16.189036371100354,16.202730409602214,16.202730409602214,16.947725605938636,18.457078863208451,13.957963134999822,21.015404724070795,16.842052633020028,16.294931486381042,15.418459955353805,18.176958638735037,15.915969043131074,17.874363774784328,14.943908607777407,17.558537588035755,19.713555709857136,16.528355056331499,18.165117717346547,17.954634373506714,17.958161847265735,17.958161847265735,17.866801220242017,15.576194582978985,12.241715190548584,19.502857388836173,20.289314809553581,16.357168137614234,19.943811501738267,15.597371909790384,17.7722608047339,14.484959030043042,13.957963134999822,16.549619852843737,20.821490025374917,16.807751972475245,17.156540581918918,20.800031060897677,17.233979038073436,12.407605671852071,17.430964870798807,20.328686097650635,15.465325129039176,16.560127115942691,15.882025628956763,17.359187054355758,19.169753128871644,18.22320150408035,17.081231235176517,16.745494161527361,17.610830123528174,16.103873952107552,17.968684056943637,16.028650542372688,19.867000455694551,17.387274630647035,17.081231235176517,16.963252535010657,16.059232993254202,12.698730272061621,18.372414567657582,15.554697906781431,20.413803337967128,18.650676627869341,16.47367201217347,18.414283726723369,14.910589323184952,15.554697906781431,16.993763602439444,20.389434103397544,17.102208213102088,18.502900532528347,8.929031646366262,14.943908607777407,17.430964870798807,18.265009639856906,17.652054395439823,20.423316194151074,15.576194582978985,13.894051557160756,16.161231149937279,20.824318473911749,14.613409556027271,18.009895483189442,7.7180145958336093,20.868027374231957,18.602053031309978,17.307017816653882,17.896778047758332,18.498396628871042,17.318796326902053,17.951096771924238,17.430964870798807,17.629319291960101,18.404562661748759,16.727154664841223,16.539029588337922,17.156540581918918,18.867629457595761,17.836080781649432],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[39.646892040571721,34.29137022137116,38.760166335985517,33.691009191920443,34.811348184117179,33.407156487619474,34.439576500607942,40.711967674408982,40.819982420293421,33.733712696107311,40.825006361962465,33.633233862726435,33.427252254295645,40.83003030363151,37.996527202290878,38.903348673553261,40.639120520207847,35.715657684545043,35.328814176028679,40.814958478624376,34.273786425529501,33.786464083632268,35.409197242733377,37.265543689445018,40.511010007647236,34.813860154951698,41.27967308301092,34.469720150622209,33.763856346121571,37.981455377283744,40.892829574494556,35.996998418011486,41.025964028724211,41.023452057889692,35.331326146863198,33.585506416870523,34.808836213282653,33.419716341792082,37.999039173125389,38.235164431570446,41.083739357918212,42.234222000129222,34.698309496563695,33.73622466694183,34.28383430886759,37.403702085343717,37.104777556035621,40.817470449458902,37.368534493660412,33.474979700151557,41.269625199672831,34.356681463068725,34.406920879759163,34.80130030077909,35.680490092861739,33.409668458453993,38.00657508562896,33.128327724987543,33.412180429288512,38.717462831798642,38.127149685686007,34.926898842505182,38.704902977626027,37.765425885514865,34.803812271613616,41.900129879137815,33.723664812769222,38.084446181499139,35.178095925957365,33.429764225130171,35.683002063696257,35.70058585953791,38.559208669223764,38.654663560935596,37.358486610322323,35.622714763667737,33.593042329374086,33.213734733361285,33.02031297910311,37.268055660279536,33.2262945875339,37.363510551991368,37.353462668653279,34.46720817978769,35.255967021827544,34.931922784174226,40.895341545329075,35.03742555922414,35.615178851164167,33.437300137633734,37.355974639487805,37.348438726984241,38.099518006506266,37.961359610607566,41.764483454073634,34.409432850593689,40.709455703574456,38.260284139915669,37.11482543937371,35.012305850878924,42.384940250200529,35.090176946749104,37.41123799784728,34.359193433903243,36.951547335129789,34.806324242448134,41.081227387083693,34.972114317526575,33.53024305851104,40.262324895029579,33.017801008268584,39.531341382183719,41.761971483239108,37.190184564409364,38.094494064837221,33.434788166799216,34.472232121456727,40.812446507789858,35.894007613796092,38.757654365150991,35.544843667797558,35.027377675886058,35.092688917583622,32.764091953981875,33.464931816813468,37.333366901977108,35.213263517640669,34.70333343823274,34.959554463353967,33.457395904309912,32.754044070643793,41.086251328752738,34.477256063125772,37.360998581156849,33.728688754438267,34.939458696677796,33.191126995850595,35.029889646720576,33.467443787647994,40.882781691156467,37.258007776941454,35.547355638632077,34.301418104709242,37.780497710521992,40.905389428667164,33.726176783603741,37.35095069781876,33.485027583489646,34.381801171413947,42.751687992040715,33.487539554324172,33.761344375287052,34.276298396364027,34.705845409067258,37.845808952219564,37.273079601948581,38.770214219323599,36.572239739116988,33.550338825187211,40.664240228553069,33.555362766856256,37.177624710236756,33.53777897101461],"y":[17,17,17,17,17,17,14.351176438342716,13.470939565024963,12.925954228479808,12.300095794918919,11.91667959883414,11.850313536857096,11.679991115860158,11.679991115860158,11.402832297756904,11.337809963389502,11.008938527585094,10.73362386970617,10.459578373628862,9.9676845068857993,9.4980763643411947,9.4566136894153772,9.3866583599442706,9.364978805459188,9.3521871636750529,9.2809145019624051,9.0522824376342808,8.7504182256228962,8.6483302195826059,8.6065261017714665,8.5832023796803423,8.0762908141991403,7.7416176903680238,7.5147611580269942,7.4842973502396868,7.384408530015576,7.176337747362882,6.9197747780501864,6.7626833745570547,6.7620003867418026,6.6071173675133661,6.5881228284148357,6.5550440809486625,6.5472473318825477,6.3776656950078081,6.2997996538847723,6.2907890528444739,6.207875593761222,6.0448021200027595,5.96073615212741,5.9567898801946875,5.9355294563972203,5.8398222050854391,5.6838023902744998,5.4632558684713128,5.4511001319042665,5.3503769254547899,5.283947583936631,5.2239655318002889,5.2004371887695804,5.1788463099172759,5.0298518203295677,4.858828026733927,4.5598893515670111,4.5367791117817404,4.4719632505128351,4.4545011622411579,4.4445384519180005,4.4445384519180005,4.4445384519180005,4.3444201579986732,4.3444201579986732,4.3265337091394214,4.2968800074860027,4.2968800074860027,4.2961398267491289,4.2961398267491289,4.2767786627304893,4.1805355107347966,4.1482260242964557,4.0980830123075735,3.9942346917796248,3.9907537649774718,3.9840053698993496,3.8936763181159217,3.71563637499731,3.7062982493025749,3.6346162529279571,3.577343060330989,3.569787510901457,3.5525126952259352,3.4800306129936085,3.416750812530482,3.3689364952899217,3.3549094705157505,3.2754922415878065,3.2724279798756077,3.2426602863107963,3.2426602863107963,3.2349443482606715,3.2036366578215798,3.1713745336480357,3.1434573286277647,3.1141639340911094,3.0714753227480043,3.064526900227448,3.0136802121796591,3.0102266398081476,3.0018563665613884,2.9870102324086658,2.9870102324086658,2.8680444426881144,2.8300553199107288,2.6944086358496948,2.6934022620566282,2.6934022620566282,2.6889708324668673,2.6667966247888604,2.6636801612492156,2.6333804879387834,2.6302453399711463,2.5867661836480926,2.5807845005550756,2.562388976792783,2.562388976792783,2.5535168403677115,2.492249012729097,2.4804358862393587,2.4687134575394993,2.3568580679442714,2.3568580679442714,2.3448831827860936,2.3088791915414726,2.2938506074041412,2.2833909023719596,2.2499875224897621,2.2412480652330329,2.1381396194928661,2.1325272330247658,2.1325272330247658,2.1006478861899645,2.1001262220329977,2.0960885843528327,2.0812065136835192,2.0785998874679388,2.0400793625706592,2.0397815818504044,1.9997424478293924,1.9972854551768766,1.9838909304708658,1.9258383455225896,1.9156612636050043,1.7101808109565857,1.6555207159715346,1.6519431571613967,1.6411377726759542,1.6098722678020654,1.6098722678020654,1.6034888933859879,1.522365355578442,1.4281145622886495,1.402085307781735,1.3141565352871993],"text":["GO:0071944 (6862) <br> cell periphery <br> 1.698e-23","GO:0009986 (1037) <br> cell surface <br> 3.229e-20","GO:0070161 (729) <br> anchoring junction <br> 9.376e-19","GO:0005886 (6367) <br> plasma membrane <br> 2.116e-18","GO:0030054 (2260) <br> cell junction <br> 7.970e-18","GO:0005737 (11561) <br> cytoplasm <br> 2.506e-17","GO:0016020 (10191) <br> membrane <br> 4.455e-15","GO:0098590 (1408) <br> plasma membrane region <br> 3.381e-14","GO:0098798 (298) <br> mitochondrial protein-containing complex <br> 1.186e-13","GO:0005911 (539) <br> cell-cell junction <br> 5.011e-13","GO:0098800 (150) <br> inner mitochondrial membrane protein complex <br> 1.211e-12","GO:0005856 (2373) <br> cytoskeleton <br> 1.412e-12","GO:0005746 (90) <br> mitochondrial respirasome <br> 2.089e-12","GO:0098803 (90) <br> respiratory chain complex <br> 2.089e-12","GO:0045177 (562) <br> apical part of cell <br> 3.955e-12","GO:0070469 (99) <br> respirasome <br> 4.594e-12","GO:0098552 (803) <br> side of membrane <br> 9.796e-12","GO:0031982 (2228) <br> vesicle <br> 1.847e-11","GO:0031252 (453) <br> cell leading edge <br> 3.471e-11","GO:0098796 (1242) <br> membrane protein complex <br> 1.077e-10","GO:0009897 (487) <br> external side of plasma membrane <br> 3.176e-10","GO:0005938 (339) <br> cell cortex <br> 3.495e-10","GO:0031410 (2077) <br> cytoplasmic vesicle <br> 4.105e-10","GO:0042995 (2564) <br> cell projection <br> 4.315e-10","GO:0097708 (2081) <br> intracellular vesicle <br> 4.444e-10","GO:0030055 (199) <br> cell-substrate junction <br> 5.237e-10","GO:0120025 (2460) <br> plasma membrane bounded cell projection <br> 8.866e-10","GO:0016324 (461) <br> apical plasma membrane <br> 1.777e-09","GO:0005925 (188) <br> focal adhesion <br> 2.247e-09","GO:0045121 (324) <br> membrane raft <br> 2.474e-09","GO:0098857 (325) <br> membrane microdomain <br> 2.611e-09","GO:0032991 (7427) <br> protein-containing complex <br> 8.389e-09","GO:0099081 (1066) <br> supramolecular polymer <br> 1.813e-08","GO:0099080 (1424) <br> supramolecular complex <br> 3.057e-08","GO:0031253 (371) <br> cell projection membrane <br> 3.279e-08","GO:0005829 (4105) <br> cytosol <br> 4.127e-08","GO:0030027 (210) <br> lamellipodium <br> 6.663e-08","GO:0005743 (539) <br> mitochondrial inner membrane <br> 1.203e-07","GO:0045178 (331) <br> basal part of cell <br> 1.727e-07","GO:0048471 (819) <br> perinuclear region of cytoplasm <br> 1.730e-07","GO:0099512 (1059) <br> supramolecular fiber <br> 2.471e-07","GO:1990204 (120) <br> oxidoreductase complex <br> 2.582e-07","GO:0019866 (593) <br> organelle inner membrane <br> 2.786e-07","GO:0005912 (165) <br> adherens junction <br> 2.836e-07","GO:0009925 (312) <br> basal plasma membrane <br> 4.191e-07","GO:0043292 (252) <br> contractile fiber <br> 5.014e-07","GO:0042383 (175) <br> sarcolemma <br> 5.119e-07","GO:0098797 (587) <br> plasma membrane protein complex <br> 6.196e-07","GO:0043235 (392) <br> receptor complex <br> 9.020e-07","GO:0005768 (973) <br> endosome <br> 1.095e-06","GO:0110165 (22653) <br> cellular anatomical entity <br> 1.105e-06","GO:0012505 (4287) <br> endomembrane system <br> 1.160e-06","GO:0015629 (503) <br> actin cytoskeleton <br> 1.446e-06","GO:0030016 (242) <br> myofibril <br> 2.071e-06","GO:0031966 (791) <br> mitochondrial membrane <br> 3.441e-06","GO:0005739 (1895) <br> mitochondrion <br> 3.539e-06","GO:0045202 (1641) <br> synapse <br> 4.463e-06","GO:0005576 (2554) <br> extracellular region <br> 5.201e-06","GO:0005740 (843) <br> mitochondrial envelope <br> 5.971e-06","GO:0070069 (40) <br> cytochrome complex <br> 6.303e-06","GO:0045335 (113) <br> phagocytic vesicle <br> 6.625e-06","GO:0030139 (209) <br> endocytic vesicle <br> 9.336e-06","GO:0070062 (102) <br> extracellular exosome <br> 1.384e-05","GO:0044291 (84) <br> cell-cell contact zone <br> 2.755e-05","GO:0030017 (219) <br> sarcomere <br> 2.906e-05","GO:1903561 (116) <br> extracellular vesicle <br> 3.373e-05","GO:0005901 (103) <br> caveola <br> 3.512e-05","GO:0045271 (50) <br> respiratory chain complex I <br> 3.593e-05","GO:0030964 (50) <br> NADH dehydrogenase complex <br> 3.593e-05","GO:0005747 (50) <br> mitochondrial respiratory chain complex I <br> 3.593e-05","GO:0031967 (1305) <br> organelle envelope <br> 4.525e-05","GO:0031975 (1305) <br> envelope <br> 4.525e-05","GO:0062023 (371) <br> collagen-containing extracellular matrix <br> 4.715e-05","GO:0065010 (123) <br> extracellular membrane-bounded organelle <br> 5.048e-05","GO:0043230 (123) <br> extracellular organelle <br> 5.048e-05","GO:0031838 (10) <br> haptoglobin-hemoglobin complex <br> 5.057e-05","GO:0005833 (10) <br> hemoglobin complex <br> 5.057e-05","GO:0005615 (1836) <br> extracellular space <br> 5.287e-05","GO:0001726 (182) <br> ruffle <br> 6.599e-05","GO:0043005 (1525) <br> neuron projection <br> 7.108e-05","GO:0005622 (17412) <br> intracellular anatomical structure <br> 7.978e-05","GO:0043232 (6555) <br> intracellular non-membrane-bounded organelle <br> 1.013e-04","GO:0043228 (6556) <br> non-membrane-bounded organelle <br> 1.022e-04","GO:0016323 (275) <br> basolateral plasma membrane <br> 1.038e-04","GO:0031090 (3365) <br> organelle membrane <br> 1.277e-04","GO:0030141 (494) <br> secretory granule <br> 1.925e-04","GO:0098858 (251) <br> actin-based cell projection <br> 1.967e-04","GO:0030670 (50) <br> phagocytic vesicle membrane <br> 2.319e-04","GO:0031674 (155) <br> I band <br> 2.646e-04","GO:0005751 (25) <br> mitochondrial respiratory chain complex IV <br> 2.693e-04","GO:0043229 (15836) <br> intracellular organelle <br> 2.802e-04","GO:0043226 (16169) <br> organelle <br> 3.311e-04","GO:0045277 (28) <br> respiratory chain complex IV <br> 3.830e-04","GO:0044853 (138) <br> plasma membrane raft <br> 4.276e-04","GO:1902495 (420) <br> transmembrane transporter complex <br> 4.417e-04","GO:0015630 (1380) <br> microtubule cytoskeleton <br> 5.303e-04","GO:0098588 (1767) <br> bounding membrane of organelle <br> 5.340e-04","GO:0048770 (103) <br> pigment granule <br> 5.719e-04","GO:0042470 (103) <br> melanosome <br> 5.719e-04","GO:0030659 (914) <br> cytoplasmic vesicle membrane <br> 5.822e-04","GO:1990351 (446) <br> transporter complex <br> 6.257e-04","GO:0030863 (113) <br> cortical cytoskeleton <br> 6.739e-04","GO:0043296 (166) <br> apical junction complex <br> 7.187e-04","GO:0012506 (928) <br> vesicle membrane <br> 7.688e-04","GO:0036477 (1103) <br> somatodendritic compartment <br> 8.483e-04","GO:0030018 (141) <br> Z disc <br> 8.619e-04","GO:0099503 (714) <br> secretory vesicle <br> 9.690e-04","GO:0030424 (801) <br> axon <br> 9.767e-04","GO:0005794 (1473) <br> Golgi apparatus <br> 9.957e-04","GO:0097517 (96) <br> contractile actin filament bundle <br> 1.030e-03","GO:0001725 (96) <br> stress fiber <br> 1.030e-03","GO:0071664 (4) <br> catenin-TCF7L2 complex <br> 1.355e-03","GO:1902494 (1783) <br> catalytic complex <br> 1.479e-03","GO:0042641 (105) <br> actomyosin <br> 2.021e-03","GO:0045275 (12) <br> respiratory chain complex III <br> 2.026e-03","GO:0005750 (12) <br> mitochondrial respiratory chain complex III <br> 2.026e-03","GO:0016327 (27) <br> apicolateral plasma membrane <br> 2.047e-03","GO:0098794 (847) <br> postsynapse <br> 2.154e-03","GO:0032432 (106) <br> actin filament bundle <br> 2.169e-03","GO:0070160 (149) <br> tight junction <br> 2.326e-03","GO:0031527 (19) <br> filopodium membrane <br> 2.343e-03","GO:0030666 (71) <br> endocytic vesicle membrane <br> 2.590e-03","GO:0030864 (82) <br> cortical actin cytoskeleton <br> 2.626e-03","GO:0000323 (535) <br> lytic vacuole <br> 2.739e-03","GO:0005764 (535) <br> lysosome <br> 2.739e-03","GO:0043209 (202) <br> myelin sheath <br> 2.796e-03","GO:0031012 (499) <br> extracellular matrix <br> 3.219e-03","GO:0019897 (140) <br> extrinsic component of plasma membrane <br> 3.308e-03","GO:0030312 (501) <br> external encapsulating structure <br> 3.398e-03","GO:0005761 (90) <br> mitochondrial ribosome <br> 4.397e-03","GO:0000313 (90) <br> organellar ribosome <br> 4.397e-03","GO:0099513 (809) <br> polymeric cytoskeletal fiber <br> 4.520e-03","GO:0016342 (32) <br> catenin complex <br> 4.910e-03","GO:0043231 (14435) <br> intracellular membrane-bounded organelle <br> 5.083e-03","GO:0005903 (144) <br> brush border <br> 5.207e-03","GO:0030175 (114) <br> filopodium <br> 5.624e-03","GO:0005604 (109) <br> basement membrane <br> 5.738e-03","GO:0030667 (104) <br> secretory granule membrane <br> 7.275e-03","GO:0005765 (298) <br> lysosomal membrane <br> 7.370e-03","GO:0098852 (298) <br> lytic vacuole membrane <br> 7.370e-03","GO:0042824 (17) <br> MHC class I peptide loading complex <br> 7.931e-03","GO:0031528 (32) <br> microvillus membrane <br> 7.941e-03","GO:0010008 (477) <br> endosome membrane <br> 8.015e-03","GO:0044297 (829) <br> cell body <br> 8.295e-03","GO:0098862 (205) <br> cluster of actin-based cell projections <br> 8.344e-03","GO:0005902 (112) <br> microvillus <br> 9.118e-03","GO:0043227 (14934) <br> membrane-bounded organelle <br> 9.125e-03","GO:0005773 (640) <br> vacuole <br> 1.001e-02","GO:0014704 (61) <br> intercalated disc <br> 1.006e-02","GO:1990836 (2) <br> lysosomal matrix <br> 1.038e-02","GO:0005774 (358) <br> vacuolar membrane <br> 1.186e-02","GO:0005923 (141) <br> bicellular tight junction <br> 1.214e-02","GO:0009898 (175) <br> cytoplasmic side of plasma membrane <br> 1.949e-02","GO:0019898 (221) <br> extrinsic component of membrane <br> 2.210e-02","GO:0044391 (202) <br> ribosomal subunit <br> 2.229e-02","GO:0043025 (743) <br> neuronal cell body <br> 2.285e-02","GO:0070210 (2) <br> Rpd3L-Expanded complex <br> 2.455e-02","GO:0034967 (2) <br> Set3 complex <br> 2.455e-02","GO:0005813 (708) <br> centrosome <br> 2.492e-02","GO:0098562 (212) <br> cytoplasmic side of membrane <br> 3.004e-02","GO:0005815 (848) <br> microtubule organizing center <br> 3.732e-02","GO:0042612 (25) <br> MHC class I protein complex <br> 3.962e-02","GO:0005797 (26) <br> Golgi medial cisterna <br> 4.851e-02"],"key":["GO:0071944","GO:0009986","GO:0070161","GO:0005886","GO:0030054","GO:0005737","GO:0016020","GO:0098590","GO:0098798","GO:0005911","GO:0098800","GO:0005856","GO:0005746","GO:0098803","GO:0045177","GO:0070469","GO:0098552","GO:0031982","GO:0031252","GO:0098796","GO:0009897","GO:0005938","GO:0031410","GO:0042995","GO:0097708","GO:0030055","GO:0120025","GO:0016324","GO:0005925","GO:0045121","GO:0098857","GO:0032991","GO:0099081","GO:0099080","GO:0031253","GO:0005829","GO:0030027","GO:0005743","GO:0045178","GO:0048471","GO:0099512","GO:1990204","GO:0019866","GO:0005912","GO:0009925","GO:0043292","GO:0042383","GO:0098797","GO:0043235","GO:0005768","GO:0110165","GO:0012505","GO:0015629","GO:0030016","GO:0031966","GO:0005739","GO:0045202","GO:0005576","GO:0005740","GO:0070069","GO:0045335","GO:0030139","GO:0070062","GO:0044291","GO:0030017","GO:1903561","GO:0005901","GO:0045271","GO:0030964","GO:0005747","GO:0031967","GO:0031975","GO:0062023","GO:0065010","GO:0043230","GO:0031838","GO:0005833","GO:0005615","GO:0001726","GO:0043005","GO:0005622","GO:0043232","GO:0043228","GO:0016323","GO:0031090","GO:0030141","GO:0098858","GO:0030670","GO:0031674","GO:0005751","GO:0043229","GO:0043226","GO:0045277","GO:0044853","GO:1902495","GO:0015630","GO:0098588","GO:0048770","GO:0042470","GO:0030659","GO:1990351","GO:0030863","GO:0043296","GO:0012506","GO:0036477","GO:0030018","GO:0099503","GO:0030424","GO:0005794","GO:0097517","GO:0001725","GO:0071664","GO:1902494","GO:0042641","GO:0045275","GO:0005750","GO:0016327","GO:0098794","GO:0032432","GO:0070160","GO:0031527","GO:0030666","GO:0030864","GO:0000323","GO:0005764","GO:0043209","GO:0031012","GO:0019897","GO:0030312","GO:0005761","GO:0000313","GO:0099513","GO:0016342","GO:0043231","GO:0005903","GO:0030175","GO:0005604","GO:0030667","GO:0005765","GO:0098852","GO:0042824","GO:0031528","GO:0010008","GO:0044297","GO:0098862","GO:0005902","GO:0043227","GO:0005773","GO:0014704","GO:1990836","GO:0005774","GO:0005923","GO:0009898","GO:0019898","GO:0044391","GO:0043025","GO:0070210","GO:0034967","GO:0005813","GO:0098562","GO:0005815","GO:0042612","GO:0005797"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[21.426941484964697,19.24365224781301,18.801413011933672,21.345600897953894,20.178985844189427,21.983573371445424,21.85055129779693,19.61743965611889,17.615477980455847,18.411860832280251,16.63145461546215,20.235796075048462,15.847237501615188,15.847237501615188,18.466384753046551,15.997377539550557,18.924022253618787,20.162344026767379,18.182836156914082,19.465198875229575,18.278630932934441,17.792558692608914,20.080197546425548,20.325534515348938,20.082455138466898,17.045498270277001,20.277595677333228,18.206066086407073,16.963252535010657,17.730645158790988,17.734869765167502,21.512496588674903,19.277731337954428,19.631080831456792,17.915151124757344,20.860958508983764,17.122851084106237,18.411860832280251,17.759921323909662,18.948920968131823,19.269597816105485,16.294931486381042,18.536148574779297,16.772535830865632,17.678803404595058,17.381706400475078,16.85887372598534,18.522960374722658,17.98946241223161,19.164653268106157,22.677165354330711,20.909469356662484,18.321215232823697,17.324645168209628,18.904992873760889,19.972226584386775,19.80132182586361,20.321016250100502,18.985297989242802,14.484959030043042,16.202730409602214,17.116006618603713,16.044026258231838,15.737378400651465,17.182873653029556,16.243009432060077,16.059232993254202,14.876507174207195,14.876507174207195,14.876507174207195,19.525422864372569,19.525422864372569,17.915151124757344,16.332614376859173,16.332614376859173,11.626277095618034,11.626277095618034,19.934819746309252,16.916107927490465,19.713555709857136,22.40891768631861,21.377259973865275,21.377425783888874,17.503982730637716,20.636892094769589,18.297446569158936,17.376113736180645,14.876507174207195,16.680165266632077,13.609357884847448,22.311247290549563,22.332714729193487,13.827468853955285,16.506747981081492,18.082033459750171,19.593161243494308,19.889400556220906,16.059232993254202,16.059232993254202,19.086683207552351,18.162139397385626,16.202730409602214,16.781428259236037,19.105667237213666,19.319786513354209,16.539029588337922,18.77491552544776,18.920872173555651,19.671853051762451,15.949106100033569,15.949106100033569,8.929031646366262,19.900100059368054,16.089153481435879,12.058806804819245,12.058806804819245,13.75799563321786,18.991251676364083,16.103873952107552,16.621495154335239,13.059967644679528,15.465325129039176,15.6987686639432,18.402120044834064,18.402120044834064,17.067055779370321,18.310709120711632,16.528355056331499,18.315973611085738,15.847237501615188,15.847237501615188,18.93342173803374,14.078535733677407,22.215387588577507,16.570552615864109,16.216288914496698,16.147114504766865,16.074274264307061,17.615477980455847,17.615477980455847,12.82781267352663,14.078535733677407,18.251233285710949,18.96421656268549,17.088261325299367,16.189036371100354,22.250614973551549,18.634689367697778,15.214137151492038,3.7795275590551185,17.866801220242017,16.539029588337922,16.85887372598534,17.19584133626687,17.067055779370321,18.825615750476654,3.7795275590551185,3.7795275590551185,18.76414696016883,17.136432378704722,18.992735342150937,13.609357884847448,13.685385010878013],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[5.7447599732664205,8.4296348300214863,17.184989133714808,7.347145528560838,15.547441512943713,5.7146210600006953,11.360644145113312,16.366215323329261,8.3316833619078778,5.204771110588835,11.315435775214723,21.884148027062544,11.365667297324265,11.363155721218789,5.7422483971609442,11.373202025640698,5.94819763781007,11.456084037121443,20.613290517691109,3.1754176173633031,3.1779291934687803,24.855342559842004,17.642095984911649,16.371238475540213,5.129423827424521,3.5395961526574888,5.8753619307512324,9.2634780970398971,7.7665787381755207,9.2308276076686955,12.011142356431893,23.049519340003936,28.936653731242362,9.0625520086017275,4.5140876815826205,12.513457577527323,11.368178873429745,6.0386143776072467,2.9041673979717717,8.8992995617457122,11.521385015863849,11.370690449535221,11.159718056675143,15.436932164302718,29.114975634731238,17.275405873511986,17.446193048684432,9.2232928793522646,5.3981624707105755,25.061291800491126,5.581507526410407,2.5801740803652198,5.7648525821102385,5.7472715493718987,8.9193921705895285,8.9168805944840521,16.170312387102044,28.552382587104361,13.181536821584242,16.17282396320752,13.176513669373286,9.8687679384598894,16.368726899434737,13.191583126006151,15.966874722558394,4.9360324673027804,5.3655119813393721,11.390783058379039,11.302877894687338,11.90816773610733,5.8452230174855071,4.5165992576880978,4.9385440434082577,11.651986973348663,2.8539358758622289,11.682125886614386,5.3881161662886665,5.9833597032867498,2.075347283164314,29.396272158544679,5.2022595344833578,4.8832893690877608,2.9217484307101116,17.041829295702609,29.006977862195722,29.403806886861108,6.116473236877038,11.51385028754742,21.894194331484453,5.5488570370392036,12.518480729738275,29.022047318828584,21.914286940328267,28.140484105806106,11.571616537973394,2.8564474519677057,5.2499794804874238,15.828738036757152,5.1495164362683372,6.1591700306701496,28.519732097733154,21.88665960316802,13.327208235701917,5.7824336148485775,29.838309553108655,5.8201072564307346,24.086800271565995,24.084288695460518,8.211127708844975,7.7088124877495456,12.405459804991803],"y":[17,15.105399319934961,15.079707488557217,13.235550687741769,11.788121803101653,11.292521801055653,10.619260216398548,10.550864924765035,9.7475733115454499,8.7847723859771207,7.8455880950507684,7.4220208204149412,7.0572431117918972,7.0254078515295104,6.8995889450759025,6.1650480728781858,6.1329493040077985,6.0694587243475819,6.0305505006425593,5.7555105666238742,5.7555105666238742,5.713519613176981,5.4319398689915479,5.2969678906149547,5.2942339677129429,5.2356891270887287,5.2246511676796912,5.0317771564582019,4.903776235852173,4.7557972704287659,4.6389909395156375,4.6123088719939611,4.5155293196617103,4.4764889699848638,4.3613303090533062,4.3363489613658732,4.2516632166843396,4.1202896697149107,4.0994338390854663,3.9738086741666017,3.9525809062785391,3.9382306568865091,3.891902522249215,3.8641100255867458,3.8356686797470121,3.6683599559154212,3.6565415736749394,3.6451379537815392,3.6238559282243026,3.6153907383146184,3.5897426318822201,3.5286349006821722,3.4709007379757635,3.4526300060612316,3.3931113455983759,3.3931113455983759,3.3709948356757153,3.3432175499618983,3.3407968589839032,3.3271908628702929,3.222078332580689,2.9883004183680884,2.9770546766868766,2.9768990197172704,2.9688425024133576,2.8674806221405276,2.7449823222862118,2.643708801026146,2.5963093874494234,2.5950428825288658,2.537624897711964,2.459306042141975,2.3908448645566369,2.3296239327556472,2.3139259284598515,2.2779454908169248,2.2101152169413933,2.1535011058046383,2.1108964910030736,2.1108964910030736,2.1041112920632963,2.0903003126405131,2.0680756908451419,2.0609663661875759,2.0445129362294394,2.0428760789807674,1.9408542058964287,1.9362418540652664,1.9108152214507064,1.8906532710542987,1.8805290773410774,1.8266364063986402,1.763028389678436,1.7384363908836993,1.7384146658314676,1.734017987910875,1.6933310312044545,1.6572319759871479,1.6051451417526812,1.5903068642475087,1.5804644035144229,1.5289427216100169,1.5269527808329419,1.4905905682908203,1.4815495209415053,1.3530429653703886,1.3522515170092642,1.3522515170092642,1.3273891805465963,1.3127692595196987,1.3050109457699932],"text":["GO:0005515 (10460) <br> protein binding <br> 8.790e-24","GO:0015453 (33) <br> oxidoreduction-driven active transmembrane transporter activity <br> 7.845e-16","GO:0044877 (2038) <br> protein-containing complex binding <br> 8.323e-16","GO:0009055 (54) <br> electron transfer activity <br> 5.814e-14","GO:0036094 (6255) <br> small molecule binding <br> 1.629e-12","GO:0005488 (16378) <br> binding <br> 5.099e-12","GO:0019899 (2330) <br> enzyme binding <br> 2.403e-11","GO:0043167 (6018) <br> ion binding <br> 2.813e-11","GO:0015399 (142) <br> primary active transmembrane transporter activity <br> 1.788e-10","GO:0005102 (1591) <br> signaling receptor binding <br> 1.641e-09","GO:0019838 (141) <br> growth factor binding <br> 1.427e-08","GO:0050839 (292) <br> cell adhesion molecule binding <br> 3.784e-08","GO:0019901 (781) <br> protein kinase binding <br> 8.765e-08","GO:0019900 (872) <br> kinase binding <br> 9.432e-08","GO:0005509 (719) <br> calcium ion binding <br> 1.260e-07","GO:0019904 (802) <br> protein domain specific binding <br> 6.838e-07","GO:0008137 (18) <br> NADH dehydrogenase (ubiquinone) activity <br> 7.363e-07","GO:0022804 (435) <br> active transmembrane transporter activity <br> 8.522e-07","GO:0050136 (19) <br> NADH dehydrogenase (quinone) activity <br> 9.321e-07","GO:0003954 (22) <br> NADH dehydrogenase activity <br> 1.756e-06","GO:0003955 (22) <br> NAD(P)H dehydrogenase (quinone) activity <br> 1.756e-06","GO:0097367 (2284) <br> carbohydrate derivative binding <br> 1.934e-06","GO:0046872 (4176) <br> metal ion binding <br> 3.699e-06","GO:0043169 (4277) <br> cation binding <br> 5.047e-06","GO:0005021 (6) <br> vascular endothelial growth factor receptor activity <br> 5.079e-06","GO:0004129 (7) <br> cytochrome-c oxidase activity <br> 5.812e-06","GO:0008092 (1005) <br> cytoskeletal protein binding <br> 5.961e-06","GO:0016675 (8) <br> oxidoreductase activity, acting on a heme group of donors <br> 9.294e-06","GO:0015078 (122) <br> proton transmembrane transporter activity <br> 1.248e-05","GO:0016655 (38) <br> oxidoreductase activity, acting on NAD(P)H, quinone or similar compound as acceptor <br> 1.755e-05","GO:0030674 (910) <br> protein-macromolecule adaptor activity <br> 2.296e-05","GO:0060090 (1119) <br> molecular adaptor activity <br> 2.442e-05","GO:0140938 (41) <br> histone H3 methyltransferase activity <br> 3.051e-05","GO:0016491 (780) <br> oxidoreductase activity <br> 3.338e-05","GO:0004713 (140) <br> protein tyrosine kinase activity <br> 4.352e-05","GO:0031720 (9) <br> haptoglobin binding <br> 4.609e-05","GO:0019902 (235) <br> phosphatase binding <br> 5.602e-05","GO:0008200 (36) <br> ion channel inhibitor activity <br> 7.581e-05","GO:0003779 (447) <br> actin binding <br> 7.954e-05","GO:0016248 (38) <br> channel inhibitor activity <br> 1.062e-04","GO:0022857 (1087) <br> transmembrane transporter activity <br> 1.115e-04","GO:0019903 (183) <br> protein phosphatase binding <br> 1.153e-04","GO:0019199 (80) <br> transmembrane receptor protein kinase activity <br> 1.283e-04","GO:0035639 (1841) <br> purine ribonucleoside triphosphate binding <br> 1.367e-04","GO:0141110 (40) <br> transporter inhibitor activity <br> 1.460e-04","GO:0045296 (68) <br> cadherin binding <br> 2.146e-04","GO:0046332 (84) <br> SMAD binding <br> 2.205e-04","GO:0016651 (71) <br> oxidoreductase activity, acting on NAD(P)H <br> 2.264e-04","GO:0005215 (1189) <br> transporter activity <br> 2.378e-04","GO:0098772 (2074) <br> molecular function regulator activity <br> 2.424e-04","GO:0005344 (13) <br> oxygen carrier activity <br> 2.572e-04","GO:0001664 (319) <br> G protein-coupled receptor binding <br> 2.961e-04","GO:0005524 (1490) <br> ATP binding <br> 3.381e-04","GO:0005516 (189) <br> calmodulin binding <br> 3.527e-04","GO:0016279 (59) <br> protein-lysine N-methyltransferase activity <br> 4.045e-04","GO:0016278 (59) <br> lysine N-methyltransferase activity <br> 4.045e-04","GO:0042800 (21) <br> histone H3K4 methyltransferase activity <br> 4.256e-04","GO:0140678 (528) <br> molecular function inhibitor activity <br> 4.537e-04","GO:0032555 (1921) <br> purine ribonucleotide binding <br> 4.563e-04","GO:0042802 (2369) <br> identical protein binding <br> 4.708e-04","GO:0032553 (1940) <br> ribonucleotide binding <br> 5.997e-04","GO:0017076 (2040) <br> purine nucleotide binding <br> 1.027e-03","GO:0043168 (2483) <br> anion binding <br> 1.054e-03","GO:0032559 (1560) <br> adenyl ribonucleotide binding <br> 1.055e-03","GO:0042054 (68) <br> histone methyltransferase activity <br> 1.074e-03","GO:0004923 (2) <br> leukemia inhibitory factor receptor activity <br> 1.357e-03","GO:0005178 (143) <br> integrin binding <br> 1.799e-03","GO:0019955 (142) <br> cytokine binding <br> 2.271e-03","GO:0019825 (22) <br> oxygen binding <br> 2.533e-03","GO:0030554 (1678) <br> adenyl nucleotide binding <br> 2.541e-03","GO:0008047 (545) <br> enzyme activator activity <br> 2.900e-03","GO:0004714 (62) <br> transmembrane receptor protein tyrosine kinase activity <br> 3.473e-03","GO:0004924 (3) <br> oncostatin-M receptor activity <br> 4.066e-03","GO:0030169 (18) <br> low-density lipoprotein particle binding <br> 4.681e-03","GO:0003712 (484) <br> transcription coregulator activity <br> 4.854e-03","GO:0030234 (1264) <br> enzyme regulator activity <br> 5.273e-03","GO:0005201 (132) <br> extracellular matrix structural constituent <br> 6.164e-03","GO:0008170 (90) <br> N-methyltransferase activity <br> 7.023e-03","GO:0000166 (2200) <br> nucleotide binding <br> 7.746e-03","GO:1901265 (2200) <br> nucleoside phosphate binding <br> 7.746e-03","GO:0005096 (231) <br> GTPase activator activity <br> 7.868e-03","GO:0004897 (4) <br> ciliary neurotrophic factor receptor activity <br> 8.123e-03","GO:0003824 (5738) <br> catalytic activity <br> 8.549e-03","GO:0044325 (165) <br> transmembrane transporter binding <br> 8.690e-03","GO:0140993 (211) <br> histone modifying activity <br> 9.026e-03","GO:1901363 (2343) <br> heterocyclic compound binding <br> 9.060e-03","GO:0008276 (97) <br> protein methyltransferase activity <br> 1.146e-02","GO:0022853 (234) <br> active monoatomic ion transmembrane transporter activity <br> 1.158e-02","GO:0050998 (22) <br> nitric-oxide synthase binding <br> 1.228e-02","GO:0005319 (179) <br> lipid transporter activity <br> 1.286e-02","GO:0031722 (2) <br> hemoglobin beta binding <br> 1.317e-02","GO:0140999 (10) <br> histone H3K4 trimethyltransferase activity <br> 1.491e-02","GO:0051015 (208) <br> actin filament binding <br> 1.726e-02","GO:0140096 (2341) <br> catalytic activity, acting on a protein <br> 1.826e-02","GO:0022890 (538) <br> inorganic cation transmembrane transporter activity <br> 1.826e-02","GO:0003713 (260) <br> transcription coactivator activity <br> 1.845e-02","GO:0005127 (6) <br> ciliary neurotrophic factor receptor binding <br> 2.026e-02","GO:0038024 (82) <br> cargo receptor activity <br> 2.202e-02","GO:0005041 (10) <br> low-density lipoprotein particle receptor activity <br> 2.482e-02","GO:0008324 (578) <br> monoatomic cation transmembrane transporter activity <br> 2.569e-02","GO:0140657 (585) <br> ATP-dependent activity <br> 2.627e-02","GO:0050840 (64) <br> extracellular matrix binding <br> 2.958e-02","GO:0033218 (440) <br> amide binding <br> 2.972e-02","GO:0005539 (223) <br> glycosaminoglycan binding <br> 3.232e-02","GO:1990409 (2) <br> adrenomedullin binding <br> 3.300e-02","GO:0008013 (95) <br> beta-catenin binding <br> 4.436e-02","GO:0071814 (31) <br> protein-lipid complex binding <br> 4.444e-02","GO:0071813 (31) <br> lipoprotein particle binding <br> 4.444e-02","GO:0015318 (657) <br> inorganic molecular entity transmembrane transporter activity <br> 4.706e-02","GO:0015026 (42) <br> coreceptor activity <br> 4.867e-02","GO:0031625 (331) <br> ubiquitin protein ligase binding <br> 4.954e-02"],"key":["GO:0005515","GO:0015453","GO:0044877","GO:0009055","GO:0036094","GO:0005488","GO:0019899","GO:0043167","GO:0015399","GO:0005102","GO:0019838","GO:0050839","GO:0019901","GO:0019900","GO:0005509","GO:0019904","GO:0008137","GO:0022804","GO:0050136","GO:0003954","GO:0003955","GO:0097367","GO:0046872","GO:0043169","GO:0005021","GO:0004129","GO:0008092","GO:0016675","GO:0015078","GO:0016655","GO:0030674","GO:0060090","GO:0140938","GO:0016491","GO:0004713","GO:0031720","GO:0019902","GO:0008200","GO:0003779","GO:0016248","GO:0022857","GO:0019903","GO:0019199","GO:0035639","GO:0141110","GO:0045296","GO:0046332","GO:0016651","GO:0005215","GO:0098772","GO:0005344","GO:0001664","GO:0005524","GO:0005516","GO:0016279","GO:0016278","GO:0042800","GO:0140678","GO:0032555","GO:0042802","GO:0032553","GO:0017076","GO:0043168","GO:0032559","GO:0042054","GO:0004923","GO:0005178","GO:0019955","GO:0019825","GO:0030554","GO:0008047","GO:0004714","GO:0004924","GO:0030169","GO:0003712","GO:0030234","GO:0005201","GO:0008170","GO:0000166","GO:1901265","GO:0005096","GO:0004897","GO:0003824","GO:0044325","GO:0140993","GO:1901363","GO:0008276","GO:0022853","GO:0050998","GO:0005319","GO:0031722","GO:0140999","GO:0051015","GO:0140096","GO:0022890","GO:0003713","GO:0005127","GO:0038024","GO:0005041","GO:0008324","GO:0140657","GO:0050840","GO:0033218","GO:0005539","GO:1990409","GO:0008013","GO:0071814","GO:0071813","GO:0015318","GO:0015026","GO:0031625"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[21.878108006865304,14.135529956784948,20.057938616091388,15.008384021416694,21.326264703622503,22.345951231091899,20.214526426207573,21.284106448848505,16.549619852843737,19.764336532730745,16.539029588337922,17.587329655016003,18.888894533536845,19.027785947980789,18.783814590379038,18.922448277443845,12.947850468301544,18.128888049401429,13.059967644679528,13.357372072457588,13.357372072457588,20.191302413460068,20.88015064879675,20.90686136579593,10.26251086029124,10.702412765043418,19.204832272187218,11.062026078826225,16.320169044907612,14.392913471576092,19.081201391218425,19.337503222180469,14.528988762818752,18.887272426331844,16.528355056331499,11.365089675535257,17.28313070194687,14.295017909649667,18.165117717346547,14.392913471576092,19.301790254335543,16.924084189674385,15.659075333497245,19.938039509290796,14.484959030043042,15.394442462025141,15.737378400651465,15.465325129039176,19.411926742312581,20.078501291615265,12.241715190548584,17.709305060191621,19.685658526484289,16.970947385124298,15.158234050640996,15.158234050640996,13.264010257621132,18.384881202247836,19.988316067880898,20.233834981047615,19.999926719230256,20.059091168750747,20.288381932139675,19.74077272727305,15.394442462025141,3.7795275590551185,16.560127115942691,16.549619852843737,13.357372072457588,19.827919663385799,18.426325337043654,15.241306099833178,7.7180145958336093,12.947850468301544,18.270476575288036,19.486596171417158,16.439763352761897,15.847237501615188,20.147571021622138,20.147571021622138,17.258790060479736,8.929031646366262,21.231965181186432,16.772535830865632,17.129659550094892,20.221000815842832,15.965383537827867,17.277088702611685,13.357372072457588,16.891883825567096,3.7795275590551185,11.626277095618034,17.109125785991207,20.220007263204501,18.409433036114937,17.42558495636295,10.26251086029124,15.6987686639432,11.626277095618034,18.502900532528347,18.518531689876312,15.294168748695853,18.144114866142381,17.208679838586786,3.7795275590551185,15.932636069737264,14.019399383087203,14.019399383087203,18.668407940029581,14.571784246357929,17.759921323909662],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[154.77321504273721,153.20855532208964,159.64299866767396,154.28096254635366,193.63353584142038,153.32408396920005,164.5228282619247,164.51529378493922,156.49861027240794,156.75227099758513,163.83970234857614,164.71872466354668,157.87993105307589,158.73132695243308,155.23532963117884,175.06858454924262,156.38057013296904,163.56092670011407,160.85856095466178,165.28381043745631,163.65887490092507,153.30901501522914,174.2322576038564,160.80581961576357,176.47250876086699,153.26631964564484,160.58983127551366,158.82927515324408,175.2569464738792,154.65015191864131,152.88457281171478,154.52206580988846,191.20241126744475,155.25291007747825,156.1068174691639,181.2217407540364,192.62140443304003,170.22642733991924,159.81629163833958,153.23869323003146,154.14283046828689,174.92040650186189,175.95262984887015,152.34209046876151,153.20604382976114,163.81709891761975,153.58527917136271,157.74933345199457,153.71336528011557,192.71935263385103,158.53291905848261,153.68322737217372,163.64380594695416,174.91287202487644,153.56518723273481,154.33119239292341,176.90699693369532,152.30441808383421,180.53861484068787,160.46927964374626,156.61413891951835,177.76090432538098,156.02393822232384,154.07250868308924,165.70574114864218,153.11060712127863,174.10166000277502,153.32157247687155,158.39227548808731,158.47766622725587,173.0870171020662,157.40777049532031,153.19850935277569,176.91201991835226,170.80407057547131,195.23837943932375,174.9153835172049,186.15431168718536,155.95612792945468,160.47932561306021,152.3069295761627,170.95476011518051,158.05573551607003,153.14074502922048,158.61077532066568,154.51704282523147,174.59391249915853,156.4081965485824,172.59978759033967,156.05407613026568,163.68147833188146,158.85690156885744,159.64048717534547,184.98646775443876,173.43360304339745,152.3094410684912,153.7108537877871,155.09217456845505,152.30190659150574,158.84685559954352,158.85439007652897,152.35967091506095,179.63447760243247,153.5099344015081,156.60409295020438,157.3299142331372,174.91036053254794,156.47349534912306,156.35545520968415,175.60855539986738,164.87946017256985,156.02142672999534,155.92096703685584,175.03091216431534,155.25793306213521,159.63044120603152,153.21608979907509,155.79539242043148,155.22779515419336,153.26380815331635,152.33706748410455,155.78785794344603,192.73442158782194,192.82985829630445,155.06957113749868,153.2512506916739,155.23281813885035,158.38976399575884,154.30105448498159,179.62694312544701,159.77861925341227,153.69829632614463,154.56476117947273,155.90338659055644,156.76985144388453,167.97110722893765,155.85315674398672,179.68470744900219,174.32769431233891,164.88197166489834,153.47728500123776,153.58025618670575,155.62712243442286,157.71668405172423,155.22528366186489,157.87490806841893,170.20131241663435,154.56727267180122,153.58779066369121,161.51908343705395,172.94386203934241,163.61115654668382,157.75184494432307,153.08800369032224,159.74596985314196,179.81781654241203,158.16372968619496,177.95177774234602,160.86358393931874,170.52027194235222,170.51776045002376,156.66185727375961,155.63716840373681,187.76417826974568,182.92453255275075,171.38924828800884,156.42326550255331,153.69076184915917,181.03086733707138,159.77359626875528,163.88490921048893,194.13332281478932,184.9839562621103,155.65474885003621,170.95727160750903,173.69730973788859,180.95301107488828,155.07961710681263,152.19140092905229,172.37124178844732,187.83952303960029],"y":[14.618708898264535,13.0350873782657,11.690507762505456,11.560692524652737,11.052381931829316,10.743541552838673,10.743541552838673,9.8142726079848437,9.2704622194982136,9.231752076482449,9.0705739146480386,9.0249767816510396,8.9924640366599817,8.9924640366599817,8.7994562799327234,8.7704959304385781,8.657687583362673,8.5026027497206957,8.5026027497206957,8.2895496429522204,8.2605169527285955,7.9315373481937907,7.8364530842002784,7.7118062954219981,7.6483494600284274,7.3841220372347021,7.2451455756432068,7.2329226817100416,6.9768614411900138,6.8728801204643357,6.847581072624398,6.7305095481827513,6.6185692547912245,6.4271752590663667,6.392054240405292,6.3339996991800156,6.3131502872498357,6.2791561045176891,6.1753746872694952,6.0756585787640756,6.0756585787640756,5.8872353476525126,5.7980131309833247,5.6883544129936929,5.5865740242543112,5.547735984736673,5.5438539815152419,5.4279837643517679,5.3199461326248496,5.2666978201374395,5.2434041487620178,5.2434041487620178,5.2284586065250318,5.1109647020815494,5.0449735270391622,5.010509300129514,5.0084613413630423,4.9692250291491336,4.9486562711761257,4.9319733851125589,4.8381724418626257,4.8381724418626257,4.797015627695485,4.7780916034260175,4.7367802464283244,4.7071236869074431,4.6816222295157353,4.6777494537706517,4.6605833026066215,4.6369107822593234,4.5369270434565987,4.5178893404202745,4.4486597416790197,4.4143795674621549,4.2895656805682778,4.1839758278990011,4.1728995609666351,4.1288820574343612,4.1288820574343612,4.1048867279621239,4.0546948505677829,4.0135957860067881,3.9818586323505989,3.9527189052449154,3.9169740286732817,3.8863236739113121,3.848379947945392,3.7684685642910178,3.7504431831625036,3.6786491958275951,3.664265970673442,3.6500398188853329,3.6425428456353672,3.5863682614283507,3.5538928550488995,3.4576586410916557,3.3562397349262767,3.3012993181265204,3.2625406237046941,3.2588383395752651,3.2588383395752651,3.2304868681157073,3.1966479445967408,3.1769426203375408,3.1098052364025648,3.1098052364025648,3.1046356349889828,3.0455772919560853,3.014156457030877,2.9985262134790016,2.9909797656661179,2.9507209936163616,2.9064416839754927,2.8796147259843656,2.876529801183926,2.8450852002612517,2.8422100703511419,2.7794139620438676,2.7700293404068534,2.7632725603197579,2.7441197685377134,2.7356041594299971,2.7332022009495263,2.7206281277486459,2.628033542798935,2.6251247980688364,2.6232368366301371,2.5805188179310639,2.559270845088069,2.5464177933285645,2.545593091113842,2.5430097027751422,2.5297732165135853,2.5247171592412316,2.5226536417993772,2.5214306624967242,2.4842842668033294,2.4760882824264621,2.4597892113984976,2.3460266168774648,2.3415867083389088,2.2951399239544985,2.2549527148621422,2.2046213338182419,2.1936851497087546,2.1434289755713514,2.0934087588508863,2.0865294968009205,2.0827267937258731,2.0519799461984887,2.0519799461984887,2.0414432443916781,1.9823338588149864,1.9784344249725214,1.976629839335607,1.9755442251756066,1.9703582309036671,1.9467985913759907,1.9413234913073127,1.9357326429406443,1.9357326429406443,1.8867336779202946,1.8856112503202869,1.868815819218838,1.8122883997838479,1.7324567792255019,1.7302118515502185,1.7020014695401009,1.6752888874357186,1.6752888874357186,1.6727890075347054,1.663884277014859,1.6188124369570138,1.6013766368816018,1.5833575515503029,1.5684389503032488,1.5049060747302792,1.4930609203366656,1.4907120541791288,1.3700002268779405,1.3233391679525224],"text":["HP:0001427 (16) <br> Mitochondrial inheritance <br> 2.406e-15","HP:0000576 (13) <br> Centrocecal scotoma <br> 9.224e-14","HP:0004309 (29) <br> Ventricular preexcitation <br> 2.039e-12","HP:0001112 (10) <br> Leber optic atrophy <br> 2.750e-12","HP:0200125 (20) <br> Mitochondrial respiratory chain defects <br> 8.864e-12","HP:0000631 (12) <br> Retinal arterial tortuosity <br> 1.805e-11","HP:0007768 (12) <br> Central retinal vessel vascular tortuosity <br> 1.805e-11","HP:0007763 (27) <br> Retinal telangiectasia <br> 1.534e-10","HP:0002401 (21) <br> Stroke-like episode <br> 5.365e-10","HP:0002572 (51) <br> Episodic vomiting <br> 5.865e-10","HP:0007327 (11) <br> Mixed demyelinating and axonal polyneuropathy <br> 8.500e-10","HP:0007924 (33) <br> Slow decrease in visual acuity <br> 9.441e-10","HP:0003200 (54) <br> Ragged-red muscle fibers <br> 1.018e-09","HP:0003737 (54) <br> Mitochondrial myopathy <br> 1.018e-09","HP:0001716 (24) <br> Wolff-Parkinson-White syndrome <br> 1.587e-09","HP:0012766 (12) <br> Widened cerebral subarachnoid space <br> 1.696e-09","HP:0002331 (25) <br> Recurrent paroxysmal headache <br> 2.199e-09","HP:0007067 (13) <br> Distal peripheral sensory neuropathy <br> 3.143e-09","HP:0005157 (13) <br> Concentric hypertrophic cardiomyopathy <br> 3.143e-09","HP:0008316 (40) <br> Abnormal mitochondria in muscle tissue <br> 5.134e-09","HP:0007159 (14) <br> Fluctuations in consciousness <br> 5.489e-09","HP:0000622 (44) <br> Blurred vision <br> 1.171e-08","HP:0012429 (16) <br> Aplasia/Hypoplasia of the cerebral white matter <br> 1.457e-08","HP:0005116 (27) <br> Arterial tortuosity <br> 1.942e-08","HP:0025268 (17) <br> Stuttering <br> 2.247e-08","HP:0000603 (51) <br> Central scotoma <br> 4.129e-08","HP:0004948 (31) <br> Vascular tortuosity <br> 5.687e-08","HP:0003800 (83) <br> Muscle abnormality related to mitochondrial dysfunction <br> 5.849e-08","HP:0012841 (57) <br> Retinal vascular tortuosity <br> 1.055e-07","HP:0001345 (22) <br> Psychotic mentation <br> 1.340e-07","HP:0000408 (43) <br> Progressive sensorineural hearing impairment <br> 1.420e-07","HP:0001271 (61) <br> Polyneuropathy <br> 1.860e-07","HP:0100027 (24) <br> Recurrent pancreatitis <br> 2.407e-07","HP:0001730 (49) <br> Progressive hearing impairment <br> 3.740e-07","HP:0002174 (67) <br> Postural tremor <br> 4.055e-07","HP:0031546 (149) <br> Cardiac conduction abnormality <br> 4.634e-07","HP:0100611 (10) <br> Multiple glomerular cysts <br> 4.862e-07","HP:0010794 (27) <br> Impaired visuospatial constructive cognition <br> 5.258e-07","HP:0004389 (28) <br> Intestinal pseudo-obstruction <br> 6.678e-07","HP:0000590 (29) <br> Progressive external ophthalmoplegia <br> 8.401e-07","HP:0001045 (29) <br> Vitiligo <br> 8.401e-07","HP:0012707 (31) <br> Elevated brain lactate level by MRS <br> 1.296e-06","HP:0025045 (32) <br> Abnormal brain lactate level by MRS <br> 1.592e-06","HP:0000114 (50) <br> Proximal tubulopathy <br> 2.049e-06","HP:0000575 (84) <br> Scotoma <br> 2.591e-06","HP:0007302 (35) <br> Bipolar affective disorder <br> 2.833e-06","HP:0000763 (100) <br> Sensory neuropathy <br> 2.859e-06","HP:0003128 (184) <br> Lactic acidosis <br> 3.733e-06","HP:0000829 (38) <br> Hypoparathyroidism <br> 4.787e-06","HP:0100651 (60) <br> Type I diabetes mellitus <br> 5.411e-06","HP:0003572 (15) <br> Low plasma citrulline <br> 5.709e-06","HP:0000816 (15) <br> Abnormality of Krebs cycle metabolism <br> 5.709e-06","HP:0007141 (93) <br> Sensorimotor neuropathy <br> 5.909e-06","HP:0012704 (41) <br> Widened subarachnoid space <br> 7.745e-06","HP:0000751 (42) <br> Personality changes <br> 9.016e-06","HP:0001138 (62) <br> Optic neuropathy <br> 9.761e-06","HP:0025454 (146) <br> Abnormal CSF metabolite concentration <br> 9.807e-06","HP:0000095 (205) <br> Abnormal renal glomerulus morphology <br> 1.073e-05","HP:0031263 (206) <br> Abnormal renal corpuscle morphology <br> 1.125e-05","HP:0004885 (17) <br> Episodic respiratory distress <br> 1.170e-05","HP:0002490 (104) <br> Increased CSF lactate <br> 1.452e-05","HP:0030085 (104) <br> Abnormal CSF lactate concentration <br> 1.452e-05","HP:0002135 (46) <br> Basal ganglia calcification <br> 1.596e-05","HP:0001009 (155) <br> Telangiectasia <br> 1.667e-05","HP:0008619 (84) <br> Bilateral sensorineural hearing impairment <br> 1.833e-05","HP:0000529 (108) <br> Progressive visual loss <br> 1.963e-05","HP:0012377 (48) <br> Hemianopia <br> 2.082e-05","HP:0000630 (69) <br> Abnormal retinal artery morphology <br> 2.100e-05","HP:0003481 (19) <br> Segmental peripheral demyelination/remyelination <br> 2.185e-05","HP:0003546 (138) <br> Exercise intolerance <br> 2.307e-05","HP:0011965 (20) <br> Abnormal circulating citrulline concentration <br> 2.905e-05","HP:0002922 (51) <br> Increased CSF protein concentration <br> 3.035e-05","HP:0000572 (232) <br> Visual loss <br> 3.559e-05","HP:0025456 (53) <br> Abnormal CSF protein concentration <br> 3.851e-05","HP:0011035 (241) <br> Abnormal renal cortex morphology <br> 5.134e-05","HP:3000036 (81) <br> Abnormal head blood vessel morphology <br> 6.547e-05","HP:0012705 (58) <br> Abnormal metabolic brain imaging by MRS <br> 6.716e-05","HP:0033578 (142) <br> Pre-capillary pulmonary hypertension <br> 7.432e-05","HP:0002092 (142) <br> Pulmonary arterial hypertension <br> 7.432e-05","HP:0004890 (143) <br> Elevated pulmonary artery pressure <br> 7.854e-05","HP:0000096 (105) <br> Glomerular sclerosis <br> 8.817e-05","HP:0011096 (50) <br> Peripheral demyelination <br> 9.692e-05","HP:0003287 (191) <br> Abnormality of mitochondrial metabolism <br> 1.043e-04","HP:0000544 (63) <br> External ophthalmoplegia <br> 1.115e-04","HP:0003648 (26) <br> Lacticaciduria <br> 1.211e-04","HP:0001269 (111) <br> Hemiparesis <br> 1.299e-04","HP:0012575 (268) <br> Abnormal nephron morphology <br> 1.418e-04","HP:0002345 (204) <br> Action tremor <br> 1.704e-04","HP:0011767 (68) <br> Abnormality of the parathyroid physiology <br> 1.776e-04","HP:0002151 (207) <br> Increased serum lactate <br> 2.096e-04","HP:0007183 (29) <br> Focal T2 hyperintense basal ganglia lesion <br> 2.166e-04","HP:0003829 (147) <br> Typified by incomplete penetrance <br> 2.239e-04","HP:0004308 (209) <br> Ventricular arrhythmia <br> 2.277e-04","HP:0033109 (30) <br> Abnormal circulating non-proteinogenic amino acid concentration <br> 2.592e-04","HP:0012103 (214) <br> Abnormality of the mitochondrion <br> 2.793e-04","HP:0000097 (76) <br> Focal segmental glomerulosclerosis <br> 3.486e-04","HP:0000828 (79) <br> Abnormality of the parathyroid gland <br> 4.403e-04","HP:0001644 (163) <br> Dilated cardiomyopathy <br> 4.997e-04","HP:0000093 (244) <br> Proteinuria <br> 5.463e-04","HP:0003812 (82) <br> Phenotypic variability <br> 5.510e-04","HP:0003828 (82) <br> Variable expressivity <br> 5.510e-04","HP:0000124 (111) <br> Renal tubular dysfunction <br> 5.882e-04","HP:0030875 (187) <br> Abnormality of pulmonary circulation <br> 6.358e-04","HP:0000726 (212) <br> Dementia <br> 6.654e-04","HP:0002483 (37) <br> Bulbar signs <br> 7.766e-04","HP:0002883 (37) <br> Hyperventilation <br> 7.766e-04","HP:0012703 (87) <br> Abnormal subarachnoid space morphology <br> 7.859e-04","HP:0002381 (89) <br> Aphasia <br> 9.004e-04","HP:0002315 (347) <br> Headache <br> 9.679e-04","HP:0020129 (262) <br> Abnormal urine protein level <br> 1.003e-03","HP:0008046 (257) <br> Abnormal retinal vascular morphology <br> 1.021e-03","HP:0002134 (181) <br> Abnormal basal ganglia morphology <br> 1.120e-03","HP:0002076 (154) <br> Migraine <br> 1.240e-03","HP:0012751 (41) <br> Abnormal basal ganglia MRI signal intensity <br> 1.319e-03","HP:0001733 (95) <br> Pancreatitis <br> 1.329e-03","HP:0004303 (259) <br> Abnormal muscle fiber morphology <br> 1.429e-03","HP:0000580 (187) <br> Pigmentary retinopathy <br> 1.438e-03","HP:0002017 (551) <br> Nausea and vomiting <br> 1.662e-03","HP:0001712 (99) <br> Left ventricular hypertrophy <br> 1.698e-03","HP:0000602 (240) <br> Ophthalmoplegia <br> 1.725e-03","HP:0000112 (100) <br> Nephropathy <br> 1.803e-03","HP:0002013 (352) <br> Vomiting <br> 1.838e-03","HP:0100659 (337) <br> Abnormal cerebral vascular morphology <br> 1.848e-03","HP:0100704 (164) <br> Cerebral visual impairment <br> 1.903e-03","HP:0001635 (250) <br> Congestive heart failure <br> 2.355e-03","HP:0000597 (275) <br> Ophthalmoparesis <br> 2.371e-03","HP:0001714 (151) <br> Ventricular hypertrophy <br> 2.381e-03","HP:0003477 (172) <br> Peripheral axonal neuropathy <br> 2.627e-03","HP:0001123 (280) <br> Visual field defect <br> 2.759e-03","HP:0030872 (155) <br> Abnormal cardiac ventricular function <br> 2.842e-03","HP:0004374 (241) <br> Hemiplegia/hemiparesis <br> 2.847e-03","HP:0000822 (390) <br> Hypertension <br> 2.864e-03","HP:0001297 (175) <br> Stroke <br> 2.953e-03","HP:0002069 (371) <br> Bilateral tonic-clonic seizure <br> 2.987e-03","HP:0002579 (109) <br> Gastrointestinal dysmotility <br> 3.002e-03","HP:0009830 (471) <br> Peripheral neuropathy <br> 3.010e-03","HP:0002045 (49) <br> Hypothermia <br> 3.279e-03","HP:0030895 (111) <br> Abnormal gastrointestinal motility <br> 3.341e-03","HP:0012469 (144) <br> Infantile spasms <br> 3.469e-03","HP:0008047 (388) <br> Abnormality of the vasculature of the eye <br> 4.508e-03","HP:0000709 (117) <br> Psychosis <br> 4.554e-03","HP:0000759 (393) <br> Abnormal peripheral nervous system morphology <br> 5.068e-03","HP:0001941 (397) <br> Acidosis <br> 5.560e-03","HP:0003110 (750) <br> Abnormality of urine homeostasis <br> 6.243e-03","HP:0001711 (124) <br> Abnormal left ventricle morphology <br> 6.402e-03","HP:0003198 (314) <br> Myopathy <br> 7.187e-03","HP:0010783 (129) <br> Erythema <br> 8.065e-03","HP:0001298 (319) <br> Encephalopathy <br> 8.194e-03","HP:0000764 (204) <br> Peripheral axonal degeneration <br> 8.266e-03","HP:0005560 (3) <br> Imbalanced hemoglobin synthesis <br> 8.872e-03","HP:0011907 (3) <br> Reduced alpha/beta synthesis ratio <br> 8.872e-03","HP:0007108 (60) <br> Demyelinating peripheral neuropathy <br> 9.090e-03","HP:0003130 (109) <br> Abnormal peripheral myelination <br> 1.042e-02","HP:0000519 (135) <br> Developmental cataract <br> 1.051e-02","HP:0004360 (426) <br> Abnormality of acid-base homeostasis <br> 1.055e-02","HP:0030956 (541) <br> Abnormality of cardiovascular system electrophysiology <br> 1.058e-02","HP:0003348 (62) <br> Hyperalaninemia <br> 1.071e-02","HP:0030163 (320) <br> Abnormal vascular physiology <br> 1.130e-02","HP:0005162 (137) <br> Abnormal left ventricular function <br> 1.145e-02","HP:0010916 (63) <br> Abnormal circulating alanine concentration <br> 1.159e-02","HP:0010915 (63) <br> Abnormal circulating pyruvate family amino acid concentration <br> 1.159e-02","HP:0002514 (140) <br> Cerebral calcification <br> 1.298e-02","HP:0001945 (436) <br> Fever <br> 1.301e-02","HP:0034221 (141) <br> Abnormal temporal lobe morphology <br> 1.353e-02","HP:0032263 (469) <br> Increased blood pressure <br> 1.541e-02","HP:0011276 (566) <br> Vascular skin abnormality <br> 1.852e-02","HP:0002354 (149) <br> Memory impairment <br> 1.861e-02","HP:0000819 (457) <br> Diabetes mellitus <br> 1.986e-02","HP:0031466 (235) <br> Impairment in personality functioning <br> 2.112e-02","HP:0004372 (235) <br> Reduced consciousness <br> 2.112e-02","HP:0007359 (335) <br> Focal-onset seizure <br> 2.124e-02","HP:0410263 (153) <br> Brain imaging abnormality <br> 2.168e-02","HP:0033108 (73) <br> Abnormal circulating proteinogenic amino acid derivative concentration <br> 2.405e-02","HP:0001952 (469) <br> Glucose intolerance <br> 2.504e-02","HP:0011097 (195) <br> Epileptic spasm <br> 2.610e-02","HP:0012211 (715) <br> Abnormal renal physiology <br> 2.701e-02","HP:0031434 (77) <br> Abnormal prosody <br> 3.127e-02","HP:0001639 (283) <br> Hypertrophic cardiomyopathy <br> 3.213e-02","HP:0000044 (164) <br> Hypogonadotropic hypogonadism <br> 3.231e-02","HP:0011675 (498) <br> Arrhythmia <br> 4.266e-02","HP:0034251 (11) <br> Abnormal corneal reflex <br> 4.750e-02"],"key":["HP:0001427","HP:0000576","HP:0004309","HP:0001112","HP:0200125","HP:0000631","HP:0007768","HP:0007763","HP:0002401","HP:0002572","HP:0007327","HP:0007924","HP:0003200","HP:0003737","HP:0001716","HP:0012766","HP:0002331","HP:0007067","HP:0005157","HP:0008316","HP:0007159","HP:0000622","HP:0012429","HP:0005116","HP:0025268","HP:0000603","HP:0004948","HP:0003800","HP:0012841","HP:0001345","HP:0000408","HP:0001271","HP:0100027","HP:0001730","HP:0002174","HP:0031546","HP:0100611","HP:0010794","HP:0004389","HP:0000590","HP:0001045","HP:0012707","HP:0025045","HP:0000114","HP:0000575","HP:0007302","HP:0000763","HP:0003128","HP:0000829","HP:0100651","HP:0003572","HP:0000816","HP:0007141","HP:0012704","HP:0000751","HP:0001138","HP:0025454","HP:0000095","HP:0031263","HP:0004885","HP:0002490","HP:0030085","HP:0002135","HP:0001009","HP:0008619","HP:0000529","HP:0012377","HP:0000630","HP:0003481","HP:0003546","HP:0011965","HP:0002922","HP:0000572","HP:0025456","HP:0011035","HP:3000036","HP:0012705","HP:0033578","HP:0002092","HP:0004890","HP:0000096","HP:0011096","HP:0003287","HP:0000544","HP:0003648","HP:0001269","HP:0012575","HP:0002345","HP:0011767","HP:0002151","HP:0007183","HP:0003829","HP:0004308","HP:0033109","HP:0012103","HP:0000097","HP:0000828","HP:0001644","HP:0000093","HP:0003812","HP:0003828","HP:0000124","HP:0030875","HP:0000726","HP:0002483","HP:0002883","HP:0012703","HP:0002381","HP:0002315","HP:0020129","HP:0008046","HP:0002134","HP:0002076","HP:0012751","HP:0001733","HP:0004303","HP:0000580","HP:0002017","HP:0001712","HP:0000602","HP:0000112","HP:0002013","HP:0100659","HP:0100704","HP:0001635","HP:0000597","HP:0001714","HP:0003477","HP:0001123","HP:0030872","HP:0004374","HP:0000822","HP:0001297","HP:0002069","HP:0002579","HP:0009830","HP:0002045","HP:0030895","HP:0012469","HP:0008047","HP:0000709","HP:0000759","HP:0001941","HP:0003110","HP:0001711","HP:0003198","HP:0010783","HP:0001298","HP:0000764","HP:0005560","HP:0011907","HP:0007108","HP:0003130","HP:0000519","HP:0004360","HP:0030956","HP:0003348","HP:0030163","HP:0005162","HP:0010916","HP:0010915","HP:0002514","HP:0001945","HP:0034221","HP:0032263","HP:0011276","HP:0002354","HP:0000819","HP:0031466","HP:0004372","HP:0007359","HP:0410263","HP:0033108","HP:0001952","HP:0011097","HP:0012211","HP:0031434","HP:0001639","HP:0000044","HP:0011675","HP:0034251"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(153,0,153,1)","opacity":0.80000000000000004,"size":[12.698730272061621,12.241715190548584,13.894051557160756,11.626277095618034,13.165094704134018,12.058806804819245,12.058806804819245,13.75799563321786,13.264010257621132,14.910589323184952,11.85527601834232,14.135529956784948,15.008384021416694,15.008384021416694,13.529597157502854,12.058806804819245,13.609357884847448,12.241715190548584,12.241715190548584,14.484959030043042,12.407605671852071,14.653923996014182,12.698730272061621,13.75799563321786,12.82781267352663,14.910589323184952,14.019399383087203,15.718205436949759,15.100112379062027,13.357372072457588,14.613409556027271,15.214137151492038,13.529597157502854,14.84162824092461,15.370018261321293,16.621495154335239,11.626277095618034,13.75799563321786,13.827468853955285,13.894051557160756,13.894051557160756,14.019399383087203,14.078535733677407,14.876507174207195,15.737378400651465,14.243648060312834,16.013102217524118,16.9320121865773,14.392913471576092,15.186452785376668,12.55922905901641,12.55922905901641,15.899100466410918,14.528988762818752,14.571784246357929,15.241306099833178,16.591163144568295,17.088261325299367,17.095253521087105,12.82781267352663,16.074274264307061,16.074274264307061,14.73183631961966,16.680165266632077,15.737378400651465,16.132851395074049,14.805916338696848,15.418459955353805,13.059967644679528,16.506747981081492,13.165094704134018,14.910589323184952,17.26491864511404,14.976496866448542,17.318796326902053,15.679061062146381,15.129460953286518,16.549619852843737,16.549619852843737,16.560127115942691,16.089153481435879,14.876507174207195,16.986202418984014,15.267977608713885,13.685385010878013,16.17520418195295,17.467999329482485,17.081231235176517,15.394442462025141,17.102208213102088,13.894051557160756,16.601350517586557,17.116006618603713,13.957963134999822,17.149872545660465,15.576194582978985,15.638803886576623,16.754569576674172,17.336263199389737,15.6987686639432,15.6987686639432,16.17520418195295,16.955512115652038,17.136432378704722,14.344740593128281,14.344740593128281,15.793380737463837,15.829513939414692,17.824361560542972,17.436322104692952,17.409307263598343,16.908082832631191,16.670564656334751,14.528988762818752,15.932636069737264,17.420182175893537,16.955512115652038,18.44061743110602,15.997377539550557,17.312920622487489,16.013102217524118,17.843832550907738,17.784479219279955,16.76358319901939,17.370496430649634,17.503982730637716,16.641340264931827,16.833560798433115,17.529072820646299,16.680165266632077,17.318796326902053,17.982575211917933,16.85887372598534,17.915151124757344,16.147114504766865,18.23449212306971,14.84162824092461,16.17520418195295,16.570552615864109,17.975649234564241,16.256176327006145,17.992891601414531,18.006513480603225,18.837532319074207,16.344946758710758,17.687594297751961,16.404980881943121,17.709305060191621,17.081231235176517,7.7180145958336093,7.7180145958336093,15.186452785376668,16.147114504766865,16.47367201217347,18.10099150922397,18.416701738523138,15.241306099833178,17.713602340529537,16.495812758659994,15.267977608713885,15.267977608713885,16.528355056331499,18.13194866318274,16.539029588337922,18.22885995331626,18.475618812957247,16.621495154335239,18.194506632993676,17.28313070194687,17.28313070194687,17.776346948075933,16.66089427540085,15.510708830809184,18.22885995331626,17.016187647017556,18.776700736793153,15.597371909790384,17.54389117666507,16.76358319901939,18.308068239435634,11.85527601834232],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(153,0,153,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"HP","legendgroup":"HP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[116.16110595420557,117.353578773306,117.2730742791895,116.99382431522295,117.15231753801478,117.16238059977934,117.15734906889705,117.15483330345593,117.14980177257364,116.91080405566532,117.16489636522047,117.36112606962942,116.86048874684252,117.35609453874713,116.97872972257611,117.21269590860214,117.09445493286856,116.7397320056678,117.00388737698751],"y":[8.8508058170044688,7.1893580619246888,6.8821256028827049,6.7627622710638011,6.2330192879141881,6.1898017531642022,5.7207043218411942,4.9761509025922681,4.8367121460237614,4.1567773835023845,4.0624293739962303,3.470191989234475,2.596397716028688,2.2219914088866171,1.8737625068034864,1.8452489510683268,1.6179020015568462,1.5749947699872213,1.3688727202656137],"text":["KEGG:00190 (134) <br> Oxidative phosphorylation <br> 1.410e-09","KEGG:05415 (205) <br> Diabetic cardiomyopathy <br> 6.466e-08","KEGG:05208 (222) <br> Chemical carcinogenesis - reactive oxygen species <br> 1.312e-07","KEGG:04714 (229) <br> Thermogenesis <br> 1.727e-07","KEGG:05012 (263) <br> Parkinson disease <br> 5.848e-07","KEGG:05020 (266) <br> Prion disease <br> 6.459e-07","KEGG:05016 (301) <br> Huntington disease <br> 1.902e-06","KEGG:05014 (367) <br> Amyotrophic lateral sclerosis <br> 1.056e-05","KEGG:05010 (381) <br> Alzheimer disease <br> 1.456e-05","KEGG:04520 (92) <br> Adherens junction <br> 6.970e-05","KEGG:05022 (470) <br> Pathways of neurodegeneration - multiple diseases <br> 8.661e-05","KEGG:05418 (146) <br> Fluid shear stress and atherosclerosis <br> 3.387e-04","KEGG:04260 (87) <br> Cardiac muscle contraction <br> 2.533e-03","KEGG:05416 (87) <br> Viral myocarditis <br> 5.998e-03","KEGG:04670 (118) <br> Leukocyte transendothelial migration <br> 1.337e-02","KEGG:05144 (55) <br> Malaria <br> 1.428e-02","KEGG:04932 (155) <br> Non-alcoholic fatty liver disease <br> 2.410e-02","KEGG:04015 (214) <br> Rap1 signaling pathway <br> 2.661e-02","KEGG:04723 (147) <br> Retrograde endocannabinoid signaling <br> 4.277e-02"],"key":["KEGG:00190","KEGG:05415","KEGG:05208","KEGG:04714","KEGG:05012","KEGG:05020","KEGG:05016","KEGG:05014","KEGG:05010","KEGG:04520","KEGG:05022","KEGG:05418","KEGG:04260","KEGG:05416","KEGG:04670","KEGG:05144","KEGG:04932","KEGG:04015","KEGG:04723"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":[16.46246361757693,17.088261325299367,17.202276580201648,17.246444436276072,17.441656841322331,17.457527867895021,17.629319291960101,17.900474581478587,17.951096771924238,15.882025628956763,18.23167930839508,16.591163144568295,15.793380737463837,15.793380737463837,16.269217457803322,15.0395982278933,16.680165266632077,17.149872545660465,16.601350517586557],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[143.57658220387162],"y":[1.3948543952123937],"text":"MIRNA:mmu-miR-2182 (6) <br> mmu-miR-2182 <br> 4.029e-02","key":["MIRNA:mmu-miR-2182"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(34,170,153,1)","opacity":0.80000000000000004,"size":10.26251086029124,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(34,170,153,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"MIRNA","legendgroup":"MIRNA","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[123.34538677933028,123.34287398171433,124.01630374279088,122.73980255388457,120.83761475860489,123.34789957694625,120.89792190138786,124.26255790915469,123.6745632670207,121.80755463836441],"y":[8.6103659203811667,7.737555728007492,7.2640237717521616,2.4864471715876677,2.446975926092132,2.3277030169196977,2.191640094959618,2.1661637887128853,2.1249974614673173,1.9531268204668872],"text":["REAC:R-MMU-163200 (113) <br> Respiratory electron transport, ATP synthesis by chemiosmotic coupling, and heat production by uncoupling proteins. <br> 2.453e-09","REAC:R-MMU-611105 (90) <br> Respiratory electron transport <br> 1.830e-08","REAC:R-MMU-1428517 (159) <br> The citric acid (TCA) cycle and respiratory electron transport <br> 5.445e-08","REAC:R-MMU-114608 (111) <br> Platelet degranulation  <br> 3.263e-03","REAC:R-MMU-6799198 (52) <br> Complex I biogenesis <br> 3.573e-03","REAC:R-MMU-76005 (115) <br> Response to elevated platelet cytosolic Ca2+ <br> 4.702e-03","REAC:R-MMU-9707564 (51) <br> Cytoprotection by HMOX1 <br> 6.432e-03","REAC:R-MMU-5218920 (29) <br> VEGFR2 mediated vascular permeability <br> 6.821e-03","REAC:R-MMU-194138 (98) <br> Signaling by VEGF <br> 7.499e-03","REAC:R-MMU-445095 (9) <br> Interaction between L1 and Ankyrins <br> 1.114e-02"],"key":["REAC:R-MMU-163200","REAC:R-MMU-611105","REAC:R-MMU-1428517","REAC:R-MMU-114608","REAC:R-MMU-6799198","REAC:R-MMU-76005","REAC:R-MMU-9707564","REAC:R-MMU-5218920","REAC:R-MMU-194138","REAC:R-MMU-445095"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":[16.202730409602214,15.847237501615188,16.717888898684524,16.17520418195295,14.943908607777407,16.229714430385808,14.910589323184952,13.894051557160756,15.981472643387029,11.365089675535257],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"REAC","legendgroup":"REAC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[136.8191993562119,128.91664745231927,136.81668741981522,127.226114257367,136.65089961763567,128.86138485159273,135.244215235506,128.9141355159226,131.58934777836558,128.49464213768036,127.19345908421042,131.6069313331422,127.65063150840255,127.18592327502044,127.16331584745049,127.23616200295365,135.22914361712606,128.61019121192672,130.38361830796873,135.26179879028263,136.55544603456258,136.56046990735589,129.0648516997222,131.58683584196893,135.24672717190268,127.23113813016032,136.56549378014921,133.50093137622392,128.80109837807288,129.08243525449882,132.33539288817363,128.63028670310001,127.16080391105383,136.57051765294256,135.27184653586929,136.26154947615333,136.78905611945197,137.53258929286335,127.78376413742555,136.56298184375257,136.25401366696337,137.14323915138104,131.77020719892511,128.85887291519609,130.37859443517542,136.66597123601562,127.22109038457369,132.12187829445753,135.25677491748931,135.25175104469599,136.78403224665865,131.29293928355969,136.45748051509284,127.610440526056,127.01511160004755,131.61697907872883,129.87871909224006,131.67475361585201,128.75085965013969,134.73429214698402,136.14348846551033,129.07992331810215,135.26682266307594,128.65540606706662,136.78654418305533,127.23867393935029,136.25903753975669,127.18089940222711,127.17336359303712,127.17085165664048,128.55744054759685,128.83375355122948,132.12690216725085,128.31127078072419,135.3396688185791,131.84305335442826,135.28943009064588,136.54288635257927,135.26933459947261,134.44290752497145,132.29520190582707,129.06987557251551,128.50971375606031,135.23919136271269,127.1758755294338,128.64033444868664,136.13344071992367,131.01913821632374,136.55795797095925,137.53007735646671,127.23365006655698,127.17838746583047,137.14575108777768,128.82872967843613,135.23416748991937,127.18341133862378,136.90209325730169,133.44315683910074,127.2135545753837,127.19094714781376,135.2643107266793,127.60792858965934,137.54514897484665,132.11434248526754,136.55042216176926,129.64762094374734,136.65341155403232,134.2118093764787,135.28691815424924],"y":[8.0862661685390851,7.3065183518909214,7.0110749566003454,6.97841557819096,6.0413640106366495,5.8274122897327612,5.4960765214011698,5.0164953344638281,4.9216034970054112,4.7528738248716165,4.4981759853696879,4.4131674255832865,4.2512788875188869,4.2111032639678632,4.1122040280068122,3.9905242206229659,3.9644020964814519,3.7401233059304997,3.7095795236233799,3.6506793867200997,3.5956789826854405,3.5929524535917738,3.5908146835010757,3.5808414981088781,3.4407433616527516,3.3639230471361934,3.331743238325569,3.3068663959706974,3.1127871187894374,3.1094614405283529,3.082888960361617,3.0611081573755916,3.057814194417233,3.0523675388880975,3.024488414334991,3.0160415047939271,2.9956370949469808,2.986654626507832,2.9689498908983545,2.909144310008656,2.8429484724013565,2.7497643156998728,2.7312307300275211,2.724085556644857,2.7017454213273968,2.6427572529813701,2.6419703575577733,2.5991088734263768,2.5713608423778549,2.4711448472665922,2.4472148909642688,2.4389262921680048,2.3738403406351889,2.3620153634688772,2.3571119290220297,2.3428646930928374,2.3359795743227569,2.3109991560055763,2.2843643001789058,2.2331398956099808,2.181372208241573,2.178299674785432,2.1714567272278034,2.1393648397668295,2.1354331871021035,2.1110752438178739,2.1042837601758815,2.0915016977393242,2.067723778093923,2.067723778093923,2.0153497561634279,1.9719417439313189,1.9717072011938093,1.941397694184221,1.941397694184221,1.941397694184221,1.941397694184221,1.8939674687742529,1.8937573519488953,1.8843842411690828,1.8678192784256065,1.8336030345767573,1.7890779018627048,1.7864674120554473,1.7801990040742564,1.7456553960024659,1.6881535803678274,1.6769807540471431,1.607218944705054,1.6057861496107362,1.6002862070271007,1.5922499148380014,1.5718221621797388,1.5680188073686701,1.558822641441572,1.5588144825485792,1.5560931414085613,1.5016095273776524,1.4825405930303366,1.4764381344958688,1.4713704494417919,1.4489752502916735,1.4468516601895633,1.4415325582673701,1.4163791144838542,1.4082824670879246,1.3918507538118468,1.3750906863290318,1.326162604939904],"text":["TF:M02933_1 (5832) <br> Factor: ZF5; motif: GYCGCGCARNGCNN; match class: 1 <br> 8.198e-09","TF:M00803_1 (7031) <br> Factor: E2F; motif: GGCGSG; match class: 1 <br> 4.937e-08","TF:M02933 (9499) <br> Factor: ZF5; motif: GYCGCGCARNGCNN <br> 9.748e-08","TF:M08867 (7232) <br> Factor: AP2; motif: GCCYGSGGSN <br> 1.051e-07","TF:M05547 (6680) <br> Factor: ZAC; motif: KGGGCCR <br> 9.092e-07","TF:M02089_1 (6886) <br> Factor: E2F-3; motif: GGCGGGN; match class: 1 <br> 1.488e-06","TF:M10375 (10720) <br> Factor: Sp1; motif: GGNGGGGGNGGGGGMGGGGCNGGG <br> 3.191e-06","TF:M00803 (10412) <br> Factor: E2F; motif: GGCGSG <br> 9.627e-06","TF:M03876_1 (5836) <br> Factor: Kaiso; motif: GCMGGGRGCRGS; match class: 1 <br> 1.198e-05","TF:M08995_1 (9327) <br> Factor: CTCF; motif: NAGGGGGCGCNNKNNNN; match class: 1 <br> 1.767e-05","TF:M00470_1 (3399) <br> Factor: AP-2gamma; motif: GCCYNNGGS; match class: 1 <br> 3.176e-05","TF:M10277 (3253) <br> Factor: Klf15; motif: RGGGMGGRGNNGGGGGNGG <br> 3.862e-05","TF:M01240_1 (11975) <br> Factor: BEN; motif: CAGCGRNV; match class: 1 <br> 5.607e-05","TF:M01858 (5104) <br> Factor: AP-2beta; motif: GCNNNGGSCNGVGGGN <br> 6.150e-05","TF:M00469_1 (2018) <br> Factor: AP-2alpha; motif: GCCNNNRGS; match class: 1 <br> 7.723e-05","TF:M00915 (7111) <br> Factor: AP-2; motif: SNNNCCNCAGGCN <br> 1.022e-04","TF:M01219 (6652) <br> Factor: SP1:SP3; motif: CCSCCCCCYCC <br> 1.085e-04","TF:M08041_1 (609) <br> Factor: Dlx1; motif: NNTAATTANN; match class: 1 <br> 1.819e-04","TF:M07289_1 (10917) <br> Factor: GKLF; motif: NNNRGGNGNGGSN; match class: 1 <br> 1.952e-04","TF:M00932_1 (4357) <br> Factor: Sp1; motif: NNGGGGCGGGGNN; match class: 1 <br> 2.235e-04","TF:M12438 (5956) <br> Factor: Wt1; motif: NNCCTCCCCCACNN <br> 2.537e-04","TF:M07436 (7691) <br> Factor: WT1; motif: NNGGGNGGGSGN <br> 2.553e-04","TF:M07354 (6088) <br> Factor: Egr-1; motif: GCGGGGGCGG <br> 2.566e-04","TF:M03876 (11107) <br> Factor: Kaiso; motif: GCMGGGRGCRGS <br> 2.625e-04","TF:M10375_1 (7376) <br> Factor: Sp1; motif: GGNGGGGGNGGGGGMGGGGCNGGG; match class: 1 <br> 3.625e-04","TF:M00189 (8023) <br> Factor: AP-2; motif: MKCCCSCNGGCG <br> 4.326e-04","TF:M01118 (8071) <br> Factor: WT1; motif: SMCNCCNSC <br> 4.659e-04","TF:M07329 (6136) <br> Factor: Osx; motif: CCNCCCCCNNN <br> 4.933e-04","TF:M10209_1 (10616) <br> Factor: E2F-1; motif: GNGGGCGGGRMN; match class: 1 <br> 7.713e-04","TF:M05494_1 (2791) <br> Factor: Egr-2; motif: GCGTGGGCGG; match class: 1 <br> 7.772e-04","TF:M01104 (7635) <br> Factor: MOVO-B; motif: GNGGGGG <br> 8.262e-04","TF:M01400_1 (1044) <br> Factor: Dlx-3; motif: NNNNNATAATTACMNNN; match class: 1 <br> 8.687e-04","TF:M00469 (3258) <br> Factor: AP-2alpha; motif: GCCNNNRGS <br> 8.754e-04","TF:M02036 (7515) <br> Factor: WT1; motif: CGCCCCCNCN <br> 8.864e-04","TF:M00931_1 (4177) <br> Factor: Sp1; motif: GGGGCGGGGC; match class: 1 <br> 9.452e-04","TF:M04157_1 (4728) <br> Factor: Tfap2a; motif: TGCCCYNRGGGCA; match class: 1 <br> 9.637e-04","TF:M00333_1 (11183) <br> Factor: ZF5; motif: NRNGNGCGCGCWN; match class: 1 <br> 1.010e-03","TF:M02835_1 (6572) <br> Factor: Zic1; motif: NNCCCCCGGGGGGG; match class: 1 <br> 1.031e-03","TF:M03814 (4958) <br> Factor: BTEB2; motif: GNAGGGGGNGGGSSNN <br> 1.074e-03","TF:M07436_1 (3983) <br> Factor: WT1; motif: NNGGGNGGGSGN; match class: 1 <br> 1.233e-03","TF:M04156 (7770) <br> Factor: Tfap2a; motif: NGCCYSAGGCN <br> 1.436e-03","TF:M05541 (6324) <br> Factor: Zfp536; motif: NCGGAKG <br> 1.779e-03","TF:M10282_1 (561) <br> Factor: Lhx2; motif: NNTAATKRGN; match class: 1 <br> 1.857e-03","TF:M02089 (11263) <br> Factor: E2F-3; motif: GGCGGGN <br> 1.888e-03","TF:M07040_1 (8412) <br> Factor: GKLF; motif: NNRRGRRNGNSNNN; match class: 1 <br> 1.987e-03","TF:M07397 (8048) <br> Factor: ZBP89; motif: CCCCKCCCCCNN <br> 2.276e-03","TF:M00800 (6721) <br> Factor: AP-2; motif: GSCCSCRGGCNRNRNN <br> 2.280e-03","TF:M00649_1 (3017) <br> Factor: MAZ; motif: GGGGAGGG; match class: 1 <br> 2.517e-03","TF:M07395_1 (4529) <br> Factor: Sp1; motif: NGGGGCGGGGN; match class: 1 <br> 2.683e-03","TF:M00933_1 (3986) <br> Factor: Sp1; motif: CCCCGCCCCN; match class: 1 <br> 3.380e-03","TF:M00716_1 (12044) <br> Factor: ZF5; motif: GSGCGCGR; match class: 1 <br> 3.571e-03","TF:M01233_1 (483) <br> Factor: ipf1; motif: MVCTAATTAS; match class: 1 <br> 3.640e-03","TF:M04445_1 (635) <br> Factor: Uncx; motif: YTAATTAN; match class: 1 <br> 4.228e-03","TF:M02844_1 (8517) <br> Factor: BCL6B; motif: NNNNCCGCCCCWNNNN; match class: 1 <br> 4.345e-03","TF:M04296 (1367) <br> Factor: ALX1; motif: NTAATYYAATTAN <br> 4.394e-03","TF:M05407 (6539) <br> Factor: Klf17; motif: NGGGCGG <br> 4.541e-03","TF:M12470 (9726) <br> Factor: Foxn1; motif: RGAMGC <br> 4.613e-03","TF:M07461_1 (3285) <br> Factor: KLF; motif: GGGNGGGG; match class: 1 <br> 4.887e-03","TF:M10363_1 (920) <br> Factor: DRRS; motif: GNNGGGWGGG; match class: 1 <br> 5.196e-03","TF:M07428_1 (802) <br> Factor: Six-3; motif: NWBTAATNYYWN; match class: 1 <br> 5.846e-03","TF:M02924 (5882) <br> Factor: TCFAP2B; motif: NTWGCCKSMGGCRNN <br> 6.586e-03","TF:M05494 (7229) <br> Factor: Egr-2; motif: GCGTGGGCGG <br> 6.633e-03","TF:M00196_1 (4242) <br> Factor: Sp1; motif: NGGGGGCGGGGYN; match class: 1 <br> 6.738e-03","TF:M10206_1 (993) <br> Factor: Dlx-5; motif: TAATTAYAGGK; match class: 1 <br> 7.255e-03","TF:M00333 (14398) <br> Factor: ZF5; motif: NRNGNGCGCGCWN <br> 7.321e-03","TF:M00915_1 (2409) <br> Factor: AP-2; motif: SNNNCCNCAGGCN; match class: 1 <br> 7.743e-03","TF:M04157 (8633) <br> Factor: Tfap2a; motif: TGCCCYNRGGGCA <br> 7.865e-03","TF:M01857 (6158) <br> Factor: AP-2alpha; motif: NGCCYSNNGSN <br> 8.100e-03","TF:M01047_1 (7237) <br> Factor: AP-2alphaA; motif: ANMGCCTNAGGCKNT; match class: 1 <br> 8.556e-03","TF:M01047 (7237) <br> Factor: AP-2alphaA; motif: ANMGCCTNAGGCKNT <br> 8.556e-03","TF:M01002 (933) <br> Factor: DEAF1; motif: RNNNRTTCGGGNRTTTCCGGRRNKN <br> 9.653e-03","TF:M02742 (3530) <br> Factor: E2F-2; motif: NNNANGGCGCGCNNN <br> 1.067e-02","TF:M02023_1 (4130) <br> Factor: MAZ; motif: NKGGGAGGGGRGGR; match class: 1 <br> 1.067e-02","TF:M05301 (7241) <br> Factor: CPBP; motif: NGGGCGG <br> 1.144e-02","TF:M05455 (7241) <br> Factor: SP6; motif: WGGGCGG <br> 1.144e-02","TF:M05391 (7241) <br> Factor: LKLF; motif: NGGGCGG <br> 1.144e-02","TF:M05665 (7241) <br> Factor: Sp3; motif: AGGGCGG <br> 1.144e-02","TF:M00444_1 (3791) <br> Factor: VDR; motif: GGGKNARNRRGGWSA; match class: 1 <br> 1.277e-02","TF:M00931 (8659) <br> Factor: Sp1; motif: GGGGCGGGGC <br> 1.277e-02","TF:M01199_1 (2726) <br> Factor: RNF96; motif: BCCCGCRGCC; match class: 1 <br> 1.305e-02","TF:M10412 (3797) <br> Factor: ZNF151; motif: GGSGGNGGGGGAGGGGMGG <br> 1.356e-02","TF:M01873 (4994) <br> Factor: Egr-1; motif: GCGGGGGCGG <br> 1.467e-02","TF:M07249_1 (1317) <br> Factor: ctcf; motif: CCNCNAGRKGGCRSTN; match class: 1 <br> 1.625e-02","TF:M01303 (7131) <br> Factor: SP1; motif: GGGGYGGGGNS <br> 1.635e-02","TF:M07348 (6188) <br> Factor: AP-2alpha; motif: NSCCNCRGGSN <br> 1.659e-02","TF:M07278_1 (638) <br> Factor: Dlx-3; motif: NTAATTNVN; match class: 1 <br> 1.796e-02","TF:M02819 (2602) <br> Factor: Tcfap2a; motif: WNNSCCYSRGGSRAN <br> 2.050e-02","TF:M04368 (656) <br> Factor: Hoxa2; motif: NNTAATTANN <br> 2.104e-02","TF:M12438_1 (1573) <br> Factor: Wt1; motif: NNCCTCCCCCACNN; match class: 1 <br> 2.470e-02","TF:M02835 (9478) <br> Factor: Zic1; motif: NNCCCCCGGGGGGG <br> 2.479e-02","TF:M00189_1 (3655) <br> Factor: AP-2; motif: MKCCCSCNGGCG; match class: 1 <br> 2.510e-02","TF:M07348_1 (1677) <br> Factor: AP-2alpha; motif: NSCCNCRGGSN; match class: 1 <br> 2.557e-02","TF:M05541_1 (1711) <br> Factor: Zfp536; motif: NCGGAKG; match class: 1 <br> 2.680e-02","TF:M00431 (5968) <br> Factor: E2F-1; motif: TTTSGCGS <br> 2.704e-02","TF:M00008 (8334) <br> Factor: Sp1; motif: GGGGCGGGGT <br> 2.762e-02","TF:M01857_1 (2198) <br> Factor: AP-2alpha; motif: NGCCYSNNGSN; match class: 1 <br> 2.762e-02","TF:M02831 (3933) <br> Factor: Zfp281; motif: NNCCCCCCCCCCMYC <br> 2.779e-02","TF:M01441_1 (1399) <br> Factor: OG-2; motif: CGNNYYAATTRRNNNNN; match class: 1 <br> 3.151e-02","TF:M03811_1 (2319) <br> Factor: AP-2gamma; motif: GCCYNCRGSN; match class: 1 <br> 3.292e-02","TF:M00470 (6995) <br> Factor: AP-2gamma; motif: GCCYNNGGS <br> 3.339e-02","TF:M00196 (8794) <br> Factor: Sp1; motif: NGGGGGCGGGGYN <br> 3.378e-02","TF:M02844 (13635) <br> Factor: BCL6B; motif: NNNNCCGCCCCWNNNN <br> 3.557e-02","TF:M02837 (3284) <br> Factor: Zic3; motif: NMCCCCCGGGGGGGN <br> 3.574e-02","TF:M07297 (3221) <br> Factor: MAZ; motif: CCCTCCCYCYN <br> 3.618e-02","TF:M05599 (5296) <br> Factor: WT1; motif: NGCGGGGGGGTSMMCYN <br> 3.834e-02","TF:M01837 (4957) <br> Factor: FKLF; motif: BGGGNGGVMD <br> 3.906e-02","TF:M05547_1 (2511) <br> Factor: ZAC; motif: KGGGCCR; match class: 1 <br> 4.056e-02","TF:M01721_1 (1837) <br> Factor: PUR1; motif: GGGNCAGNN; match class: 1 <br> 4.216e-02","TF:M03807_1 (2951) <br> Factor: SP2; motif: GNNGGGGGCGGGGSN; match class: 1 <br> 4.719e-02"],"key":["TF:M02933_1","TF:M00803_1","TF:M02933","TF:M08867","TF:M05547","TF:M02089_1","TF:M10375","TF:M00803","TF:M03876_1","TF:M08995_1","TF:M00470_1","TF:M10277","TF:M01240_1","TF:M01858","TF:M00469_1","TF:M00915","TF:M01219","TF:M08041_1","TF:M07289_1","TF:M00932_1","TF:M12438","TF:M07436","TF:M07354","TF:M03876","TF:M10375_1","TF:M00189","TF:M01118","TF:M07329","TF:M10209_1","TF:M05494_1","TF:M01104","TF:M01400_1","TF:M00469","TF:M02036","TF:M00931_1","TF:M04157_1","TF:M00333_1","TF:M02835_1","TF:M03814","TF:M07436_1","TF:M04156","TF:M05541","TF:M10282_1","TF:M02089","TF:M07040_1","TF:M07397","TF:M00800","TF:M00649_1","TF:M07395_1","TF:M00933_1","TF:M00716_1","TF:M01233_1","TF:M04445_1","TF:M02844_1","TF:M04296","TF:M05407","TF:M12470","TF:M07461_1","TF:M10363_1","TF:M07428_1","TF:M02924","TF:M05494","TF:M00196_1","TF:M10206_1","TF:M00333","TF:M00915_1","TF:M04157","TF:M01857","TF:M01047_1","TF:M01047","TF:M01002","TF:M02742","TF:M02023_1","TF:M05301","TF:M05455","TF:M05391","TF:M05665","TF:M00444_1","TF:M00931","TF:M01199_1","TF:M10412","TF:M01873","TF:M07249_1","TF:M01303","TF:M07348","TF:M07278_1","TF:M02819","TF:M04368","TF:M12438_1","TF:M02835","TF:M00189_1","TF:M07348_1","TF:M05541_1","TF:M00431","TF:M00008","TF:M01857_1","TF:M02831","TF:M01441_1","TF:M03811_1","TF:M00470","TF:M00196","TF:M02844","TF:M02837","TF:M07297","TF:M05599","TF:M01837","TF:M05547_1","TF:M01721_1","TF:M03807_1"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.80000000000000004,"size":[21.249765668339915,21.453293403509363,21.775964998974256,21.48377346610371,21.397780783048212,21.43072547410927,21.904038993437013,21.873246206592913,21.250516358609133,21.75653217882709,20.648295847158394,20.598437872681757,22.020505946544663,21.10316085614685,20.046345943226477,21.4655341998903,21.393219761864632,18.570619476587346,21.923247347626511,20.927545807169494,21.272786343893337,21.550133761804936,21.296739236995887,21.941428845327732,21.505062339763633,21.595566014832009,21.601969179982547,21.305312857100862,21.893747316577215,20.423316194151074,21.542265833362837,19.251971624778417,20.600184528531347,21.5251978647171,20.88041847159187,21.018460117273616,21.948609489469991,21.380075098237505,21.071085784534294,20.827139344700495,21.561130453472881,21.338220221243322,18.464065056701649,21.956112500322757,21.646327809158134,21.598906049405716,21.404422932642859,20.5125627594233,20.970680025422197,20.827984134175182,22.026529889943095,18.267746195110309,18.624586661685317,21.659603612629319,19.581708252717657,21.374603354778994,21.801048950015041,20.609567288316502,19.094857466267616,18.922448277443845,21.259110205994862,21.483325178888432,20.897681975063438,19.189929315806637,22.212724500992163,20.253288272315878,21.674070024496281,21.309218651796105,21.484520173179416,21.484520173179416,19.112372290916607,20.6911234245858,20.86775637239236,21.485117144835954,21.485117144835954,21.485117144835954,21.485117144835954,20.771613564159129,21.677284234148473,20.396211944030625,20.773393727796986,21.079087162392025,19.53653855160978,21.468571569045693,21.31452091461691,18.630658612830207,20.342533623705943,18.666450866177925,19.750715071460966,21.773612447612759,20.730437070696922,19.827209119463117,19.851115827647231,21.274987073775812,21.636351578251091,20.146508109192371,20.812958814613001,19.609692982930195,20.209017811302175,21.447736739297476,21.693810648318088,22.156135233005944,20.609221249021619,20.587190954108927,21.143887011719087,21.070862642541588,20.301369614619986,19.935464451182543,20.487258597622066],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[199.72237129981235,199.52551612813295,199.57346802892667],"y":[7.8684190517213723,3.5336874406332233,2.3336123623933656],"text":["WP:WP295 (92) <br> Electron transport chain <br> 1.354e-08","WP:WP1248 (54) <br> Oxidative phosphorylation <br> 2.926e-04","WP:WP1272 (22) <br> Selenium micronutrient network <br> 4.639e-03"],"key":["WP:WP295","WP:WP1248","WP:WP1272"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,153,198,1)","opacity":0.80000000000000004,"size":[15.882025628956763,15.008384021416694,13.357372072457588],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,153,198,1)"}},"hoveron":"points","set":"SharedDatab5111929","name":"WP","legendgroup":"WP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[2,30.122117653027573],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[32.633469470586746,42.82453514624185],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[45.335886963801016,113.60698612414703],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[116.1183379417062,117.54980847771492],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[120.06116029527409,124.42588975419193],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[126.9372415717511,137.72852033180286],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[140.23987214936201,145.36554120900027],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[147.87689302655946,149.58210091068213],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[152.09345272824129,196.98135511529389],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[199.49270693285305,200.00251135181756],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.199999999999999],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.559055118110237,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.75590551181102372,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.283520132835207,"r":6.6417600664176026,"b":55.878166138820198,"l":61.037775010377771},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"tickvals":[16.061058826513786,37.729002308414294,79.471436543974022,116.83407320971057,122.243525024733,132.33288095177699,142.80270667918114,148.72949696862079,174.53740392176758,199.74760914233531],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.393109173931101},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"d3b2aa66cdc":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"d3b23434e4fa":{"x":{},"xend":{},"y":{},"yend":{}},"d3b27ca4f3dc":{"x":{},"xend":{},"y":{},"yend":{}},"d3b231863072":{"x":{},"xend":{},"y":{},"yend":{}},"d3b260d6a5c4":{"x":{},"xend":{},"y":{},"yend":{}},"d3b22c1d1887":{"x":{},"xend":{},"y":{},"yend":{}},"d3b22a355db1":{"x":{},"xend":{},"y":{},"yend":{}},"d3b2159a251d":{"x":{},"xend":{},"y":{},"yend":{}},"d3b23afa9dff":{"x":{},"xend":{},"y":{},"yend":{}},"d3b21b9aee99":{"x":{},"xend":{},"y":{},"yend":{}},"d3b258a68cf7":{"x":{},"xend":{},"y":{},"yend":{}},"d3b21e70d999":{"x":{},"y":{}},"d3b22dddb6c":{"yintercept":{}}},"cur_data":"d3b2aa66cdc","visdat":{"d3b2aa66cdc":["function (y) ","x"],"d3b23434e4fa":["function (y) ","x"],"d3b27ca4f3dc":["function (y) ","x"],"d3b231863072":["function (y) ","x"],"d3b260d6a5c4":["function (y) ","x"],"d3b22c1d1887":["function (y) ","x"],"d3b22a355db1":["function (y) ","x"],"d3b2159a251d":["function (y) ","x"],"d3b23afa9dff":["function (y) ","x"],"d3b21b9aee99":["function (y) ","x"],"d3b258a68cf7":["function (y) ","x"],"d3b21e70d999":["function (y) ","x"],"d3b22dddb6c":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDatab5111929"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Modify the <code>go_enrich</code> data frame to only include relevant columns:</p>
<ul>
<li><p><code>query</code>: query name/identifier. A query is the set of genes you are interested in investigating.</p></li>
<li><p><code>source</code>: source of the GO term (e.g.&nbsp;GO, KEGG).</p></li>
<li><p><code>term_id</code>: ID of the GO term.</p></li>
<li><p><code>term_name</code>: name/description of the GO term.</p></li>
<li><p><code>p_value</code>: p-value for the enrichment of the GO term.</p></li>
<li><p><code>query_size</code>: number of genes in the query list.</p></li>
<li><p><code>intersection_size</code>: number of genes in the query list that overlap with the GO term.</p></li>
<li><p><code>term_size</code>: total number of genes associated with the GO term.</p></li>
<li><p><code>effective_domain_size</code>: the size of the background gene set used for the enrichment analysis. This adjusts for the fact that some GO terms are more likely to be enriched simply because they are associated with a larger number of genes in the genome.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>gp_mod <span class="ot">=</span> go_enrich<span class="sc">$</span>result[,<span class="fu">c</span>(<span class="st">"query"</span>, <span class="st">"source"</span>, <span class="st">"term_id"</span>,</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"term_name"</span>, <span class="st">"p_value"</span>, <span class="st">"query_size"</span>, </span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"intersection_size"</span>, <span class="st">"term_size"</span>, </span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"effective_domain_size"</span>)]</span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(gp_mod))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 6%">
<col style="width: 5%">
<col style="width: 8%">
<col style="width: 27%">
<col style="width: 7%">
<col style="width: 8%">
<col style="width: 13%">
<col style="width: 7%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">query</th>
<th style="text-align: left;">source</th>
<th style="text-align: left;">term_id</th>
<th style="text-align: left;">term_name</th>
<th style="text-align: right;">p_value</th>
<th style="text-align: right;">query_size</th>
<th style="text-align: right;">intersection_size</th>
<th style="text-align: right;">term_size</th>
<th style="text-align: right;">effective_domain_size</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">CORUM</td>
<td style="text-align: left;">CORUM:538</td>
<td style="text-align: left;">Cytochrome c oxidase, mitochondrial</td>
<td style="text-align: right;">0.0295358</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">1082</td>
</tr>
<tr class="even">
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">anatomical structure development</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">166</td>
<td style="text-align: right;">6269</td>
<td style="text-align: right;">26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">system development</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">132</td>
<td style="text-align: right;">4117</td>
<td style="text-align: right;">26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">developmental process</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">6854</td>
<td style="text-align: right;">26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">anatomical structure morphogenesis</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">292</td>
<td style="text-align: right;">107</td>
<td style="text-align: right;">2761</td>
<td style="text-align: right;">26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048514</td>
<td style="text-align: left;">blood vessel morphogenesis</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">258</td>
<td style="text-align: right;">53</td>
<td style="text-align: right;">657</td>
<td style="text-align: right;">26944</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Calculate gene and background ratios to assess the proportion of genes involved in a term relative to the query and background.</p>
<ul>
<li><p><strong>GeneRatio</strong>: the ratio of the number of query genes that intersect with the GO term to the total number of genes in the query list. It helps to understand what proportion of the entire query list is associated with a specific GO term.</p></li>
<li><p><strong>BgRatio</strong>: the ratio of the number of genes associated with the GO term to the effective domain size. This helps to understand the proportion of background genes associated with the GO term, offering insight into its general relevance.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>gp_mod<span class="sc">$</span>GeneRatio <span class="ot">=</span> <span class="fu">paste0</span>(gp_mod<span class="sc">$</span>intersection_size,  <span class="st">"/"</span>, gp_mod<span class="sc">$</span>query_size)</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>gp_mod<span class="sc">$</span>BgRatio <span class="ot">=</span> <span class="fu">paste0</span>(gp_mod<span class="sc">$</span>term_size, <span class="st">"/"</span>, gp_mod<span class="sc">$</span>effective_domain_size)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Rename columns.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gp_mod) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Cluster"</span>, <span class="st">"Category"</span>, <span class="st">"ID"</span>, <span class="st">"Description"</span>, <span class="st">"p.adjust"</span>, </span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"query_size"</span>, <span class="st">"Count"</span>, <span class="st">"term_size"</span>, <span class="st">"effective_domain_size"</span>, </span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"GeneRatio"</span>, <span class="st">"BgRatio"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Set the row names of the <code>gp_mod</code> data frame to the GO term IDs.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(gp_mod) <span class="ot">=</span> gp_mod<span class="sc">$</span>ID</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(gp_mod))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 23%">
<col style="width: 6%">
<col style="width: 7%">
<col style="width: 3%">
<col style="width: 6%">
<col style="width: 14%">
<col style="width: 6%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Cluster</th>
<th style="text-align: left;">Category</th>
<th style="text-align: left;">ID</th>
<th style="text-align: left;">Description</th>
<th style="text-align: right;">p.adjust</th>
<th style="text-align: right;">query_size</th>
<th style="text-align: right;">Count</th>
<th style="text-align: right;">term_size</th>
<th style="text-align: right;">effective_domain_size</th>
<th style="text-align: left;">GeneRatio</th>
<th style="text-align: left;">BgRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">CORUM:538</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">CORUM</td>
<td style="text-align: left;">CORUM:538</td>
<td style="text-align: left;">Cytochrome c oxidase, mitochondrial</td>
<td style="text-align: right;">0.0295358</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">12</td>
<td style="text-align: right;">1082</td>
<td style="text-align: left;">3/19</td>
<td style="text-align: left;">12/1082</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">anatomical structure development</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">166</td>
<td style="text-align: right;">6269</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">166/288</td>
<td style="text-align: left;">6269/26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">system development</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">132</td>
<td style="text-align: right;">4117</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">132/288</td>
<td style="text-align: left;">4117/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">developmental process</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">6854</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">172/288</td>
<td style="text-align: left;">6854/26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">anatomical structure morphogenesis</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">292</td>
<td style="text-align: right;">107</td>
<td style="text-align: right;">2761</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">107/292</td>
<td style="text-align: left;">2761/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0048514</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048514</td>
<td style="text-align: left;">blood vessel morphogenesis</td>
<td style="text-align: right;">0.0000000</td>
<td style="text-align: right;">258</td>
<td style="text-align: right;">53</td>
<td style="text-align: right;">657</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">53/258</td>
<td style="text-align: left;">657/26944</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The <code>result</code> data frame, which contains the enrichment analysis, is ordered first by query name, data source (such as GO:BP, GO:CC, etc.), and then by the adjusted p-value.</p>
<p>Filter for the GO:BP data source (to look for biological processes related to stress responses).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>gp_mod_bp <span class="ot">=</span> gp_mod <span class="sc">%&gt;%</span></span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category <span class="sc">==</span> <span class="st">"GO:BP"</span>)</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(gp_mod_bp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 22%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 3%">
<col style="width: 6%">
<col style="width: 14%">
<col style="width: 6%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Cluster</th>
<th style="text-align: left;">Category</th>
<th style="text-align: left;">ID</th>
<th style="text-align: left;">Description</th>
<th style="text-align: right;">p.adjust</th>
<th style="text-align: right;">query_size</th>
<th style="text-align: right;">Count</th>
<th style="text-align: right;">term_size</th>
<th style="text-align: right;">effective_domain_size</th>
<th style="text-align: left;">GeneRatio</th>
<th style="text-align: left;">BgRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">anatomical structure development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">166</td>
<td style="text-align: right;">6269</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">166/288</td>
<td style="text-align: left;">6269/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">system development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">132</td>
<td style="text-align: right;">4117</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">132/288</td>
<td style="text-align: left;">4117/26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">developmental process</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">172</td>
<td style="text-align: right;">6854</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">172/288</td>
<td style="text-align: left;">6854/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">anatomical structure morphogenesis</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">292</td>
<td style="text-align: right;">107</td>
<td style="text-align: right;">2761</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">107/292</td>
<td style="text-align: left;">2761/26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048514</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048514</td>
<td style="text-align: left;">blood vessel morphogenesis</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">258</td>
<td style="text-align: right;">53</td>
<td style="text-align: right;">657</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">53/258</td>
<td style="text-align: left;">657/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0007275</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0007275</td>
<td style="text-align: left;">multicellular organism development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">288</td>
<td style="text-align: right;">141</td>
<td style="text-align: right;">4879</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">141/288</td>
<td style="text-align: left;">4879/26944</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Create an <code>enrichResult</code> object from the modified data frame. This class is used by the <code>enrichplot</code> package to generate plots.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define as enrichResult object</span></span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>gp_mod_bp_enrich  <span class="ot">=</span> <span class="fu">new</span>(<span class="st">"enrichResult"</span>, <span class="at">result =</span> gp_mod_bp)</span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate a dotplot</span></span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a><span class="co"># The order of the y-axis is determined by the orderBy= argument</span></span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Default for orderBy="x" (in this case, "GeneRatio")</span></span>
<span id="cb65-6"><a href="#cb65-6" aria-hidden="true" tabindex="-1"></a>dotplot_top300 <span class="ot">=</span> enrichplot<span class="sc">::</span><span class="fu">dotplot</span>(gp_mod_bp_enrich) <span class="sc">+</span></span>
<span id="cb65-7"><a href="#cb65-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Top 300 expressed genes"</span>, <span class="at">subtitle =</span> <span class="st">"Gene Ontology:Biological Process"</span>) <span class="sc">+</span> </span>
<span id="cb65-8"><a href="#cb65-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>))</span>
<span id="cb65-9"><a href="#cb65-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb65-10"><a href="#cb65-10" aria-hidden="true" tabindex="-1"></a>dotplot_top300</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>This plot shows the 10 most enriched GO terms (ordered by <code>GeneRatio</code>) in genes with the highest expression levels across all the samples.</p>
<p>The GO terms largely relate to normal developmental processes and anatomical structure formation.</p>
<p>Anatomical structure morphogenesis, vasculature development, and blood vessel development are related to normal EC function.</p>
</section>
<section id="explorting-the-outputs-of-dea" class="level2">
<h2 class="anchored" data-anchor-id="explorting-the-outputs-of-dea">Explorting the outputs of DEA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the results of the DEA from dds</span></span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">=</span> <span class="fu">results</span>(dds)</span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition LV vs BR 
Wald test p-value: condition LV vs BR 
DataFrame with 6 rows and 6 columns
                    baseMean log2FoldChange     lfcSE      stat      pvalue
                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
ENSMUSG00000000001  807.4914     -0.0965592  0.507050 -0.190433 0.848969778
ENSMUSG00000000028   14.9649     -0.5079879  2.019062 -0.251596 0.801353310
ENSMUSG00000000049   18.0004      8.6969814  2.299635  3.781896 0.000155638
ENSMUSG00000000056  128.7510      0.8479185  1.118891  0.757821 0.448558366
ENSMUSG00000000058 1541.3723     -0.2564915  0.363457 -0.705701 0.480374291
ENSMUSG00000000078 3540.1042     -1.8111155  0.492120 -3.680230 0.000233024
                         padj
                    &lt;numeric&gt;
ENSMUSG00000000001 0.94797771
ENSMUSG00000000028 0.92256791
ENSMUSG00000000049 0.00142440
ENSMUSG00000000056 0.68954221
ENSMUSG00000000058 0.71748582
ENSMUSG00000000078 0.00202678</code></pre>
</div>
</div>
<p>The default output columns are:</p>
<ul>
<li><p><strong>log2FoldChange</strong>: log2 fold change of expression for each gene between conditions.</p>
<ul>
<li><p>Fold change values are represented in log2 scale to symmetrically represent both upregulation and downregulation of gene expression. For example, a fold change of 2 (upregulation) and a fold change of -0.5 (downregulation) are symetrically represented as +1 and -1 on a log2 scale.</p></li>
<li><p>As raw fold change values can vary widely, using a log2 scale also compresses this range.</p></li>
</ul></li>
<li><p><strong>lfcSE</strong>: standard error of the log2 fold change.</p></li>
<li><p><strong>stat</strong>: test statistics (Wald test or likelihood ratio test) for each gene.</p></li>
<li><p><strong>pvalue</strong>: p-value for test of differential expression.</p></li>
<li><p><strong>padj</strong>: the adjusted p-value (using the Benjamini-Hochberg procedure) to control for false discovery rate.</p></li>
</ul>
<p>List the names of the coefficients (i.e.&nbsp;terms in the design formula) that are available in <code>dds</code> for which you can extract results. This is useful for understanding which comparisons are available in the model.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">resultsNames</span>(dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Intercept"          "condition_LG_vs_BR" "condition_LV_vs_BR"</code></pre>
</div>
</div>
<section id="log-fold-shrinkage" class="level3">
<h3 class="anchored" data-anchor-id="log-fold-shrinkage">Log-fold shrinkage</h3>
<p>Shrinking log2 fold changes is used to reduce noise in low-count genes.</p>
<ul>
<li><p>Genes with low counts often show exaggerated or highly variable log2 fold changes (i.e.&nbsp;1 vs 4 counts is a 4x fold change).</p></li>
<li><p>Shrinking the log2fold changes helps reduce this noise by pulling extreme values towards more conservative or â€œshrunkâ€ estimates, reflecting more accurate differences in gene expression.</p></li>
</ul>
<p>Extract the results of the DEA for lung compared to brain.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>resLGBR <span class="ot">=</span> <span class="fu">results</span>(dds, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">"condition"</span>, <span class="st">"LG"</span>, <span class="st">"BR"</span>))</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(resLGBR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition LG vs BR 
Wald test p-value: condition LG vs BR 
DataFrame with 6 rows and 6 columns
                    baseMean log2FoldChange     lfcSE       stat    pvalue
                   &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;
ENSMUSG00000000001  807.4914     -0.0249977  0.508523 -0.0491574 0.9607939
ENSMUSG00000000028   14.9649      1.9215833  2.022589  0.9500614 0.3420811
ENSMUSG00000000049   18.0004      0.0000000  2.582076  0.0000000 1.0000000
ENSMUSG00000000056  128.7510     -1.5845830  1.130799 -1.4012946 0.1611260
ENSMUSG00000000058 1541.3723      1.1950509  0.363832  3.2846211 0.0010212
ENSMUSG00000000078 3540.1042     -0.7840804  0.492355 -1.5925119 0.1112697
                         padj
                    &lt;numeric&gt;
ENSMUSG00000000001 0.99118033
ENSMUSG00000000028 0.59634897
ENSMUSG00000000049 1.00000000
ENSMUSG00000000056 0.38283399
ENSMUSG00000000058 0.00882073
ENSMUSG00000000078 0.30247313</code></pre>
</div>
</div>
<p>Visualise the relationship between log2 fold changes, significant DE genes, and the gene mean count.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLGBR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We can see that genes with low counts tend to have exaggerated fold changes and high levels of variance. However, since these counts are very low, this is probably noise.</p>
<p>Shrinking the log2 fold changes will â€œshrinkâ€ the fold changes of genes with low counts (i.e.&nbsp;genes towards the left of the x-axis).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>resLGBR_Lfc <span class="ot">=</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef =</span> <span class="st">"condition_LG_vs_BR"</span>, <span class="at">res =</span> resLGBR)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plotMA</span>(resLGBR_Lfc)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-46-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="visualising-differentially-expressed-genes-between-conditions" class="level3">
<h3 class="anchored" data-anchor-id="visualising-differentially-expressed-genes-between-conditions">Visualising differentially expressed genes between conditions</h3>
<section id="heatmap" class="level4">
<h4 class="anchored" data-anchor-id="heatmap">Heatmap</h4>
<p>A heatmap can reveal expression patterns across sample groups (columns) and automatically ordered genes (rows) according to their similarity.</p>
<p>Use the transformed data.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get top 50 differentially expressed genes from res</span></span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">=</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj), ] <span class="sc">|&gt;</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>(<span class="dv">50</span>) <span class="sc">|&gt;</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>()</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract data for heatmap</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>heatmapdata <span class="ot">=</span> <span class="fu">assay</span>(vsd)[genes, ]</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a gene ID to gene name mapping</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>id_to_name <span class="ot">=</span> <span class="fu">setNames</span>(tx2gene<span class="sc">$</span>V3, tx2gene<span class="sc">$</span>V2)</span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace gene IDs with gene names</span></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(heatmapdata) <span class="ot">=</span> id_to_name[<span class="fu">rownames</span>(heatmapdata)]</span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale transformed counts for visualisation</span></span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Scale each row of the matrix so that each row has a mean of 0 and a standard deviation of 1</span></span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a><span class="co"># This helps to ensure that the data is centered and scaled for each gene, making it easier to visualise patterns in the heatmap</span></span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>heatmapdata <span class="ot">=</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(heatmapdata)))</span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract condition data</span></span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a>heatmapColAnnot <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">colData</span>(vsd)[, <span class="st">"condition"</span>])</span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename column title</span></span>
<span id="cb76-22"><a href="#cb76-22" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(heatmapColAnnot) <span class="ot">=</span> <span class="st">"Condition"</span></span>
<span id="cb76-23"><a href="#cb76-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Create annotation object for use in the heatmap</span></span>
<span id="cb76-24"><a href="#cb76-24" aria-hidden="true" tabindex="-1"></a>heatmapColAnnot <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df =</span> heatmapColAnnot)</span>
<span id="cb76-25"><a href="#cb76-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-26"><a href="#cb76-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot heatmap</span></span>
<span id="cb76-27"><a href="#cb76-27" aria-hidden="true" tabindex="-1"></a>ComplexHeatmap<span class="sc">::</span><span class="fu">Heatmap</span>(heatmapdata,</span>
<span id="cb76-28"><a href="#cb76-28" aria-hidden="true" tabindex="-1"></a>                        <span class="at">top_annotation =</span> heatmapColAnnot,</span>
<span id="cb76-29"><a href="#cb76-29" aria-hidden="true" tabindex="-1"></a>                        <span class="at">name =</span> <span class="st">"Expression"</span>,</span>
<span id="cb76-30"><a href="#cb76-30" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb76-31"><a href="#cb76-31" aria-hidden="true" tabindex="-1"></a>                        <span class="at">row_names_gp =</span> grid<span class="sc">::</span><span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">10</span>),</span>
<span id="cb76-32"><a href="#cb76-32" aria-hidden="true" tabindex="-1"></a>                        <span class="at">column_title =</span> <span class="st">"Top 50 DE genes"</span>,</span>
<span id="cb76-33"><a href="#cb76-33" aria-hidden="true" tabindex="-1"></a>                        <span class="at">column_title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize=</span><span class="dv">15</span>, <span class="at">fontface=</span><span class="st">"bold"</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-47-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="volcano-plot" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot">Volcano plot</h4>
<p>We want to look for genes that are overexpressed in the brain compared to both lung and liver. Here, this will be represented by genes that have a negative <code>log2FoldChange</code> in the comparisons named above.</p>
<p>What kind of threshold should we use for <code>log2FoldChange</code> and <code>padj</code>? Use a volcano plot to help decide.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set thresholds</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>foldchange<span class="ot">=</span><span class="dv">2</span> <span class="co"># Threshold for log2 fold change</span></span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>pval<span class="ot">=</span><span class="fl">0.001</span> <span class="co"># Threshold for adjusted p-value</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>A volcano plot is a common way to visualise DEA data. It shows the relationship between the magnitude of change (log2 fold change) and statistical significance (adjusted p-value).</p>
<ul>
<li><p>The adjusted p-value is plotted on the volcano plot as <code>-log10(padj)</code>. The raw p-values are often very small for significant results, and visualising them on a log scale compresses a wide range of p-values into a more manageable range where highly significant results are visually distinct.</p></li>
<li><p>Lower p-values (high significance) are also emphasised as larger values on the y-axis.</p></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create function that generates a volcano plot that takes a coefficient of dds as an argument</span></span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>PlotVolcano <span class="ot">&lt;-</span><span class="cf">function</span>(comparison){</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Shrink the log2 fold changes using the apeglm method to stabilise the estimates,       especially for genes with low counts or low variance</span></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The output, de, is a DESeq2 object that contains shruken log2 fold changes</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>  de <span class="ot">=</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span>comparison, <span class="at">type=</span><span class="st">"apeglm"</span>)</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Convert de into a data frame, resul</span></span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>  resul <span class="ot">=</span> <span class="fu">as.data.frame</span>(de)</span>
<span id="cb78-8"><a href="#cb78-8" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract gene IDs and assign them to resul as a new column</span></span>
<span id="cb78-9"><a href="#cb78-9" aria-hidden="true" tabindex="-1"></a>  resul<span class="sc">$</span>geneID <span class="ot">&lt;-</span> <span class="fu">rownames</span>(resul)</span>
<span id="cb78-10"><a href="#cb78-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Merge resul with additional gene annotations (gene names) stored in tx2gene</span></span>
<span id="cb78-11"><a href="#cb78-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Column "geneID" in res corresponds to column "V2" in tx2gene</span></span>
<span id="cb78-12"><a href="#cb78-12" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Keep all rows in resul</span></span>
<span id="cb78-13"><a href="#cb78-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Only matching rows in tx2gene are kept</span></span>
<span id="cb78-14"><a href="#cb78-14" aria-hidden="true" tabindex="-1"></a>  de <span class="ot">&lt;-</span> <span class="fu">merge</span>(resul, tx2gene, <span class="at">by.x =</span> <span class="st">"geneID"</span>, <span class="at">by.y =</span> <span class="st">"V2"</span>, <span class="at">all.x =</span> <span class="cn">TRUE</span>, <span class="at">all.y =</span> <span class="cn">FALSE</span>)</span>
<span id="cb78-15"><a href="#cb78-15" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-16"><a href="#cb78-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filtering and selection</span></span>
<span id="cb78-17"><a href="#cb78-17" aria-hidden="true" tabindex="-1"></a>  de <span class="ot">=</span> de <span class="sc">%&gt;%</span> </span>
<span id="cb78-18"><a href="#cb78-18" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(log2FoldChange)) <span class="sc">%&gt;%</span> <span class="co"># Remove any rows with NA values in the log2FoldChange column</span></span>
<span id="cb78-19"><a href="#cb78-19" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>V1) <span class="sc">%&gt;%</span> <span class="co"># Remove column "V1" (transcript ID)</span></span>
<span id="cb78-20"><a href="#cb78-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="co"># Remove duplicate rows</span></span>
<span id="cb78-21"><a href="#cb78-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-22"><a href="#cb78-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Count how many genes are in each category (unchanged, upregulated, downregulated)</span></span>
<span id="cb78-23"><a href="#cb78-23" aria-hidden="true" tabindex="-1"></a>  n_unchanged <span class="ot">&lt;-</span> de <span class="sc">%&gt;%</span> </span>
<span id="cb78-24"><a href="#cb78-24" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Filter for genes where log2FoldChange is between -foldchange and +foldchange</span></span>
<span id="cb78-25"><a href="#cb78-25" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(log2FoldChange <span class="sc">&lt;</span> foldchange <span class="sc">&amp;</span> log2FoldChange <span class="sc">&gt;</span> <span class="sc">-</span>(foldchange)) <span class="sc">%&gt;%</span> </span>
<span id="cb78-26"><a href="#cb78-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Count how many genes unchanged</span></span>
<span id="cb78-27"><a href="#cb78-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">nrow</span>()</span>
<span id="cb78-28"><a href="#cb78-28" aria-hidden="true" tabindex="-1"></a>  n_up <span class="ot">&lt;-</span> de <span class="sc">%&gt;%</span></span>
<span id="cb78-29"><a href="#cb78-29" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Filter for genes where log2FoldChange is greater than foldchange and significant (based on pval)</span></span>
<span id="cb78-30"><a href="#cb78-30" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(log2FoldChange <span class="sc">&gt;=</span> foldchange <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> pval) <span class="sc">%&gt;%</span></span>
<span id="cb78-31"><a href="#cb78-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Count how many genes upregulated</span></span>
<span id="cb78-32"><a href="#cb78-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">nrow</span>()</span>
<span id="cb78-33"><a href="#cb78-33" aria-hidden="true" tabindex="-1"></a>  n_down <span class="ot">&lt;-</span> de <span class="sc">%&gt;%</span> </span>
<span id="cb78-34"><a href="#cb78-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Filter for genes where log2FoldChange is less than -foldchange and significant (based on pval)</span></span>
<span id="cb78-35"><a href="#cb78-35" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">filter</span>(log2FoldChange <span class="sc">&lt;=</span> <span class="sc">-</span>(foldchange) <span class="sc">&amp;</span> padj <span class="sc">&lt;</span> pval) <span class="sc">%&gt;%</span> </span>
<span id="cb78-36"><a href="#cb78-36" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Count how many genes downregulated</span></span>
<span id="cb78-37"><a href="#cb78-37" aria-hidden="true" tabindex="-1"></a>    <span class="fu">nrow</span>()</span>
<span id="cb78-38"><a href="#cb78-38" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-39"><a href="#cb78-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Assign DE labels to genes</span></span>
<span id="cb78-40"><a href="#cb78-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create "Expression" column and label all genes as "Unchanged" with total count of unchanged genes</span></span>
<span id="cb78-41"><a href="#cb78-41" aria-hidden="true" tabindex="-1"></a>  de<span class="sc">$</span>Expression <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Unchanged ("</span>,n_unchanged,<span class="st">")"</span>)</span>
<span id="cb78-42"><a href="#cb78-42" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For genes where log2FoldChange is greater than foldchange and significant, the label is updated to "Up" with a total count of upregulated genes</span></span>
<span id="cb78-43"><a href="#cb78-43" aria-hidden="true" tabindex="-1"></a>  de<span class="sc">$</span>Expression[de<span class="sc">$</span>log2FoldChange <span class="sc">&gt;=</span> foldchange <span class="sc">&amp;</span> de<span class="sc">$</span>padj <span class="sc">&lt;</span> pval] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Up ("</span>,n_up,<span class="st">")"</span>)</span>
<span id="cb78-44"><a href="#cb78-44" aria-hidden="true" tabindex="-1"></a>  <span class="co"># For genes where log2FoldChange is less than -foldchange and significant, the label is updated to "Down" with a total count of downregulated genes</span></span>
<span id="cb78-45"><a href="#cb78-45" aria-hidden="true" tabindex="-1"></a>  de<span class="sc">$</span>Expression[de<span class="sc">$</span>log2FoldChange <span class="sc">&lt;=</span> <span class="sc">-</span>(foldchange) <span class="sc">&amp;</span> de<span class="sc">$</span>padj <span class="sc">&lt;</span> pval] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"Down ("</span>,n_down,<span class="st">")"</span>)</span>
<span id="cb78-46"><a href="#cb78-46" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-47"><a href="#cb78-47" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set colours for plot - the order of colours will correspond to the order of levels in the factor variable de$Expression</span></span>
<span id="cb78-48"><a href="#cb78-48" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (n_up <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> n_down <span class="sc">==</span> <span class="dv">0</span>){ <span class="co"># If there are no up or downregulated genes</span></span>
<span id="cb78-49"><a href="#cb78-49" aria-hidden="true" tabindex="-1"></a>      cvec <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#84A1AB"</span>) <span class="co"># Use grey-blue for all genes</span></span>
<span id="cb78-50"><a href="#cb78-50" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (n_up <span class="sc">==</span> <span class="dv">0</span>){ <span class="co"># If there are only downregulated genes</span></span>
<span id="cb78-51"><a href="#cb78-51" aria-hidden="true" tabindex="-1"></a>      cvec <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#B02302"</span>, <span class="st">"#84A1AB"</span>) <span class="co"># Use red for down genes and grey-blue for unchanged genes</span></span>
<span id="cb78-52"><a href="#cb78-52" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> <span class="cf">if</span> (n_down <span class="sc">==</span> <span class="dv">0</span>){ <span class="co"># If there are only upregulated genes</span></span>
<span id="cb78-53"><a href="#cb78-53" aria-hidden="true" tabindex="-1"></a>      cvec <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#84A1AB"</span>, <span class="st">"#61B002"</span>) <span class="co"># Use grey-blue for unchanged genes and green for up genes</span></span>
<span id="cb78-54"><a href="#cb78-54" aria-hidden="true" tabindex="-1"></a>  } <span class="cf">else</span> { <span class="co"># If there are downregulated and upregulated genes</span></span>
<span id="cb78-55"><a href="#cb78-55" aria-hidden="true" tabindex="-1"></a>      cvec <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"#B02302"</span>, <span class="st">"#84A1AB"</span>, <span class="st">"#61B002"</span>) <span class="co"># Use red for down genes, and grey-blue for unchanged genes, and green for up genes</span></span>
<span id="cb78-56"><a href="#cb78-56" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb78-57"><a href="#cb78-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-58"><a href="#cb78-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add gene names to genes that are differentially expressed</span></span>
<span id="cb78-59"><a href="#cb78-59" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Create "delabel" column and initialise with NA values for all rows (by default, no genes are labelled on the plot)</span></span>
<span id="cb78-60"><a href="#cb78-60" aria-hidden="true" tabindex="-1"></a>  de<span class="sc">$</span>delabel <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb78-61"><a href="#cb78-61" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select "delabel" entries where "Expression" does not contain "Unchanged"</span></span>
<span id="cb78-62"><a href="#cb78-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Select and assign gene names (from "V3") where "Expression" does not contain "Unchanged"</span></span>
<span id="cb78-63"><a href="#cb78-63" aria-hidden="true" tabindex="-1"></a>  de<span class="sc">$</span>delabel[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Unchanged"</span>, de<span class="sc">$</span>Expression)] <span class="ot">&lt;-</span> de<span class="sc">$</span>V3[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"Unchanged"</span>, de<span class="sc">$</span>Expression)]</span>
<span id="cb78-64"><a href="#cb78-64" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort the data frame by adjusted p-values</span></span>
<span id="cb78-65"><a href="#cb78-65" aria-hidden="true" tabindex="-1"></a>  de<span class="ot">=</span>de[<span class="fu">order</span>(de<span class="sc">$</span>padj),]</span>
<span id="cb78-66"><a href="#cb78-66" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-67"><a href="#cb78-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb78-68"><a href="#cb78-68" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Generate the volcano plot</span></span>
<span id="cb78-69"><a href="#cb78-69" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The x-axis is mapped to the log2 fold change values (i.e. magnitude of change)</span></span>
<span id="cb78-70"><a href="#cb78-70" aria-hidden="true" tabindex="-1"></a>  <span class="co"># The y-axis is mapped to the -log10  of the padj values (i.e. significance)</span></span>
<span id="cb78-71"><a href="#cb78-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="at">data=</span>de, <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(padj))) <span class="sc">+</span></span>
<span id="cb78-72"><a href="#cb78-72" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Add vertical dashed lines at the foldchange thresholds</span></span>
<span id="cb78-73"><a href="#cb78-73" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="fu">c</span>(<span class="sc">-</span>(foldchange), foldchange), <span class="at">col=</span><span class="st">"light grey"</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb78-74"><a href="#cb78-74" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Add horizontal dashed line at the pval threshold</span></span>
<span id="cb78-75"><a href="#cb78-75" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="sc">-</span><span class="fu">log10</span>(pval), <span class="at">col=</span><span class="st">"light grey"</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb78-76"><a href="#cb78-76" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Add points to the plot, coloured by "Expression" column</span></span>
<span id="cb78-77"><a href="#cb78-77" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color=</span>Expression), <span class="at">alpha=</span><span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb78-78"><a href="#cb78-78" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Add text lables (avoiding overlap) to the top 50 most signficant DE genes</span></span>
<span id="cb78-79"><a href="#cb78-79" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Set x- and y-coordinates for labels using "log2FoldChange" and "padj" columns</span></span>
<span id="cb78-80"><a href="#cb78-80" aria-hidden="true" tabindex="-1"></a>          <span class="fu">geom_text_repel</span>(<span class="at">data=</span>de[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>,],<span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj),<span class="at">label=</span>delabel),<span class="at">max.overlaps=</span><span class="dv">50</span>,<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb78-81"><a href="#cb78-81" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Use colours specified earlier</span></span>
<span id="cb78-82"><a href="#cb78-82" aria-hidden="true" tabindex="-1"></a>          <span class="fu">scale_color_manual</span>(<span class="at">values=</span>cvec) <span class="sc">+</span></span>
<span id="cb78-83"><a href="#cb78-83" aria-hidden="true" tabindex="-1"></a>          <span class="co"># Set appearance</span></span>
<span id="cb78-84"><a href="#cb78-84" aria-hidden="true" tabindex="-1"></a>          <span class="fu">theme_few</span>() <span class="sc">+</span><span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb78-85"><a href="#cb78-85" aria-hidden="true" tabindex="-1"></a>          <span class="fu">labs</span>(<span class="at">title =</span> comparison)</span>
<span id="cb78-86"><a href="#cb78-86" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PlotVolcano</span>(<span class="st">"condition_LG_vs_BR"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 950 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">PlotVolcano</span>(<span class="st">"condition_LV_vs_BR"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using 'apeglm' for LFC shrinkage. If used in published research, please cite:
    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for
    sequence count data: removing the noise and preserving large differences.
    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 635 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-51-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="go-enrichment-analysis" class="level4">
<h4 class="anchored" data-anchor-id="go-enrichment-analysis">GO enrichment analysis</h4>
<p>Look at genes overexpressed in the brain compared to both lung and liver.</p>
<p>Identify genes that are significantly downregulated (negative fold change) in lung samples compared to brain samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink the log2 fold changes using the apeglm method to stabilise the estimates,       especially for genes with low counts</span></span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>brainlung <span class="ot">=</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="st">"condition_LG_vs_BR"</span>, <span class="at">type=</span><span class="st">"apeglm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Covert the DESeq2 object to a data frame</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter genes for signficance and log2 fold change</span></span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj<span class="sc">&lt;</span><span class="fl">0.001</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;=</span> <span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract row names of filtered data frame</span></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>()</span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brainlung) <span class="co"># The output is a vector of gene IDs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENSMUSG00000000142" "ENSMUSG00000001247" "ENSMUSG00000001864"
[4] "ENSMUSG00000002489" "ENSMUSG00000002985" "ENSMUSG00000003418"</code></pre>
</div>
</div>
<p>Identify genes that are significantly downregulated (negative fold change) in liver samples compared to brain samples.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Shrink the log2 fold changes using the apeglm method to stabilise the estimates,       especially for genes with low counts</span></span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>brainliver <span class="ot">=</span> <span class="fu">lfcShrink</span>(dds, <span class="at">coef=</span><span class="st">"condition_LV_vs_BR"</span>, <span class="at">type=</span><span class="st">"apeglm"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Covert the DESeq2 object to a data frame</span></span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter genes for signficance and log2 fold change</span></span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(padj<span class="sc">&lt;</span><span class="fl">0.001</span> <span class="sc">&amp;</span> log2FoldChange <span class="sc">&lt;=</span> <span class="sc">-</span><span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Extract row names of filtered data frame</span></span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>()</span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(brainliver) <span class="co"># The output is a vector of gene IDs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENSMUSG00000000142" "ENSMUSG00000000159" "ENSMUSG00000000296"
[4] "ENSMUSG00000000416" "ENSMUSG00000000489" "ENSMUSG00000000530"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(brainlung) <span class="co"># 366 genes are downregulated in LG compared to BR</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 366</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(brainliver) <span class="co"># 686 genes are downregulated in LV compared to BR</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 686</code></pre>
</div>
</div>
<p>Get the genes that are downregulated in both the lung and liver compared to the brain.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="co"># interest() returns the gene IDs that are common to both vectors</span></span>
<span id="cb93-2"><a href="#cb93-2" aria-hidden="true" tabindex="-1"></a>brain_enriched <span class="ot">=</span> <span class="fu">intersect</span>(brainlung,brainliver) <span class="sc">%&gt;%</span> </span>
<span id="cb93-3"><a href="#cb93-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove duplicates</span></span>
<span id="cb93-4"><a href="#cb93-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span>
<span id="cb93-5"><a href="#cb93-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb93-6"><a href="#cb93-6" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(brain_enriched) <span class="co"># 209 genes are downregulated in LG and LV compared to the brain</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 209</code></pre>
</div>
</div>
<p>Perform GO analysis to see what kind of gene networks they belong to.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb95"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb95-1"><a href="#cb95-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Perform GO enrichment analysis using the brain enriched genes</span></span>
<span id="cb95-2"><a href="#cb95-2" aria-hidden="true" tabindex="-1"></a>go_enrich <span class="ot">=</span> <span class="fu">gost</span>(brain_enriched, <span class="at">organism =</span> <span class="st">"mmusculus"</span>)  <span class="co"># Specify the organism is a mouse</span></span>
<span id="cb95-3"><a href="#cb95-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Visualise result of GO enrichment analysis</span></span>
<span id="cb95-4"><a href="#cb95-4" aria-hidden="true" tabindex="-1"></a><span class="fu">gostplot</span>(go_enrich, <span class="at">interactive =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="plotly html-widget html-fill-item" id="htmlwidget-6561f66661cb46a4f9ad" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-6561f66661cb46a4f9ad">{"x":{"data":[{"x":[148.73955719212594,147.93725436759033],"y":[1.3016791398310068,1.3016791398310068],"text":["CORUM:3143 (3) <br> Sos1-Abi1-Eps8 complex <br> 4.993e-02","CORUM:141 (3) <br> Sos1-Abi1-Eps8 complex <br> 4.993e-02"],"key":["CORUM:3143","CORUM:141"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(102,170,0,1)","opacity":0.80000000000000004,"size":7.7180145958336093,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(102,170,0,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"CORUM","legendgroup":"CORUM","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[79.300658338305112,65.133601600388999,78.542202189599095,52.453319829472768,79.019376587791626,54.394666196856058,54.841703264636429,82.254116718696778,52.696929916971051,80.586517769223946,62.602065845767576,56.238066240399831,79.715046631472319,57.61433766255513,78.682843064855845,77.628036500430241,57.933291076083826,80.498617222188471,90.253066498924198,83.243625733896025,78.559782299006187,78.549736522202139,54.982344139893179,88.962184179603341,51.659703461952553,83.005038534799766,94.077996017067463,68.501448273947858,65.131090156187994,68.581814488380289,53.332325299827424,45.94114501624523,62.40868464228955,80.573960548218878,62.531745408139201,86.483388753203215,52.732090135785242,52.267472958597779,71.266548339263522,51.760161229993095,58.62393823136248,110.76654273280097,107.9989312232843,105.38200636582842,79.333307112918291,61.296114861240653,54.984855584094184,62.607088734169601,56.240577684600851,82.834260329130856,46.64937228093099,78.861155603127798,79.305681226707151,52.435739720065669,52.739624468388286,72.429347004332683,71.957195494542177,62.722615167416215,79.330795668717272,56.682591863979191,67.592305473181042,78.808415274906508,66.050278733758859,51.662214906153565,79.33581855711931,45.474016394856761,46.232472543562778,57.825298975440248,79.19517768186256,61.281046196034573,86.480877309002196,78.469370307769708,95.652671531102811,46.739784272167469,78.552247966403158,78.685354509056864,80.312770351313489,52.747158800991322,91.717238468114957,63.749795845630658,78.544713633800114,87.661256083478449,80.589029213424951,46.885448035826236,91.714727023913937,66.118087727186222,79.747695406085484,52.237335628185619,79.707512298869275,78.93901037335921,52.239847072386631,85.644566390064753,78.697911730061918,64.819671075262335,62.619645955174676,78.710468951066986,78.562293743207206,57.631917771962222,84.197974530281087,69.066523219175849,97.242415710344261,79.014353699389602,62.707546502210135,52.935517116067317,61.846121141262572,102.83289050179989,55.509747422105967,63.398193657488804,81.588584005428245,73.037116500977902,72.406744006523567,93.942378030212751,54.83919182043541,53.334836744028436,62.750241053627363,78.858644158926779,53.475477619285186,101.05478800748247,51.664726350354584,97.066614616273313,62.639737508782773,52.71451002637815,54.510192630102672,90.52932536103566,51.845550332827536,54.999924249300264,83.374220832348726,82.495215361994056,94.130736345288753,91.888016673783866,46.270144206577982,83.550021926419646],"y":[11.330652618501608,10.42304301609707,9.8509608007562335,9.1587097196242038,9.1176199658540096,8.217427447031616,8.1821126181089028,7.5983125314845621,7.3387830244712999,7.078031786060266,6.8988336301386601,6.8988336301386601,6.3845717693244044,6.3741302875258308,6.1962573464044439,5.9195198295745675,5.8852315804073703,5.7560648223573683,5.5877065190708697,5.4849093325774625,5.4005474134351443,5.3792381470454087,5.3414535189489136,5.2651383846388011,4.9913164888982884,4.8204071240010791,4.7983127276615294,4.7810563232521472,4.622149762513895,4.6070576981092275,4.5431285521135223,4.4264392510243598,4.41853341296141,4.3259468568433466,4.321658282476065,4.2877950449926336,4.1092042920408796,4.0603201404012337,3.9770206315336036,3.9545052286901834,3.9078014020313305,3.834238399664581,3.7596955307606819,3.7341010188589787,3.6738536622554823,3.6362227038052723,3.4609575442886977,3.4588609786078868,3.4588609786078868,3.3741944845640739,3.3741944845640739,3.3257562820054662,3.3055970901055756,3.3022101688768424,3.2873364365040616,3.2751558516021562,3.2674725712555266,3.2624948064528274,3.25932125971286,3.2343877439420465,3.1463367646074203,3.1434609788457695,3.0634414867022897,3.0265829624381215,2.9862997222282712,2.977967755940691,2.9523156445298682,2.8412080888659355,2.8409049566410762,2.8143407365261446,2.7803404018967588,2.7217567110864747,2.7096126286866702,2.6866804058899798,2.6859441531511461,2.6666507729200255,2.6541838014581254,2.6200077052981121,2.6110583703515786,2.6102583575693763,2.5570892024505967,2.4632149073032124,2.4484911673032674,2.34245415092317,2.3298585650806536,2.3153705251005787,2.3129668413861721,2.2641849275930235,2.2615766884402992,2.2478237359688253,2.2296185835826416,2.2257392037384855,2.1338503467002545,2.0634094124038245,2.0031860627172295,1.9877005356955944,1.9691297496106575,1.9635260057465243,1.9629652104523654,1.9629652104523654,1.9629652104523654,1.9625841821530599,1.9531652296381377,1.9459240062435947,1.9323468278008529,1.903341976512114,1.8682729209585256,1.8432697548670041,1.8411659056240217,1.8069670531474342,1.7909200422965375,1.7851563637211993,1.7543744092223748,1.7365039540170122,1.7283252110541916,1.6877986820985627,1.6803724543893044,1.6759059065665394,1.6573532105570723,1.5757483727378647,1.5214422799896719,1.5071400388020144,1.4088246644505242,1.3969166014015262,1.3635606195330749,1.3346614130843961,1.3225849023511735,1.3225849023511735,1.3155097405712199,1.3155097405712199,1.3069898295026223,1.3035673129363867],"text":["GO:0048856 (6269) <br> anatomical structure development <br> 4.670e-12","GO:0032502 (6854) <br> developmental process <br> 3.775e-11","GO:0048513 (3260) <br> animal organ development <br> 1.409e-10","GO:0007275 (4879) <br> multicellular organism development <br> 6.939e-10","GO:0048731 (4117) <br> system development <br> 7.627e-10","GO:0009653 (2761) <br> anatomical structure morphogenesis <br> 6.061e-09","GO:0009888 (2122) <br> tissue development <br> 6.575e-09","GO:0055085 (1525) <br> transmembrane transport <br> 2.522e-08","GO:0007399 (2529) <br> nervous system development <br> 4.584e-08","GO:0051239 (3165) <br> regulation of multicellular organismal process <br> 8.355e-08","GO:0023051 (3616) <br> regulation of signaling <br> 1.262e-07","GO:0010646 (3616) <br> regulation of cell communication <br> 1.262e-07","GO:0050793 (2698) <br> regulation of developmental process <br> 4.125e-07","GO:0015711 (464) <br> organic anion transport <br> 4.225e-07","GO:0048583 (3987) <br> regulation of response to stimulus <br> 6.364e-07","GO:0046942 (384) <br> carboxylic acid transport <br> 1.204e-06","GO:0015849 (386) <br> organic acid transport <br> 1.302e-06","GO:0051179 (5364) <br> localization <br> 1.754e-06","GO:0072359 (1216) <br> circulatory system development <br> 2.584e-06","GO:0060429 (1309) <br> epithelium development <br> 3.274e-06","GO:0048522 (6078) <br> positive regulation of cellular process <br> 3.976e-06","GO:0048518 (6589) <br> positive regulation of biological process <br> 4.176e-06","GO:0009966 (3005) <br> regulation of signal transduction <br> 4.556e-06","GO:0071702 (2419) <br> organic substance transport <br> 5.431e-06","GO:0006810 (4397) <br> transport <br> 1.020e-05","GO:0060322 (778) <br> head development <br> 1.512e-05","GO:0098739 (201) <br> import across plasma membrane <br> 1.591e-05","GO:0035239 (927) <br> tube morphogenesis <br> 1.656e-05","GO:0032501 (9132) <br> multicellular organismal process <br> 2.387e-05","GO:0035295 (1181) <br> tube development <br> 2.471e-05","GO:0008283 (2168) <br> cell population proliferation <br> 2.863e-05","GO:0000902 (1045) <br> cell morphogenesis <br> 3.746e-05","GO:0022008 (1907) <br> neurogenesis <br> 3.815e-05","GO:0051234 (4696) <br> establishment of localization <br> 4.721e-05","GO:0022603 (904) <br> regulation of anatomical structure morphogenesis <br> 4.768e-05","GO:0065008 (3134) <br> regulation of biological quality <br> 5.155e-05","GO:0007417 (1006) <br> central nervous system development <br> 7.777e-05","GO:0007166 (2766) <br> cell surface receptor signaling pathway <br> 8.703e-05","GO:0042127 (1799) <br> regulation of cell population proliferation <br> 1.054e-04","GO:0006869 (459) <br> lipid transport <br> 1.110e-04","GO:0016477 (1538) <br> cell migration <br> 1.237e-04","GO:2000026 (1550) <br> regulation of multicellular organismal development <br> 1.465e-04","GO:1905039 (159) <br> carboxylic acid transmembrane transport <br> 1.739e-04","GO:1903825 (160) <br> organic acid transmembrane transport <br> 1.845e-04","GO:0048870 (1760) <br> cell motility <br> 2.119e-04","GO:0021543 (208) <br> pallium development <br> 2.311e-04","GO:0009967 (1613) <br> positive regulation of signal transduction <br> 3.460e-04","GO:0023056 (1990) <br> positive regulation of signaling <br> 3.476e-04","GO:0010647 (1990) <br> positive regulation of cell communication <br> 3.476e-04","GO:0060249 (271) <br> anatomical structure homeostasis <br> 4.225e-04","GO:0001894 (271) <br> tissue homeostasis <br> 4.225e-04","GO:0048667 (647) <br> cell morphogenesis involved in neuron differentiation <br> 4.723e-04","GO:0048858 (721) <br> cell projection morphogenesis <br> 4.948e-04","GO:0007267 (1734) <br> cell-cell signaling <br> 4.986e-04","GO:0007420 (723) <br> brain development <br> 5.160e-04","GO:0042908 (42) <br> xenobiotic transport <br> 5.307e-04","GO:0042592 (1835) <br> homeostatic process <br> 5.402e-04","GO:0030154 (4777) <br> cell differentiation <br> 5.464e-04","GO:0048869 (4778) <br> cellular developmental process <br> 5.504e-04","GO:0010876 (519) <br> lipid localization <br> 5.829e-04","GO:0034330 (814) <br> cell junction organization <br> 7.139e-04","GO:0048646 (1221) <br> anatomical structure formation involved in morphogenesis <br> 7.187e-04","GO:0032989 (824) <br> cellular anatomical entity morphogenesis <br> 8.641e-04","GO:0006811 (1239) <br> monoatomic ion transport <br> 9.406e-04","GO:0048871 (912) <br> multicellular organismal-level homeostasis <br> 1.032e-03","GO:0000165 (758) <br> MAPK cascade <br> 1.052e-03","GO:0001568 (761) <br> blood vessel development <br> 1.116e-03","GO:0015801 (12) <br> aromatic amino acid transport <br> 1.441e-03","GO:0048812 (699) <br> neuron projection morphogenesis <br> 1.442e-03","GO:0021537 (306) <br> telencephalon development <br> 1.533e-03","GO:0065007 (13527) <br> biological regulation <br> 1.658e-03","GO:0048468 (3202) <br> cell development <br> 1.898e-03","GO:0120039 (714) <br> plasma membrane bounded cell projection morphogenesis <br> 1.952e-03","GO:0001944 (793) <br> vasculature development <br> 2.057e-03","GO:0048519 (5966) <br> negative regulation of biological process <br> 2.061e-03","GO:0048584 (2359) <br> positive regulation of response to stimulus <br> 2.155e-03","GO:0051094 (1480) <br> positive regulation of developmental process <br> 2.217e-03","GO:0007423 (650) <br> sensory organ development <br> 2.399e-03","GO:0089718 (54) <br> amino acid import across plasma membrane <br> 2.449e-03","GO:0031175 (1131) <br> neuron projection development <br> 2.453e-03","GO:0048514 (657) <br> blood vessel morphogenesis <br> 2.773e-03","GO:0070887 (3162) <br> cellular response to chemical stimulus <br> 3.442e-03","GO:0051240 (1804) <br> positive regulation of multicellular organismal process <br> 3.560e-03","GO:0002009 (536) <br> morphogenesis of an epithelium <br> 4.545e-03","GO:0089709 (5) <br> L-histidine transmembrane transport <br> 4.679e-03","GO:0033036 (3092) <br> macromolecule localization <br> 4.838e-03","GO:0050808 (539) <br> synapse organization <br> 4.864e-03","GO:0007154 (7582) <br> cell communication <br> 5.443e-03","GO:0050789 (13144) <br> regulation of biological process <br> 5.475e-03","GO:0048699 (1651) <br> generation of neurons <br> 5.652e-03","GO:0007155 (1465) <br> cell adhesion <br> 5.894e-03","GO:0061564 (548) <br> axon development <br> 5.946e-03","GO:0048589 (785) <br> developmental growth <br> 7.348e-03","GO:0032328 (18) <br> alanine transport <br> 8.642e-03","GO:0030001 (884) <br> metal ion transport <br> 9.927e-03","GO:0048598 (648) <br> embryonic morphogenesis <br> 1.029e-02","GO:0048523 (5599) <br> negative regulation of cellular process <br> 1.074e-02","GO:0015718 (197) <br> monocarboxylic acid transport <br> 1.088e-02","GO:0060856 (19) <br> establishment of blood-brain barrier <br> 1.089e-02","GO:0035633 (19) <br> maintenance of blood-brain barrier <br> 1.089e-02","GO:0150104 (19) <br> transport across blood-brain barrier <br> 1.089e-02","GO:0048729 (651) <br> tissue morphogenesis <br> 1.090e-02","GO:0030111 (309) <br> regulation of Wnt signaling pathway <br> 1.114e-02","GO:0007507 (653) <br> heart development <br> 1.133e-02","GO:0021766 (107) <br> hippocampus development <br> 1.169e-02","GO:1902531 (1722) <br> regulation of intracellular signal transduction <br> 1.249e-02","GO:0010232 (20) <br> vascular transport <br> 1.354e-02","GO:0030900 (447) <br> forebrain development <br> 1.435e-02","GO:0051897 (204) <br> positive regulation of phosphatidylinositol 3-kinase/protein kinase B signal transduction <br> 1.442e-02","GO:0043408 (670) <br> regulation of MAPK cascade <br> 1.560e-02","GO:0042886 (385) <br> amide transport <br> 1.618e-02","GO:0098657 (916) <br> import into cell <br> 1.640e-02","GO:0009887 (1094) <br> animal organ morphogenesis <br> 1.760e-02","GO:0008284 (1097) <br> positive regulation of cell population proliferation <br> 1.834e-02","GO:0030178 (160) <br> negative regulation of Wnt signaling pathway <br> 1.869e-02","GO:0048666 (1287) <br> neuron development <br> 2.052e-02","GO:0008610 (686) <br> lipid biosynthetic process <br> 2.088e-02","GO:1901700 (2077) <br> response to oxygen-containing compound <br> 2.109e-02","GO:0006812 (1022) <br> monoatomic cation transport <br> 2.201e-02","GO:0141124 (1794) <br> intracellular signaling cassette <br> 2.656e-02","GO:0030030 (1706) <br> cell projection organization <br> 3.010e-02","GO:0007409 (481) <br> axonogenesis <br> 3.111e-02","GO:0009719 (1935) <br> response to endogenous stimulus <br> 3.901e-02","GO:0072657 (644) <br> protein localization to membrane <br> 4.009e-02","GO:0006915 (2050) <br> apoptotic process <br> 4.330e-02","GO:0009987 (20967) <br> cellular process <br> 4.627e-02","GO:0060485 (297) <br> mesenchyme development <br> 4.758e-02","GO:0060070 (297) <br> canonical Wnt signaling pathway <br> 4.758e-02","GO:0098773 (133) <br> skin epidermis development <br> 4.836e-02","GO:0090090 (133) <br> negative regulation of canonical Wnt signaling pathway <br> 4.836e-02","GO:0001654 (431) <br> eye development <br> 4.932e-02","GO:0060562 (363) <br> epithelial tube morphogenesis <br> 4.971e-02"],"key":["GO:0048856","GO:0032502","GO:0048513","GO:0007275","GO:0048731","GO:0009653","GO:0009888","GO:0055085","GO:0007399","GO:0051239","GO:0023051","GO:0010646","GO:0050793","GO:0015711","GO:0048583","GO:0046942","GO:0015849","GO:0051179","GO:0072359","GO:0060429","GO:0048522","GO:0048518","GO:0009966","GO:0071702","GO:0006810","GO:0060322","GO:0098739","GO:0035239","GO:0032501","GO:0035295","GO:0008283","GO:0000902","GO:0022008","GO:0051234","GO:0022603","GO:0065008","GO:0007417","GO:0007166","GO:0042127","GO:0006869","GO:0016477","GO:2000026","GO:1905039","GO:1903825","GO:0048870","GO:0021543","GO:0009967","GO:0023056","GO:0010647","GO:0060249","GO:0001894","GO:0048667","GO:0048858","GO:0007267","GO:0007420","GO:0042908","GO:0042592","GO:0030154","GO:0048869","GO:0010876","GO:0034330","GO:0048646","GO:0032989","GO:0006811","GO:0048871","GO:0000165","GO:0001568","GO:0015801","GO:0048812","GO:0021537","GO:0065007","GO:0048468","GO:0120039","GO:0001944","GO:0048519","GO:0048584","GO:0051094","GO:0007423","GO:0089718","GO:0031175","GO:0048514","GO:0070887","GO:0051240","GO:0002009","GO:0089709","GO:0033036","GO:0050808","GO:0007154","GO:0050789","GO:0048699","GO:0007155","GO:0061564","GO:0048589","GO:0032328","GO:0030001","GO:0048598","GO:0048523","GO:0015718","GO:0060856","GO:0035633","GO:0150104","GO:0048729","GO:0030111","GO:0007507","GO:0021766","GO:1902531","GO:0010232","GO:0030900","GO:0051897","GO:0043408","GO:0042886","GO:0098657","GO:0009887","GO:0008284","GO:0030178","GO:0048666","GO:0008610","GO:1901700","GO:0006812","GO:0141124","GO:0030030","GO:0007409","GO:0009719","GO:0072657","GO:0006915","GO:0009987","GO:0060485","GO:0060070","GO:0098773","GO:0090090","GO:0001654","GO:0060562"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(255,153,0,1)","opacity":0.80000000000000004,"size":[21.328701751057359,21.425677032812782,20.600882389716016,21.053308382885966,20.864226972468124,20.410891339422463,20.105330786772971,19.713555709857136,20.309637264065447,20.567218733002786,20.718327126675387,20.718327126675387,20.384322580975084,18.214664035347667,20.828265580083379,17.961679248961399,17.968684056943637,21.157935465906252,19.439378205030948,19.529140283062222,21.294944028415323,21.382882501585598,20.508006265108179,20.258097580088869,20.937736855395432,18.884021439413772,17.059909601881486,19.10432153871146,21.734036327290926,19.403663741365257,20.130438904075231,19.253155187873404,19.979681740149239,21.010922982113723,19.072929603778331,20.555999759703631,19.206065496864341,20.41297214249937,19.910696974658698,18.200300109965355,19.723742560374109,19.73306396628913,16.717888898684524,16.727154664841223,19.884686769488304,17.109125785991207,19.780761776682866,20.029907540636515,20.029907540636515,17.483546122126494,17.483546122126494,18.648689983393385,18.787355432242641,19.867000455694551,18.790885633929687,14.571784246357929,19.934174664443724,21.029896765233428,21.030128864275088,18.362347375265848,18.941197134781184,19.444389654278609,18.956593895682929,19.462249447371921,19.083945550207464,18.851004453529015,18.856016803374629,12.058806804819245,18.747807006360492,17.652054395439823,22.147855770297312,20.580456506606325,18.77491552544776,18.908186071536839,21.274620612141288,20.228916694392645,19.677558310711198,18.654639942998166,15.008384021416694,19.350612301236584,18.668407940029581,20.566138165724581,19.913987772666282,18.404562661748759,9.700176674170466,20.540609330678421,18.411860832280251,21.534762819396946,22.117920782678972,19.808573485961332,19.665296933776261,18.433492686902198,18.895360528242147,12.947850468301544,19.044921758323945,18.650676627869341,21.20506702705476,17.030925456144942,13.059967644679528,13.059967644679528,13.059967644679528,18.656616635122333,17.665500266604447,18.660560141784572,16.118438885246281,19.858741250062707,13.165094704134018,18.165117717346547,17.081231235176517,18.693559386636259,17.965186633900949,19.089414393008106,19.309698562268999,19.313071133340067,16.727154664841223,19.508541196334715,18.723797222092436,20.080197546425548,19.225618918894778,19.907396343450667,19.847632302417448,18.262266882511316,19.99688315058361,18.642709972404521,20.064835818205584,22.598716152232853,17.610830123528174,17.610830123528174,16.451161238395375,16.451161238395375,18.116568226369793,17.885621674177841],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(255,153,0,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"GO:BP","legendgroup":"GO:BP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[39.646892040571721,40.711967674408982,34.469720150622209,34.439576500607942,33.691009191920443,37.996527202290878,34.28383430886759,37.999039173125389,34.46720817978769,35.213263517640669,34.959554463353967,33.128327724987543,34.811348184117179,41.27967308301092,36.951547335129789,41.269625199672831,37.265543689445018,34.29137022137116,37.268055660279536,34.472232121456727,38.760166335985517,33.213734733361285,34.406920879759163,38.559208669223764,34.974626288361101,40.179429857490355,38.00657508562896,34.972114317526575],"y":[13.708698611578779,11.153251590529941,10.625338289372182,10.168320938564658,9.5620199758893953,8.8491493892735757,5.9590344559521089,5.6125816134630195,4.8037112798388168,4.0407025246009383,4.017462690935325,3.6555934850088598,3.6089454454793959,3.1804662049622077,2.7966894391087362,2.784045284038398,2.7613756889228736,2.6718348997710621,2.4656417276660152,1.9672023149156543,1.9462831308206199,1.9397573141042481,1.9047698867888667,1.8255128198165256,1.5793640463118674,1.5664522005427455,1.487097197841565,1.4647933514097278],"text":["GO:0071944 (6862) <br> cell periphery <br> 1.956e-14","GO:0098590 (1408) <br> plasma membrane region <br> 7.027e-12","GO:0016324 (461) <br> apical plasma membrane <br> 2.370e-11","GO:0016020 (10191) <br> membrane <br> 6.787e-11","GO:0005886 (6367) <br> plasma membrane <br> 2.741e-10","GO:0045177 (562) <br> apical part of cell <br> 1.415e-09","GO:0009925 (312) <br> basal plasma membrane <br> 1.099e-06","GO:0045178 (331) <br> basal part of cell <br> 2.440e-06","GO:0016323 (275) <br> basolateral plasma membrane <br> 1.571e-05","GO:0031012 (499) <br> extracellular matrix <br> 9.105e-05","GO:0030312 (501) <br> external encapsulating structure <br> 9.606e-05","GO:0005576 (2554) <br> extracellular region <br> 2.210e-04","GO:0030054 (2260) <br> cell junction <br> 2.461e-04","GO:0120025 (2460) <br> plasma membrane bounded cell projection <br> 6.600e-04","GO:0036477 (1103) <br> somatodendritic compartment <br> 1.597e-03","GO:0110165 (22653) <br> cellular anatomical entity <br> 1.644e-03","GO:0042995 (2564) <br> cell projection <br> 1.732e-03","GO:0009986 (1037) <br> cell surface <br> 2.129e-03","GO:0043005 (1525) <br> neuron projection <br> 3.423e-03","GO:0016327 (27) <br> apicolateral plasma membrane <br> 1.078e-02","GO:0070161 (729) <br> anchoring junction <br> 1.132e-02","GO:0005615 (1836) <br> extracellular space <br> 1.149e-02","GO:0015629 (503) <br> actin cytoskeleton <br> 1.245e-02","GO:0062023 (371) <br> collagen-containing extracellular matrix <br> 1.494e-02","GO:0030425 (783) <br> dendrite <br> 2.634e-02","GO:0097447 (785) <br> dendritic tree <br> 2.714e-02","GO:0045202 (1641) <br> synapse <br> 3.258e-02","GO:0030424 (801) <br> axon <br> 3.429e-02"],"key":["GO:0071944","GO:0098590","GO:0016324","GO:0016020","GO:0005886","GO:0045177","GO:0009925","GO:0045178","GO:0016323","GO:0031012","GO:0030312","GO:0005576","GO:0030054","GO:0120025","GO:0036477","GO:0110165","GO:0042995","GO:0009986","GO:0043005","GO:0016327","GO:0070161","GO:0005615","GO:0015629","GO:0062023","GO:0030425","GO:0097447","GO:0045202","GO:0030424"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(16,150,24,1)","opacity":0.80000000000000004,"size":[21.426941484964697,19.61743965611889,18.206066086407073,21.85055129779693,21.345600897953894,18.466384753046551,17.678803404595058,17.759921323909662,17.503982730637716,18.310709120711632,18.315973611085738,20.321016250100502,20.178985844189427,20.277595677333228,19.319786513354209,22.677165354330711,20.325534515348938,19.24365224781301,19.713555709857136,13.75799563321786,18.801413011933672,19.934819746309252,18.321215232823697,17.915151124757344,18.892132005694162,18.895360528242147,19.80132182586361,18.920872173555651],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(16,150,24,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"GO:CC","legendgroup":"GO:CC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[6.4404665544835895,5.3981624707105755,11.521385015863849,17.704885387548579,5.5764843741994525,11.456084037121443,5.7447599732664205,5.5488570370392036,8.1684309150518644,16.230590213633494,7.9825742832465547,5.8276419847471663,5.5036486671406148,28.258528182763534,28.250993454447102,21.884148027062544,7.9875974354575092,29.443992104548744,28.248481878341625,9.9365804933077726,28.785959164913731,8.211127708844975,5.3655119813393721],"y":[8.4060876790349131,8.1404939740592255,6.4182268937921325,6.2317560877051816,6.2036095130202265,5.903008837223445,5.6455351577554982,5.4501086410795798,4.903178724462439,3.7913183766296568,3.5141281851042074,3.4459783729647535,3.0612662515352982,2.8454643687559367,2.4764796902212884,2.3221616635838274,2.1471055310929987,1.6461227353838663,1.5680422876214031,1.5059640646588468,1.5026226616096798,1.3738620070247876,1.3478185736385011],"text":["GO:0008514 (264) <br> organic anion transmembrane transporter activity <br> 3.926e-09","GO:0005215 (1189) <br> transporter activity <br> 7.236e-09","GO:0022857 (1087) <br> transmembrane transporter activity <br> 3.817e-07","GO:0046943 (189) <br> carboxylic acid transmembrane transporter activity <br> 5.865e-07","GO:0005342 (190) <br> organic acid transmembrane transporter activity <br> 6.257e-07","GO:0022804 (435) <br> active transmembrane transporter activity <br> 1.250e-06","GO:0005515 (10460) <br> protein binding <br> 2.262e-06","GO:0005319 (179) <br> lipid transporter activity <br> 3.547e-06","GO:0015291 (290) <br> secondary active transmembrane transporter activity <br> 1.250e-05","GO:0042910 (38) <br> xenobiotic transmembrane transporter activity <br> 1.617e-04","GO:0015173 (10) <br> aromatic amino acid transmembrane transporter activity <br> 3.061e-04","GO:0008028 (69) <br> monocarboxylic acid transmembrane transporter activity <br> 3.581e-04","GO:0005290 (4) <br> L-histidine transmembrane transporter activity <br> 8.684e-04","GO:0140333 (14) <br> glycerophospholipid flippase activity <br> 1.427e-03","GO:0140327 (17) <br> flippase activity <br> 3.338e-03","GO:0050839 (292) <br> cell adhesion molecule binding <br> 4.763e-03","GO:0015175 (42) <br> neutral L-amino acid transmembrane transporter activity <br> 7.127e-03","GO:1901618 (53) <br> organic hydroxy compound transmembrane transporter activity <br> 2.259e-02","GO:0140326 (28) <br> ATPase-coupled intramembrane lipid transporter activity <br> 2.704e-02","GO:0017147 (29) <br> Wnt-protein binding <br> 3.119e-02","GO:0140830 (2) <br> L-glutamine, sodium:proton antiporter activity <br> 3.143e-02","GO:0015318 (657) <br> inorganic molecular entity transmembrane transporter activity <br> 4.228e-02","GO:0005178 (143) <br> integrin binding <br> 4.489e-02"],"key":["GO:0008514","GO:0005215","GO:0022857","GO:0046943","GO:0005342","GO:0022804","GO:0005515","GO:0005319","GO:0015291","GO:0042910","GO:0015173","GO:0008028","GO:0005290","GO:0140333","GO:0140327","GO:0050839","GO:0015175","GO:1901618","GO:0140326","GO:0017147","GO:0140830","GO:0015318","GO:0005178"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(220,57,18,1)","opacity":0.80000000000000004,"size":[17.446969261783856,19.411926742312581,19.301790254335543,16.970947385124298,16.978597178450606,18.128888049401429,21.878108006865304,16.891883825567096,17.577805217626835,14.392913471576092,11.626277095618034,15.418459955353805,8.929031646366262,12.407605671852071,12.82781267352663,17.587329655016003,14.571784246357929,14.976496866448542,13.827468853955285,13.894051557160756,3.7795275590551185,18.668407940029581,16.560127115942691],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(220,57,18,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"GO:MF","legendgroup":"GO:MF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[117.25546392110152,117.31584229168888],"y":[1.6779078274736754,1.5373583064309611],"text":["KEGG:05200 (543) <br> Pathways in cancer <br> 2.099e-02","KEGG:05226 (150) <br> Gastric cancer <br> 2.902e-02"],"key":["KEGG:05200","KEGG:05226"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(221,68,119,1)","opacity":0.80000000000000004,"size":[18.421523189960372,16.63145461546215],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(221,68,119,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"KEGG","legendgroup":"KEGG","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[123.42077070780898,124.19219957590789,120.31746565210173,124.18214838544407,122.12416713797502],"y":[6.6760072082430799,6.2860346622113781,2.7017012226793353,2.1112089248183974,1.6650274879331972],"text":["REAC:R-MMU-425407 (207) <br> SLC-mediated transmembrane transport <br> 2.109e-07","REAC:R-MMU-382551 (628) <br> Transport of small molecules <br> 5.176e-07","REAC:R-MMU-352230 (29) <br> Amino acid transport across the plasma membrane <br> 1.987e-03","REAC:R-MMU-425393 (93) <br> Transport of inorganic cations/anions and amino acids/oligopeptides <br> 7.741e-03","REAC:R-MMU-556833 (570) <br> Metabolism of lipids <br> 2.163e-02"],"key":["REAC:R-MMU-425407","REAC:R-MMU-382551","REAC:R-MMU-352230","REAC:R-MMU-425393","REAC:R-MMU-556833"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(51,102,204,1)","opacity":0.80000000000000004,"size":[17.102208213102088,18.610296681163369,13.894051557160756,15.899100466410918,18.484782106384632],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(51,102,204,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"REAC","legendgroup":"REAC","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[130.38110637157206,136.69862640917222,136.8191993562119,136.78654418305533,136.81668741981522,130.38361830796873,136.78905611945197,135.244215235506,136.78152031026198,136.56046990735589,132.12690216725085,132.20728413194396,137.53258929286335,128.9141355159226,132.12439023085417,136.69611447277555,128.80109837807288,128.79858644167624,128.91664745231927,128.83375355122948,136.78403224665865,137.53007735646671,136.56549378014921,135.2944539634392,129.88374296503341,130.37608249877874,131.8531011000149,128.85887291519609,127.226114257367,127.6481195720059],"y":[4.3711437604363175,4.0548517837920199,3.8586030303728345,3.6864908591676366,3.6697321989753031,3.5131302378202665,3.4835034750940599,3.3577829771046641,2.9150487089915869,2.8228788380056646,2.7267594388493106,2.559170027731513,2.5268785436244858,2.5146136534564465,2.4436895808448282,2.3600946613107472,2.3116318573199544,2.2493283739510828,2.1611011804703062,2.099448038749383,1.9269990680719247,1.805209055105887,1.8019905711125441,1.6583799847109144,1.5228346119943565,1.4814765618024976,1.4023250749274241,1.3665604837065681,1.3451257850865932,1.310320282376227],"text":["TF:M07289 (15234) <br> Factor: GKLF; motif: NNNRGGNGNGGSN <br> 4.255e-05","TF:M10276_1 (9977) <br> Factor: Kaiso; motif: SARNYCTCGCGAGAN; match class: 1 <br> 8.813e-05","TF:M02933_1 (5832) <br> Factor: ZF5; motif: GYCGCGCARNGCNN; match class: 1 <br> 1.385e-04","TF:M00333 (14398) <br> Factor: ZF5; motif: NRNGNGCGCGCWN <br> 2.058e-04","TF:M02933 (9499) <br> Factor: ZF5; motif: GYCGCGCARNGCNN <br> 2.139e-04","TF:M07289_1 (10917) <br> Factor: GKLF; motif: NNNRGGNGNGGSN; match class: 1 <br> 3.068e-04","TF:M00333_1 (11183) <br> Factor: ZF5; motif: NRNGNGCGCGCWN; match class: 1 <br> 3.285e-04","TF:M10375 (10720) <br> Factor: Sp1; motif: GGNGGGGGNGGGGGMGGGGCNGGG <br> 4.387e-04","TF:M00716 (14823) <br> Factor: ZF5; motif: GSGCGCGR <br> 1.216e-03","TF:M07436 (7691) <br> Factor: WT1; motif: NNGGGNGGGSGN <br> 1.504e-03","TF:M02023_1 (4130) <br> Factor: MAZ; motif: NKGGGAGGGGRGGR; match class: 1 <br> 1.876e-03","TF:M07326_1 (26) <br> Factor: Mef-2A; motif: GNNYTAAAWATAR; match class: 1 <br> 2.759e-03","TF:M02835_1 (6572) <br> Factor: Zic1; motif: NNCCCCCGGGGGGG; match class: 1 <br> 2.972e-03","TF:M00803 (10412) <br> Factor: E2F; motif: GGCGSG <br> 3.058e-03","TF:M02023 (7941) <br> Factor: MAZ; motif: NKGGGAGGGGRGGR <br> 3.600e-03","TF:M10276 (13469) <br> Factor: Kaiso; motif: SARNYCTCGCGAGAN <br> 4.364e-03","TF:M10209_1 (10616) <br> Factor: E2F-1; motif: GNGGGCGGGRMN; match class: 1 <br> 4.879e-03","TF:M10209 (14210) <br> Factor: E2F-1; motif: GNGGGCGGGRMN <br> 5.632e-03","TF:M00803_1 (7031) <br> Factor: E2F; motif: GGCGSG; match class: 1 <br> 6.901e-03","TF:M02742 (3530) <br> Factor: E2F-2; motif: NNNANGGCGCGCNNN <br> 7.953e-03","TF:M00716_1 (12044) <br> Factor: ZF5; motif: GSGCGCGR; match class: 1 <br> 1.183e-02","TF:M02835 (9478) <br> Factor: Zic1; motif: NNCCCCCGGGGGGG <br> 1.566e-02","TF:M01118 (8071) <br> Factor: WT1; motif: SMCNCCNSC <br> 1.578e-02","TF:M00665 (7220) <br> Factor: Sp3; motif: ASMCTTGGGSRGGG <br> 2.196e-02","TF:M04662 (15957) <br> Factor: FOXN4; motif: NNWANNCGWMCGCGTCNNNNMT <br> 3.000e-02","TF:M07040 (12890) <br> Factor: GKLF; motif: NNRRGRRNGNSNNN <br> 3.300e-02","TF:M08819 (5785) <br> Factor: LKLF; motif: CNCCACCCS <br> 3.960e-02","TF:M02089 (11263) <br> Factor: E2F-3; motif: GGCGGGN <br> 4.300e-02","TF:M08867 (7232) <br> Factor: AP2; motif: GCCYGSGGSN <br> 4.517e-02","TF:M01240 (16194) <br> Factor: BEN; motif: CAGCGRNV <br> 4.894e-02"],"key":["TF:M07289","TF:M10276_1","TF:M02933_1","TF:M00333","TF:M02933","TF:M07289_1","TF:M00333_1","TF:M10375","TF:M00716","TF:M07436","TF:M02023_1","TF:M07326_1","TF:M02835_1","TF:M00803","TF:M02023","TF:M10276","TF:M10209_1","TF:M10209","TF:M00803_1","TF:M02742","TF:M00716_1","TF:M02835","TF:M01118","TF:M00665","TF:M04662","TF:M07040","TF:M08819","TF:M02089","TF:M08867","TF:M01240"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(85,116,166,1)","opacity":0.80000000000000004,"size":[22.271200406486955,21.828073175697263,21.249765668339915,22.212724500992163,21.775964998974256,21.923247347626511,21.948609489469991,21.904038993437013,22.242887463254103,21.550133761804936,20.86775637239236,13.685385010878013,21.380075098237505,21.873246206592913,21.584532734254186,22.143380514398363,21.893747316577215,22.199080490222887,21.453293403509363,20.6911234245858,22.026529889943095,21.773612447612759,21.601969179982547,21.481979131958287,22.3191024237445,22.097555432375856,21.240903848232236,21.956112500322757,21.48377346610371,22.334307522813916],"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(85,116,166,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"TF","legendgroup":"TF","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isNestedKey":false,"frame":null},{"x":[199.94446431401471],"y":[2.3130650713495498],"text":"WP:WP5298 (72) <br> Dravet syndrome Scn1a A1783V point mutation model <br> 4.863e-03","key":["WP:WP5298"],"type":"scatter","mode":"markers","marker":{"autocolorscale":false,"color":"rgba(0,153,198,1)","opacity":0.80000000000000004,"size":15.488196555088839,"symbol":"circle","line":{"width":1.8897637795275593,"color":"rgba(0,153,198,1)"}},"hoveron":"points","set":"SharedDataad3764ff","name":"WP","legendgroup":"WP","showlegend":true,"xaxis":"x","yaxis":"y","hoverinfo":"text","_isSimpleKey":true,"_isNestedKey":false,"frame":null},{"x":[2,30.122117653027573],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(220,57,18,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[32.633469470586746,42.82453514624185],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(16,150,24,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[45.335886963801016,113.60698612414703],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(255,153,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[116.1183379417062,117.54980847771492],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(221,68,119,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[120.06116029527409,124.42588975419193],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(51,102,204,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[126.9372415717511,137.72852033180286],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(85,116,166,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[140.23987214936201,145.36554120900027],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(34,170,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[147.87689302655946,149.58210091068213],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(102,170,0,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[152.09345272824129,196.98135511529389],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(153,0,153,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[199.49270693285305,200.00251135181756],"y":[-1,-1],"text":"","type":"scatter","mode":"lines","line":{"width":11.338582677165356,"color":"rgba(0,153,198,1)","dash":"solid"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[180],"y":[16.199999999999999],"text":"values above this threshold are capped","hovertext":"","textfont":{"size":7.559055118110237,"color":"rgba(190,190,190,1)"},"type":"scatter","mode":"text","hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null},{"x":[0,210],"y":[16,16],"text":"","type":"scatter","mode":"lines","line":{"width":0.75590551181102372,"color":"rgba(190,190,190,1)","dash":"dash"},"hoveron":"points","showlegend":false,"xaxis":"x","yaxis":"y","hoverinfo":"text","frame":null}],"layout":{"margin":{"t":29.283520132835207,"r":6.6417600664176026,"b":55.878166138820198,"l":61.037775010377771},"plot_bgcolor":"rgba(255,255,255,1)","paper_bgcolor":"rgba(255,255,255,1)","font":{"color":"rgba(0,0,0,1)","family":"","size":14.611872146118724},"xaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[0,210],"tickmode":"array","ticktext":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"tickvals":[16.061058826513786,37.729002308414294,79.471436543974022,116.83407320971057,122.243525024733,132.33288095177699,142.80270667918114,148.72949696862079,174.53740392176758,199.74760914233531],"categoryorder":"array","categoryarray":["GO:MF","GO:CC","GO:BP","KEGG","REAC","TF","MIRNA","CORUM","HP","WP"],"nticks":null,"ticks":"","tickcolor":null,"ticklen":3.6529680365296811,"tickwidth":0,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":10.62681610626816},"tickangle":-45,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.13283520132835203,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"y","title":"","hoverformat":".2f"},"yaxis":{"domain":[0,1],"automargin":true,"type":"linear","autorange":false,"range":[-1,18.5],"tickmode":"array","ticktext":["0","2","4","6","8","10","12","14",">16"],"tickvals":[0,2,4,6,8,10,12,14,16],"categoryorder":"array","categoryarray":["0","2","4","6","8","10","12","14",">16"],"nticks":null,"ticks":"outside","tickcolor":"rgba(190,190,190,1)","ticklen":3.6529680365296811,"tickwidth":0.66417600664176002,"showticklabels":true,"tickfont":{"color":"rgba(77,77,77,1)","family":"","size":11.68949771689498},"tickangle":-0,"showline":true,"linecolor":"rgba(190,190,190,1)","linewidth":0.66417600664176002,"showgrid":false,"gridcolor":null,"gridwidth":0,"zeroline":false,"anchor":"x","title":"","hoverformat":".2f"},"annotations":[{"text":"-log10(p-adj)","x":0,"y":0.5,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(0,0,0,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-90,"xanchor":"right","yanchor":"center","annotationType":"axis","xshift":-37.393109173931101},{"text":"query_1","x":0.5,"y":1,"showarrow":false,"ax":0,"ay":0,"font":{"color":"rgba(169,169,169,1)","family":"","size":13.283520132835205},"xref":"paper","yref":"paper","textangle":-0,"xanchor":"center","yanchor":"bottom"}],"shapes":[{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":1},{"type":"rect","fillcolor":null,"line":{"color":null,"width":0,"linetype":[]},"yref":"paper","xref":"paper","x0":0,"x1":1,"y0":0,"y1":24.973017849730184,"yanchor":1,"ysizemode":"pixel"}],"showlegend":false,"legend":{"bgcolor":"rgba(255,255,255,1)","bordercolor":"transparent","borderwidth":1.8897637795275593,"font":{"color":"rgba(0,0,0,1)","family":"","size":11.68949771689498}},"hovermode":"closest","barmode":"relative","dragmode":"zoom"},"config":{"doubleClick":"reset","modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"source":"A","attrs":{"d3b21f039bb1":{"x":{},"y":{},"text":{},"colour":{},"size":{},"alpha":{},"type":"scatter"},"d3b225e2915f":{"x":{},"xend":{},"y":{},"yend":{}},"d3b23db60767":{"x":{},"xend":{},"y":{},"yend":{}},"d3b279a01ad7":{"x":{},"xend":{},"y":{},"yend":{}},"d3b27e4257a2":{"x":{},"xend":{},"y":{},"yend":{}},"d3b235878d70":{"x":{},"xend":{},"y":{},"yend":{}},"d3b25656cf84":{"x":{},"xend":{},"y":{},"yend":{}},"d3b25d520f64":{"x":{},"xend":{},"y":{},"yend":{}},"d3b236709e19":{"x":{},"xend":{},"y":{},"yend":{}},"d3b21b9b973b":{"x":{},"xend":{},"y":{},"yend":{}},"d3b23e5b0a6":{"x":{},"xend":{},"y":{},"yend":{}},"d3b25cb06449":{"x":{},"y":{}},"d3b240882429":{"yintercept":{}}},"cur_data":"d3b21f039bb1","visdat":{"d3b21f039bb1":["function (y) ","x"],"d3b225e2915f":["function (y) ","x"],"d3b23db60767":["function (y) ","x"],"d3b279a01ad7":["function (y) ","x"],"d3b27e4257a2":["function (y) ","x"],"d3b235878d70":["function (y) ","x"],"d3b25656cf84":["function (y) ","x"],"d3b25d520f64":["function (y) ","x"],"d3b236709e19":["function (y) ","x"],"d3b21b9b973b":["function (y) ","x"],"d3b23e5b0a6":["function (y) ","x"],"d3b25cb06449":["function (y) ","x"],"d3b240882429":["function (y) ","x"]},"highlight":{"on":"plotly_click","off":"plotly_doubleclick","persistent":false,"dynamic":false,"color":null,"selectize":false,"defaultValues":null,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0,"ctGroups":["SharedDataad3764ff"]},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<p>Modify the <code>go_enrich</code> data frame as done earlier.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb96"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb96-1"><a href="#cb96-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Only include relevant columns</span></span>
<span id="cb96-2"><a href="#cb96-2" aria-hidden="true" tabindex="-1"></a>gp_mod <span class="ot">=</span> go_enrich<span class="sc">$</span>result[,<span class="fu">c</span>(<span class="st">"query"</span>, <span class="st">"source"</span>, <span class="st">"term_id"</span>,</span>
<span id="cb96-3"><a href="#cb96-3" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"term_name"</span>, <span class="st">"p_value"</span>, <span class="st">"query_size"</span>, </span>
<span id="cb96-4"><a href="#cb96-4" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"intersection_size"</span>, <span class="st">"term_size"</span>, </span>
<span id="cb96-5"><a href="#cb96-5" aria-hidden="true" tabindex="-1"></a>                                <span class="st">"effective_domain_size"</span>)]</span>
<span id="cb96-6"><a href="#cb96-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-7"><a href="#cb96-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate GeneRatio and BgRatio</span></span>
<span id="cb96-8"><a href="#cb96-8" aria-hidden="true" tabindex="-1"></a>gp_mod<span class="sc">$</span>GeneRatio <span class="ot">=</span> <span class="fu">paste0</span>(gp_mod<span class="sc">$</span>intersection_size,  <span class="st">"/"</span>, gp_mod<span class="sc">$</span>query_size)</span>
<span id="cb96-9"><a href="#cb96-9" aria-hidden="true" tabindex="-1"></a>gp_mod<span class="sc">$</span>BgRatio <span class="ot">=</span> <span class="fu">paste0</span>(gp_mod<span class="sc">$</span>term_size, <span class="st">"/"</span>, gp_mod<span class="sc">$</span>effective_domain_size)</span>
<span id="cb96-10"><a href="#cb96-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-11"><a href="#cb96-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename columns</span></span>
<span id="cb96-12"><a href="#cb96-12" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gp_mod) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"Cluster"</span>, <span class="st">"Category"</span>, <span class="st">"ID"</span>, <span class="st">"Description"</span>, <span class="st">"p.adjust"</span>, </span>
<span id="cb96-13"><a href="#cb96-13" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"query_size"</span>, <span class="st">"Count"</span>, <span class="st">"term_size"</span>, <span class="st">"effective_domain_size"</span>, </span>
<span id="cb96-14"><a href="#cb96-14" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"GeneRatio"</span>, <span class="st">"BgRatio"</span>)</span>
<span id="cb96-15"><a href="#cb96-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-16"><a href="#cb96-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Set row names to the GO term IDs</span></span>
<span id="cb96-17"><a href="#cb96-17" aria-hidden="true" tabindex="-1"></a><span class="fu">row.names</span>(gp_mod) <span class="ot">=</span> gp_mod<span class="sc">$</span>ID</span>
<span id="cb96-18"><a href="#cb96-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-19"><a href="#cb96-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter</span></span>
<span id="cb96-20"><a href="#cb96-20" aria-hidden="true" tabindex="-1"></a>gp_mod_bp <span class="ot">=</span> gp_mod <span class="sc">%&gt;%</span></span>
<span id="cb96-21"><a href="#cb96-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Category <span class="sc">==</span> <span class="st">"GO:BP"</span>)</span>
<span id="cb96-22"><a href="#cb96-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-23"><a href="#cb96-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Define as enrichResult object</span></span>
<span id="cb96-24"><a href="#cb96-24" aria-hidden="true" tabindex="-1"></a>gp_mod_bp_enrich  <span class="ot">=</span> <span class="fu">new</span>(<span class="st">"enrichResult"</span>, <span class="at">result =</span> gp_mod_bp)</span>
<span id="cb96-25"><a href="#cb96-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb96-26"><a href="#cb96-26" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(<span class="fu">head</span>(gp_mod_bp))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<colgroup>
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 22%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 3%">
<col style="width: 6%">
<col style="width: 14%">
<col style="width: 6%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th style="text-align: left;">Cluster</th>
<th style="text-align: left;">Category</th>
<th style="text-align: left;">ID</th>
<th style="text-align: left;">Description</th>
<th style="text-align: right;">p.adjust</th>
<th style="text-align: right;">query_size</th>
<th style="text-align: right;">Count</th>
<th style="text-align: right;">term_size</th>
<th style="text-align: right;">effective_domain_size</th>
<th style="text-align: left;">GeneRatio</th>
<th style="text-align: left;">BgRatio</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048856</td>
<td style="text-align: left;">anatomical structure development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">89</td>
<td style="text-align: right;">6269</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">89/176</td>
<td style="text-align: left;">6269/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0032502</td>
<td style="text-align: left;">developmental process</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">92</td>
<td style="text-align: right;">6854</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">92/176</td>
<td style="text-align: left;">6854/26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048513</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048513</td>
<td style="text-align: left;">animal organ development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">59</td>
<td style="text-align: right;">3260</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">59/176</td>
<td style="text-align: left;">3260/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0007275</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0007275</td>
<td style="text-align: left;">multicellular organism development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">73</td>
<td style="text-align: right;">4879</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">73/176</td>
<td style="text-align: left;">4879/26944</td>
</tr>
<tr class="odd">
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0048731</td>
<td style="text-align: left;">system development</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">66</td>
<td style="text-align: right;">4117</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">66/176</td>
<td style="text-align: left;">4117/26944</td>
</tr>
<tr class="even">
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">query_1</td>
<td style="text-align: left;">GO:BP</td>
<td style="text-align: left;">GO:0009653</td>
<td style="text-align: left;">anatomical structure morphogenesis</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">176</td>
<td style="text-align: right;">51</td>
<td style="text-align: right;">2761</td>
<td style="text-align: right;">26944</td>
<td style="text-align: left;">51/176</td>
<td style="text-align: left;">2761/26944</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Generate a bar plot to visualise the result of the GO enrichment analysis, showing the top 10 enriched GO terms.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>barplot_BR_OE <span class="ot">=</span> <span class="fu">barplot</span>(gp_mod_bp_enrich, <span class="at">showCategory =</span> <span class="dv">10</span>, <span class="at">font.size =</span> <span class="dv">12</span>) <span class="sc">+</span> </span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a>  ggplot2<span class="sc">::</span><span class="fu">facet_grid</span>(<span class="sc">~</span>Cluster) <span class="sc">+</span></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Brain-enriched genes"</span>, <span class="at">subtitle =</span> <span class="st">"Gene Ontology:Biological Process"</span>)  <span class="sc">+</span> </span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face=</span><span class="st">"bold"</span>))</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>barplot_BR_OE</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="embl_ebi_rna-seq_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p>This plot shows the 10 most enriched GO terms (ordered by <code>p.adjust</code>) in genes overexpressed in the brain compared to lung and liver.</p>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "î§‹";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>